<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><link href=https://canineglioma.verhaaklab.com/figures/A1_preload/ rel=canonical><meta name=author content="Samir B. Amin"><meta name=lang:clipboard.copy content="Copy to clipboard"><meta name=lang:clipboard.copied content="Copied to clipboard"><meta name=lang:search.language content=en><meta name=lang:search.pipeline.stopwords content=True><meta name=lang:search.pipeline.trimmer content=True><meta name=lang:search.result.none content="No matching documents"><meta name=lang:search.result.one content="1 matching document"><meta name=lang:search.result.other content="# matching documents"><meta name=lang:search.tokenizer content=[\s\-\.]+><link rel="shortcut icon" href=../../assets/images/logo.png><meta name=generator content="mkdocs-1.0.4, mkdocs-material-4.4.0"><title>Start Here &middot; Canine Glioma</title><link rel=stylesheet href=../../assets/stylesheets/application.4031d38b.css><link rel=stylesheet href=../../assets/stylesheets/application-palette.224b79ff.css><meta name=theme-color content=#2196f3><script src=../../assets/javascripts/modernizr.74668098.js></script><link href=https://fonts.gstatic.com rel=preconnect crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,400,400i,700|Fira+Code&display=fallback"><style>body,input{font-family:"IBM Plex Sans","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Fira Code","Courier New",Courier,monospace}</style><link rel=stylesheet href=../../assets/fonts/material-icons.css><link rel=stylesheet href=../../assets/css/katex.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css><script async src="https://www.googletagmanager.com/gtag/js?id=UA-64053990-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-64053990-2');
</script><meta content=https://canineglioma.verhaaklab.com/figures/A1_preload/ property=og:url><meta property=og:title content="Start Here · Canine Glioma"><meta property=og:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta property=og:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta property=og:image:alt content="Image alt"><meta property=og:image:type content=image/jpeg><meta property=og:image:width content=300><meta property=og:image:height content=300><meta name=twitter:card content=summary><meta name=twitter:site content=@roelverhaak><meta name=twitter:creator content=@roelverhaak><meta property=twitter:title content="Start Here · Canine Glioma"><meta name=twitter:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta name=twitter:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta name=twitter:image:alt content="Image alt"></head> <body dir=ltr data-md-color-primary=blue data-md-color-accent=blue> <svg class=md-svg> <defs> <!-- <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg> --> <svg xmlns=http://www.w3.org/2000/svg xmlns:xlink=http://www.w3.org/1999/xlink version=1.1 style=width:0;height:0;position:absolute;overflow:hidden;> <defs> <symbol viewbox="0 0 1024 998" aria-labelledby=dpsi-ant-github-title id=__github><title id=dpsi-ant-github-title>icon github</title><path fill=currentColor d="M0 512q0 166 95.5 298.5T343 996q6 1 10 1t6.5-1.5 4-3 2-5 .5-4.5V882q-37 4-66-.5t-45.5-14-29-23.5-17-25.5-9-24T194 780q-9-15-27-27.5t-27-20-2-14.5q50-26 113 66 34 51 119 30 10-41 40-70-116-21-172-86t-56-158q0-87 55-151-22-65 6-137 29-2 65 11.5t50.5 23T384 264q57-16 128.5-16T642 264q13-9 29-19t49-21.5 61-9.5q27 71 6 135 56 64 56 152 0 92-56.5 157.5T615 744q43 43 43 104v129q0 1 1 3 0 6 .5 9t4.5 6 11 3q154-52 251.5-185.5T1024 512q0-104-40.5-199t-109-163.5T711 40.5 512 0 313 40.5t-163.5 109T40.5 313 0 512z"/></symbol> </defs> </svg> </defs> </svg> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay data-md-component=overlay for=__drawer></label> <a href=#raw-sequence-data tabindex=1 class=md-skip> Skip to content </a> <header class=md-header data-md-component=header> <nav class="md-header-nav md-grid"> <div class=md-flex> <div class="md-flex__cell md-flex__cell--shrink"> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-header-nav__button md-logo"> <img src=../../assets/images/logo.png width=24 height=24> </a> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--menu md-header-nav__button" for=__drawer></label> </div> <div class="md-flex__cell md-flex__cell--stretch"> <div class="md-flex__ellipsis md-header-nav__title" data-md-component=title> <span class=md-header-nav__topic> Canine Glioma Project &middot; Verhaak Lab </span> <span class=md-header-nav__topic> Start Here &middot; Canine Glioma </span> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--search md-header-nav__button" for=__search></label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=query data-md-state=active> <label class="md-icon md-search__icon" for=__search></label> <button type=reset class="md-icon md-search__icon" data-md-component=reset tabindex=-1> &#xE5CD; </button> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=result> <div class=md-search-result__meta> Type to start searching </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <div class=md-header-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> </div> </div> </nav> </header> <div class=md-container> <main class=md-main> <div class="md-main__inner md-grid" data-md-component=container> <div class="md-sidebar md-sidebar--primary" data-md-component=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" data-md-level=0> <label class="md-nav__title md-nav__title--site" for=__drawer> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-nav__button md-logo"> <img src=../../assets/images/logo.png width=48 height=48> </a> Canine Glioma Project &middot; Verhaak Lab </label> <div class=md-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. title=Home class=md-nav__link> Home </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-2 type=checkbox id=nav-2> <label class=md-nav__link for=nav-2> Methods </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-2> Methods </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../methods/S01_samples/ title=Samples class=md-nav__link> Samples </a> </li> <li class=md-nav__item> <a href=../../methods/S02_data_sources/ title="Data Sources" class=md-nav__link> Data Sources </a> </li> <li class=md-nav__item> <a href=../../methods/S03_sample_prep/ title="Sample Preparations" class=md-nav__link> Sample Preparations </a> </li> <li class=md-nav__item> <a href=../../methods/S04_alignments_fprint/ title=Alignments class=md-nav__link> Alignments </a> </li> <li class=md-nav__item> <a href=../../methods/S05_somatic_variants/ title="Somatic SNVs and Indels" class=md-nav__link> Somatic SNVs and Indels </a> </li> <li class=md-nav__item> <a href=../../methods/S06_smgs/ title="Significantly Mutated Genes" class=md-nav__link> Significantly Mutated Genes </a> </li> <li class=md-nav__item> <a href=../../methods/S07_hallmarks/ title="Cancer Hallmarks" class=md-nav__link> Cancer Hallmarks </a> </li> <li class=md-nav__item> <a href=../../methods/S09_mutation_rate/ title="Somatic Mutation Rate" class=md-nav__link> Somatic Mutation Rate </a> </li> <li class=md-nav__item> <a href=../../methods/S10_SCNA/ title="Somatic Copy Number Aberrations" class=md-nav__link> Somatic Copy Number Aberrations </a> </li> <li class=md-nav__item> <a href=../../methods/S11_aneuploidy/ title="Aneuploidy Metrics" class=md-nav__link> Aneuploidy Metrics </a> </li> <li class=md-nav__item> <a href=../../methods/S12_ith/ title="Intratumor Heterogeneity" class=md-nav__link> Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../../methods/S13_mut_sigs/ title="Mutational Signatures" class=md-nav__link> Mutational Signatures </a> </li> <li class=md-nav__item> <a href=../../methods/S14_mut_timing/ title="Molecular Timing of Somatic Drivers" class=md-nav__link> Molecular Timing of Somatic Drivers </a> </li> <li class=md-nav__item> <a href=../../methods/S15_class_predict_methy/ title="Class Prediction using methylation classifier" class=md-nav__link> Class Prediction using methylation classifier </a> </li> <li class=md-nav__item> <a href=../../methods/S16_expression_immune/ title="Immune Microenvironment Analysis" class=md-nav__link> Immune Microenvironment Analysis </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3 type=checkbox id=nav-3 checked> <label class=md-nav__link for=nav-3> Figures </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-3> Figures </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-toggle md-nav__toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Start Here </label> <a href=./ title="Start Here" class="md-nav__link md-nav__link--active"> Start Here </a> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#raw-sequence-data title="Raw sequence data" class=md-nav__link> Raw sequence data </a> </li> <li class=md-nav__item> <a href=#r-session title="R session" class=md-nav__link> R session </a> </li> <li class=md-nav__item> <a href=#load-r-packages title="Load R packages" class=md-nav__link> Load R packages </a> </li> <li class=md-nav__item> <a href=#load-r-objects title="Load R objects" class=md-nav__link> Load R objects </a> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-2 type=checkbox id=nav-3-2> <label class=md-nav__link for=nav-3-2> F1 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-2> F1 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F1A/ title="F1A · Somatic Mutational Landscape" class=md-nav__link> F1A &middot; Somatic Mutational Landscape </a> </li> <li class=md-nav__item> <a href=../F1B/ title="F1B · Somatic Recurrent Hotspot Mutations" class=md-nav__link> F1B &middot; Somatic Recurrent Hotspot Mutations </a> </li> <li class=md-nav__item> <a href=../F1C/ title="F1C · Comparative Pathway Analysis" class=md-nav__link> F1C &middot; Comparative Pathway Analysis </a> </li> <li class=md-nav__item> <a href=../F1D/ title="F1D · Comparative Somatic Mutational Burden" class=md-nav__link> F1D &middot; Comparative Somatic Mutational Burden </a> </li> <li class=md-nav__item> <a href=../S1/ title="S1A · DNA fingerprinting" class=md-nav__link> S1A &middot; DNA fingerprinting </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-3 type=checkbox id=nav-3-3> <label class=md-nav__link for=nav-3-3> F2 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-3> F2 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F2A/ title="F2A · Somatic Copy Number Landscape" class=md-nav__link> F2A &middot; Somatic Copy Number Landscape </a> </li> <li class=md-nav__item> <a href=../F2B/ title="F2B · Comparative Aneuploidy Burden" class=md-nav__link> F2B &middot; Comparative Aneuploidy Burden </a> </li> <li class=md-nav__item> <a href=../F2C/ title="F2C · Comparative Shared Syntenic Aneuploidy" class=md-nav__link> F2C &middot; Comparative Shared Syntenic Aneuploidy </a> </li> <li class=md-nav__item> <a href=../F2D/ title="F2D · Comparative Intratumor Heterogeneity" class=md-nav__link> F2D &middot; Comparative Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../S2/ title="S2 · IGV Plots" class=md-nav__link> S2 &middot; IGV Plots </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-4 type=checkbox id=nav-3-4> <label class=md-nav__link for=nav-3-4> F3 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-4> F3 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F3A/ title="F3A · Somatic Mutational Signatures - Deconvoluted" class=md-nav__link> F3A &middot; Somatic Mutational Signatures - Deconvoluted </a> </li> <li class=md-nav__item> <a href=../F3B/ title="F3B · Somatic Mutational Landscape - de novo" class=md-nav__link> F3B &middot; Somatic Mutational Landscape - de novo </a> </li> <li class=md-nav__item> <a href=../F3C/ title="F3C · Comparative Somatic Mutational Signatures in Driver Genes" class=md-nav__link> F3C &middot; Comparative Somatic Mutational Signatures in Driver Genes </a> </li> <li class=md-nav__item> <a href=../F3D/ title="F3D · Comparative Molecular Timing of Driver Events" class=md-nav__link> F3D &middot; Comparative Molecular Timing of Driver Events </a> </li> <li class=md-nav__item> <a href=../S3/ title="S3 · Suppl Figures" class=md-nav__link> S3 &middot; Suppl Figures </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-5 type=checkbox id=nav-3-5> <label class=md-nav__link for=nav-3-5> F4 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-5> F4 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F4/ title="F4 · DNA Methylation based Prediction of Canine Gliomas" class=md-nav__link> F4 &middot; DNA Methylation based Prediction of Canine Gliomas </a> </li> <li class=md-nav__item> <a href=../S4/ title="S4 · Suppl Figure 4" class=md-nav__link> S4 &middot; Suppl Figure 4 </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-6 type=checkbox id=nav-3-6> <label class=md-nav__link for=nav-3-6> F5 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-6> F5 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F5/ title="F5B · Comparative Immunohistochemistry - Quantitative" class=md-nav__link> F5B &middot; Comparative Immunohistochemistry - Quantitative </a> </li> <li class=md-nav__item> <a href=../S5/ title="S5 · Suppl Figure 5" class=md-nav__link> S5 &middot; Suppl Figure 5 </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-4 type=checkbox id=nav-4> <label class=md-nav__link for=nav-4> R Markdown </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-4> R Markdown </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../run_code/ title=How-To class=md-nav__link> How-To </a> </li> <li class=md-nav__item> <a href=../../run_code/run_all/ title=Output class=md-nav__link> Output </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=https://verhaaklab.com/contact title=Contact class=md-nav__link> Contact </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#raw-sequence-data title="Raw sequence data" class=md-nav__link> Raw sequence data </a> </li> <li class=md-nav__item> <a href=#r-session title="R session" class=md-nav__link> R session </a> </li> <li class=md-nav__item> <a href=#load-r-packages title="Load R packages" class=md-nav__link> Load R packages </a> </li> <li class=md-nav__item> <a href=#load-r-objects title="Load R objects" class=md-nav__link> Load R objects </a> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content> <article class="md-content__inner md-typeset"> <!-- enable git edit link only if config.repo_edit is set to true --> <h1>Start Here</h1> <h3 id=raw-sequence-data>Raw sequence data<a class=headerlink href=#raw-sequence-data title="Permanent link">&para;</a></h3> <p>Sequencing data generated from canine patients with glioma (n=83 donors) is available in the Binary Alignment Map (BAM) format (n=373) at the NCBI SRA database under BioProject ID: <a href=https://www.ncbi.nlm.nih.gov/bioproject/PRJNA579792>PRJNA579792</a>.</p> <p><a href=/methods/S02_data_sources/ >View details</a> on comparative sequencing data used from patients of human pediatric and glioma.</p> <h3 id=r-session>R session<a class=headerlink href=#r-session title="Permanent link">&para;</a></h3> <ul class=task-list> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled checked><span class=task-list-indicator></span></label> Prefer using RStudio or similar GUI over copy-pasting code in terminal. The latter may emit errors due to issues with syntax highlighting and font ligatures used on this website.</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled checked><span class=task-list-indicator></span></label> Code on this website was evaluated in an isolated, new environment and syntax-related errors were addressed on the top of respective code blocks. Please <a href=https://github.com/TheJacksonLaboratory/canineglioma/issues>report bugs</a> by submitting a GitHub issue. </li> </ul> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=nf>sessionInfo</span><span class=p>()</span>
</pre></div> </td></tr></table> <blockquote> <p>Package versions may have updated versions here than what was used at the time of publication. Versions for core R packages that were used in the manuscript are listed in the manuscript under STAR methods <code>=&gt;</code> Key Resources Table.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91</pre></div></td><td class=code><div class=codehilite><pre><span></span>R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS release 6.5 (Final)

Matrix products: default
BLAS/LAPACK: /home/foo/anaconda3/lib/libopenblasp-r0.3.6.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GenomicFeatures_1.36.4    AnnotationDbi_1.46.0     
 [3] MutationalPatterns_1.10.0 rtracklayer_1.44.0       
 [5] cowplot_1.0.0             ggrepel_0.8.1            
 [7] ggforce_0.3.1             gridExtra_2.3            
 [9] forcats_0.4.0             stringr_1.4.0            
[11] dplyr_0.8.3               purrr_0.3.3              
[13] readr_1.3.1               tidyr_1.0.2              
[15] tibble_2.1.3              ggplot2_3.2.1            
[17] tidyverse_1.2.1           Palimpsest_1.0.0         
[19] GenomicRanges_1.36.0      GenomeInfoDb_1.20.0      
[21] IRanges_2.18.1            S4Vectors_0.22.0         
[23] NMF_0.21.0                Biobase_2.44.0           
[25] BiocGenerics_0.30.0       cluster_2.1.0            
[27] rngtools_1.4              pkgmaker_0.27            
[29] registry_0.5-1            extrafont_0.17          

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1            ggsignif_0.5.0             
  [3] ellipsis_0.3.0              rprojroot_1.3-2            
  [5] IRdisplay_0.7.0             lsa_0.73.1                 
  [7] XVector_0.24.0              ggdendro_0.1-20            
  [9] base64enc_0.1-3             rstudioapi_0.10            
 [11] ggpubr_0.2.1.999            farver_1.1.0               
 [13] SnowballC_0.6.0             bit64_0.9-7                
 [15] fansi_0.4.0                 lubridate_1.7.4            
 [17] xml2_1.2.2                  codetools_0.2-16           
 [19] doParallel_1.0.14           polyclip_1.10-0            
 [21] IRkernel_1.0.2              jsonlite_1.6               
 [23] Rsamtools_2.0.0             broom_0.5.4                
 [25] gridBase_0.4-7              graph_1.62.0               
 [27] compiler_3.6.1              httr_1.4.1                 
 [29] backports_1.1.5             assertthat_0.2.1           
 [31] Matrix_1.2-17               lazyeval_0.2.2             
 [33] cli_2.0.1                   tweenr_1.0.1               
 [35] htmltools_0.3.6             prettyunits_1.0.2          
 [37] tools_3.6.1                 gtable_0.3.0               
 [39] glue_1.3.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.3              Rcpp_1.0.3                 
 [43] cellranger_1.1.0            vctrs_0.2.2                
 [45] Biostrings_2.52.0           gdata_2.18.0               
 [47] nlme_3.1-140                iterators_1.0.10           
 [49] rvest_0.3.5                 lifecycle_0.1.0            
 [51] gtools_3.8.1                XML_3.98-1.20              
 [53] MASS_7.3-51.4               zlibbioc_1.30.0            
 [55] RCircos_1.2.1               scales_1.0.0               
 [57] BSgenome_1.52.0             VariantAnnotation_1.30.1   
 [59] hms_0.5.3                   SummarizedExperiment_1.14.0
 [61] RColorBrewer_1.1-2          memoise_1.1.0              
 [63] biomaRt_2.40.1              stringi_1.4.3              
 [65] RSQLite_2.1.1               foreach_1.4.4              
 [67] plotrix_3.7-6               caTools_1.17.1.2           
 [69] BiocParallel_1.18.0         bibtex_0.4.2               
 [71] repr_1.0.1                  rlang_0.4.3                
 [73] pkgconfig_2.0.3             matrixStats_0.54.0         
 [75] bitops_1.0-6                pracma_2.2.5               
 [77] evaluate_0.14               lattice_0.20-38            
 [79] labeling_0.3                GenomicAlignments_1.20.1   
 [81] bit_1.1-14                  tidyselect_0.2.5           
 [83] here_0.1                    plyr_1.8.4                 
 [85] magrittr_1.5                R6_2.4.1                   
 [87] gplots_3.0.1.1              generics_0.0.2             
 [89] pbdZMQ_0.3-3                DelayedArray_0.10.0        
 [91] DBI_1.1.0                   pillar_1.4.3               
 [93] haven_2.2.0                 withr_2.1.2                
 [95] RCurl_1.95-4.12             modelr_0.1.5               
 [97] crayon_1.3.4                uuid_0.1-2                 
 [99] KernSmooth_2.23-15          progress_1.2.2             
[101] grid_3.6.1                  readxl_1.3.1               
[103] data.table_1.12.2           blob_1.2.0                 
[105] Rgraphviz_2.28.0            digest_0.6.23              
[107] xtable_1.8-4                munsell_0.5.0              
</pre></div> </td></tr></table> <h3 id=load-r-packages>Load R packages<a class=headerlink href=#load-r-packages title="Permanent link">&para;</a></h3> <blockquote> <p>PS: Not all of R packages were loaded for generating all figures. If you are loading all of packages, be aware of warnings related to masked functions and use pacakge name prefix, e.g., <code>dplyr::select</code> to explicitly call a function.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=c1>## For Figure 1 and 2, including supplemental figures</span>
<span class=nf>library</span><span class=p>(</span><span class=n>knitr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>htmlTable</span><span class=p>)</span>

<span class=nf>library</span><span class=p>(</span><span class=n>tidyverse</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>broom</span><span class=p>)</span>

<span class=nf>library</span><span class=p>(</span><span class=n>maftools</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>tint</span><span class=p>)</span>

<span class=nf>library</span><span class=p>(</span><span class=n>DBI</span><span class=p>)</span>

<span class=nf>library</span><span class=p>(</span><span class=n>ggplot2</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>gridExtra</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggforce</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggrepel</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>cowplot</span><span class=p>)</span>

<span class=c1>## For Figure 3, including supplemental figures,</span>
<span class=c1>## load these extra packages.</span>

<span class=nf>library</span><span class=p>(</span><span class=n>Palimpsest</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>BSgenome.Cfamiliaris.UCSC.canFam3</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>rtracklayer</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>MutationalPatterns</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>tidyverse</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>GenomicFeatures</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>NMF</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>BradleyTerryScalable</span><span class=p>)</span>

<span class=c1>## Import Arial fonts to avoid error</span>
<span class=c1>## related to font not found</span>
<span class=nf>library</span><span class=p>(</span><span class=n>extrafont</span><span class=p>)</span>
<span class=nf>loadfonts</span><span class=p>()</span>
</pre></div> </td></tr></table> <h3 id=load-r-objects>Load R objects<a class=headerlink href=#load-r-objects title="Permanent link">&para;</a></h3> <p><img alt=🚀 class=emojione src=https://cdn.jsdelivr.net/emojione/assets/4.0/png/64/1f680.png title=:rocket:> Download <a href=/data/cgp_base_objects_v1_20200207.RData>/data/cgp_base_objects_v1_20200207.RData</a> md5sum: 12812097f49d8c8300a470491387c7d1.</p> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=nf>load</span><span class=p>(</span><span class=s>&quot;cgp_base_objects_v1_20200207.RData&quot;</span><span class=p>)</span>
<span class=nf>ls</span><span class=p>()</span>
<span class=c1>## 37 objects</span>
</pre></div> </td></tr></table> <table> <thead> <tr> <th>R object</th> <th>Description</th> </tr> </thead> <tbody> <tr> <td>cgp_suppl_sample_info_master</td> <td>Donor level information on 81 canine patients with glioma.</td> </tr> <tr> <td>maf_nogistic</td> <td>maftools compatible MAF object, does not contain copy number data on 81 canine patients. Note that three patients had no detectable somatic mutations following somatic mutation filtration.</td> </tr> <tr> <td>cgp_maftools_gistic_matched</td> <td>maftools compatible MAF object <strong>only</strong> on 67 canine patients with paired tumor-normal samples where somatic copy number data was available.</td> </tr> <tr> <td>cgp_maftools_gistic_n81</td> <td>maftools compatible MAF object of 81 canine patients. This object <strong>also</strong> contain copy number data on 67 of 81 canine patients where paired tumor-normal samples were available.</td> </tr> <tr> <td>cgp_maftools_gistic_methy</td> <td>maftools compatible MAF object of 42 canine patients where DNA methylation data was available.</td> </tr> <tr> <td>cancer_genes_maftools</td> <td>Cancer genes with observed somatic mutation in canine patients (n=48). Based on cancer hallmark analysis (Fig 1B).</td> </tr> <tr> <td>HALLMARK</td> <td>Genes (n=277) to Cancer Hallmarks (n=11) mapping</td> </tr> <tr> <td>CG_all</td> <td>Somatic variants in canine glioma cohort (n=62/81 cases)</td> </tr> <tr> <td>ALL_HALLMARK_v1</td> <td>Cancer Hallmark table for Figure 1C</td> </tr> <tr> <td>mutrate_table</td> <td>Somatic mutation burden across patients of canine and human pediatric and adult cancers (n=4,840).</td> </tr> <tr> <td>merged_anp_scores</td> <td>Comparative aneuploidy metrics showing fraction of genome with aneuploidy</td> </tr> <tr> <td>cgp_aneuploidy_metrics_master</td> <td>aneuploidy metrics for 67 canine glioma patients.</td> </tr> <tr> <td>mat_aneuploidy_summary_mostvar_cgAll</td> <td>Comparative shared syntenic aneuploidy across canine and human pediatric and adult glioma.</td> </tr> <tr> <td>updated_merged_entropy_df_ridges</td> <td>Shannon entropy across canine and human pediatric and adult glioma where whole-genome sequence data was available (n=105).</td> </tr> <tr> <td>merged_ccf_vaf_df</td> <td>Comparative intra-tumor heterogeneity (Shannon entropy) and cellular prevalence matrix.</td> </tr> <tr> <td>mut_mat</td> <td>Canine glioma mutational matrix based on 96-trinucleotide context - Derived using <code>mut_matrix</code> function in <em>MutationalPatterns</em> R package.</td> </tr> <tr> <td>cancer_signatures</td> <td>Known human mutational signatures in adult (n=30, COSMIC v2) and pediatric cancers (n=12).</td> </tr> <tr> <td>fit_res_hm</td> <td>Somatic mutational signatures in canine glioma patients with outlier mutation profile (n=8). Signatures were fitted to known human mutational signatures.</td> </tr> <tr> <td>fit_res_nonhm</td> <td>Somatic mutational signatures in canine glioma patients without outlier mutation profile (n=73). Signatures were fitted to known human mutational signatures.</td> </tr> <tr> <td>hm_nonhm_topsigs_tbl, driver_sigs_groups</td> <td>Mutational signatures and their grouping to proposed mechnaism</td> </tr> <tr> <td>instability_sigs_df</td> <td>subset of <code>fit_res$contribution</code> matrix for suppl fig 3B.</td> </tr> <tr> <td>merged_ag_pg_cg_denovo_sig</td> <td>De-novo mutational signatures from canine, human pediatric and adult cohort.</td> </tr> <tr> <td>trimmed_{cg,pg,ag}_vcf</td> <td>Trimmed VCFs for plotting Figure 3C</td> </tr> <tr> <td>win_table_{cg,pg,ag}_vcf</td> <td>Trimmed VCFs for plotting Figure 3D</td> </tr> <tr> <td>cgp_RRBS.noNormals_class_prediction</td> <td>class prediction table for Figure 4</td> </tr> <tr> <td>cgp_RRBS.noNormals_sample_info</td> <td>Donor level information for 45 canine patients with glioma where DNA methylation data was available.</td> </tr> <tr> <td>{cgp, tcga}_info</td> <td>Sample metadata for suppl figure 5.</td> </tr> <tr> <td>{cgp, ped, tcga}_cibersort</td> <td>CIBERSORT output for three cohorts.</td> </tr> <tr> <td>fig5b_dat</td> <td>Data related to IHC panel, Figure 5B.</td> </tr> </tbody> </table> <p>Last updated: February 10, 2020</p> <h2 id=__comments>Comments</h2> <div class="admonition bug"> <p class=admonition-title>Issues with code ?</p> <p>Please limit comments below for general questions. For code-related query, prefer <a href=https://github.com/TheJacksonLaboratory/canineglioma/issues>submitting a GitHub issue</a>.</p> </div> <div id=disqus_thread></div> <script>
    var disqus_config = function () {
      this.page.url = "https://canineglioma.verhaaklab.com/figures/A1_preload/";
      this.page.identifier =
        "/figures/A1_preload/";
    };
    (function() {
      var d = document, s = d.createElement("script");
      s.src = "//verhaaklab.disqus.com/embed.js";
      s.setAttribute("data-timestamp", +new Date());
      (d.head || d.body).appendChild(s);
    })();
  </script> </article> </div> </div> </main> <footer class=md-footer> <div class=md-footer-nav> <nav class="md-footer-nav__inner md-grid"> <a href=../../methods/S16_expression_immune/ title="Immune Microenvironment Analysis" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel=prev> <div class="md-flex__cell md-flex__cell--shrink"> <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i> </div> <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"> <span class=md-flex__ellipsis> <span class=md-footer-nav__direction> Previous </span> Immune Microenvironment Analysis </span> </div> </a> <a href=../F1A/ title="F1A · Somatic Mutational Landscape" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel=next> <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"> <span class=md-flex__ellipsis> <span class=md-footer-nav__direction> Next </span> F1A &middot; Somatic Mutational Landscape </span> </div> <div class="md-flex__cell md-flex__cell--shrink"> <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i> </div> </a> </nav> </div> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-footer-copyright> <div class=md-footer-copyright__highlight> Copyright &copy; 2017 - 2020 <a href=https://verhaaklab.com>Verhaak Lab at the Jackson Laboratory for Genomic Medicine</a> </div> powered by <a href=https://www.mkdocs.org>MkDocs</a> and <a href=https://squidfunk.github.io/mkdocs-material/ > Material for MkDocs</a> | <a href=https://github.com/sbamin/theme-mkdocs-material/ > Forked Code</a> </div> <div class=md-footer-social> <link rel=stylesheet href=../../assets/fonts/font-awesome.css> <a href=https://verhaaklab.com/contact class="md-footer-social__link fa fa-envelope"></a> <a href=https://twitter.com/roelverhaak class="md-footer-social__link fa fa-twitter"></a> <a href=https://github.com/sbamin class="md-footer-social__link fa fa-github"></a> <a href=https://twitter.com/sbamin class="md-footer-social__link fa fa-twitter"></a> </div> </div> </div> </footer> </div> <script src=../../assets/javascripts/application.ffb616e8.js></script> <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script> <script src=../../assets/js/katex.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js></script> <script src=../../search/main.js></script> </body> </html>