---
title: S1A &middot; DNA fingerprinting
---

*   DNA fingerprinting of whole genome and exome samples from canine glioma patients was performed using [NGSCheckMate](https://github.com/parklab/NGSCheckMate)[^1]. [Details](/methods/s04_alignments_fprint/#dna-fingerprinting)

```sh
#!/bin/bash

## Run NGSCheckMate Fingerprint
## https://github.com/parklab/NGSCheckMate

## @sbamin

NGSCM_EXE="/projects/verhaak-lab/verhaak_env/verhaak_apps/NGSCheckMate/ncm.py"
OUTDIR="/fastscratch/foo/cgp/ngs_checkmate"

## dbsnp file
DBSNP="Canis_familiaris_dbsnp_protein_coding_cds_exons_canonical.bed"

REF_FASTA="CanFam3_1.fa"

BAMLIST="${HOME}/pipelines/cgp_qc/runner/fingerprint/cgp_bams.tsv"

## Number of parallel bcftools
THREADS=12

mkdir -p "${OUTDIR}"/vcfs
mkdir -p "${OUTDIR}"/results

export NGSCM_EXE OUTDIR DBSNP REF_FASTA BAMLIST

source "${RVPYTHON}"/v4.2.0/bin/activate base && \
echo "switched to python2 legacy env; using samtools, bcftools, htslib 1.9"

command -v python && \
command -v samtools && \
command -v bcftools

## load ngs_checkmate
module load rvngscheckmate/1.3.0

## Run mpileup and get per-bam vcf for DBSNP regions
mpileup_fn() {

  MYBAM="$1"
  if [[ ! -f "$MYBAM" ]]; then
    echo "ERROR: BAM file at $MYBAM does not exitst" >&2
    exit 1
  fi

  VCFID=$(printf "%s/vcfs/fingerprint_%s.vcf" "$OUTDIR" "$(basename "$MYBAM" .bam)")

  echo -e "\n\nGetting mpileup...\nfrom: $MYBAM\nto: $VCFID\nBED: $DBSNP\n"
  sleep 2

  set -euo pipefail

  if [[ ! -s "${VCFID}" ]]; then
    bcftools mpileup -Ou -f "${REF_FASTA}" -R "${DBSNP}" "${MYBAM}" | \
      bcftools call -Ou -mv | \
      bcftools filter -s LowQual -e '%QUAL<20 || DP>100' > "$VCFID"
  else
    echo -e "\n\nSKIP: $VCFID is present - skipping mpileup\n\n" >&2
  fi
}

export -f mpileup_fn

############# Run mpileup in parallel #########################
THREADS="${THREADS:-2}"
echo "Running $THREADS parallel bcftools mpileups"
cat "$BAMLIST" | parallel -j"${THREADS}" mpileup_fn {}
###############################################################

echo "Running NGSCheckMate"
sleep 2

python "${NGSCM_EXE}" -V -d "${OUTDIR}"/vcfs -O "${OUTDIR}"/results -N cgp_fingerprint_ngscheckmate_v2 -bed "${DBSNP}" -nz

exitstat=$?
echo "ending NGSCheckMate with exit: $exitstat"
```

[^1]: Lee S, Lee S, Ouellette S, Park W-Y, Lee EA, Park PJ. NGSCheckMate: software for validating sample identity in next-generation sequencing studies within and across data types. Nucleic Acids Res 2017;45:e103. doi: [10.1093/nar/gkx193](https://doi.org/10.1093/nar/gkx193)
