---
title: "cgp_figures"
author: "sbamin"
date: "2/8/2020"
output:
  html_document:
    code_folding: hide
    df_print: paged
    fig_caption: yes
    fig_height: 18
    fig_width: 12
    highlight: textmate
    keep_md: yes
    number_sections: yes
    theme: default
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '4'
---

## Comparative molecular life history of spontaneous canine and human gliomas

>Amin et al., Comparative Molecular Life History of Spontaneous Canine and Human Gliomas, *Cancer Cell* (2020). DOI: [10.1016/j.ccell.2020.01.004](https://doi.org/10.1016/j.ccell.2020.01.004) Feb 10, 2020.

## Set Env

```{r global_opts}
library(knitr)
opts_chunk$set(comment = NA, fig.width = 18, fig.height = 12, 
               message = TRUE, cache = FALSE, echo = FALSE)
```

### Load R packages

```{r load_pkgs, message = FALSE}
## For Figure 1 and 2, including supplemental figures
library(knitr)
library(htmlTable)

library(tidyverse)
library(broom)

library(maftools)
library(tint)

library(DBI)

library(ggplot2)
library(gridExtra)
library(ggforce)
library(ggrepel)
library(cowplot)

## For Figure 3, including supplemental figures,
## load these extra packages.

library(Palimpsest)
## genome package is required for preprocessing
## but not for reproducing figure panels.
# library(BSgenome.Cfamiliaris.UCSC.canFam3)
library(rtracklayer)
library(MutationalPatterns)
library(tidyverse)
library(GenomicFeatures)
library(NMF)
library(BradleyTerryScalable)

## Import Arial fonts to avoid error
## related to font not found
library(extrafont)
loadfonts()
```

### Load R objects

```{r load_rdata}
glue::glue("Work directory is {getwd()}\nLoading base objects...")
load("../data/cgp_base_objects_v1_20200207.RData")
glue::glue("List objects")
ls()
```

***

## Fig 1A

### Required metadata

*   Genes to plot

>Based on [significantly mutated genes analysis](/methods/S06_smgs). See also Table S2.

```{r}
oncoprint_genes <- c("PDGFRA", "PIK3CA", "NF1", "TP53", "EGFR", "IDH1",
    "FGFR1", "SPOP", "TRIM11", "SMARCC2", "SMARCA4", "SMARCAD1", "ARID5B",
    "KMT2A", "KMT2D", "RB1", "CDKN2C")

oncoprint_genes_gistic <- c("PDGFRA", "MYC", "TP53", "EGFR", "RB1", "PTEN",
    "CDKN2A")

oncoprint_genes_snv_scna <- c("PDGFRA", "PIK3CA", "NF1", "TP53", "ZFHX3",
    "EGFR", "IDH1", "FGFR1", "SPOP", "TRIM11", "SMARCC2",
    "SMARCA4", "SMARCAD1", "ARID5B",
    "KMT2A", "KMT2C", "KMT2D",
    "RB1", "PTEN", "CDKN2A")
```

*   Sample and Variant Annotations 

```{r}
## Specify colors for sample annotations
ann_colors <- list(Tumor_Morphology = c(Astro = "#d95f02",
    Oligo = "#7570b3", Undefined = "#1b9e77"),
    Tumor_Grade = c(High = "#FEB24C", Low = "#FFEDA0"),
    Tumor_Location = c(Hemispheric = "#a6cee3",
    Cerebellar = "#1f78b4",
    midline = "#e31a1c", UN = "#fb9a99"),
    Tissue_Archival = c(`snap-frozen` = "#ffcc33", ffpe = "#cc0033"),
    Matched_Normal = c(paired = "#ffcc33", tumor_only = "#cc0033"))

## specify colors for variant types
varcall_colors <- c(Amp = "#006400",
    LOH = "#FF9999",
    Del = "#9933FF",
    Frame_Shift_Del = "#1E90FF",
    Frame_Shift_Ins = "#FF8C69",
    Splice_Site = "#00CED1",
    Splice_Region = "#00CED1",
    Translation_Start_Site = "#CD6600",
    Nonsense_Mutation = "#FFC100",
    Nonstop_Mutation = "#E7B98A",
    In_Frame_Del = "#E78A96",
    In_Frame_Ins = "#FFC1C9",
    Missense_Mutation = "#CD6600",
    Translation_Start_Site = "#008B8B",
    Multi_Hit = "#3D3D3D")
```

### Plot Fig 1A

>Somatic mutational lanscape of glioma driver genes. Based on [significantly mutated genes analysis](/methods/S06_smgs).

??? error "font family 'Arial' not found"
    If you get an error, `font family 'Arial' not found in PostScript font database`, either remove *fonts* argument from `pdf` or import system fonts using *extrafont* R package. See [documentation by Gavin Simpson](https://www.fromthebottomoftheheap.net/2013/09/09/preparing-figures-for-plos-one-with-r/) for details. Run `extrafont::loadfonts()` as detailed in [Start Here](/figures/a1_preload/#load-r-packages).

```{r}
pdf("F1A.pdf",
    width = 18, height = 12, pointsize = 12,
    fonts = "Arial", bg = "white",
    compress = FALSE, useDingbats = FALSE)

fig_1A <- oncoplot(maf = maf_nogistic,
    genes = oncoprint_genes,
    additionalFeature = c("dnds_mutsig", "Driver"),
    removeNonMutated = TRUE,
    keepGeneOrder = FALSE,
    writeMatrix = FALSE,
    sortByMutation = FALSE,
    fontSize = 0.9,
    legendFontSize = 1.8,
    annotationFontSize = 1.8,
    clinicalFeatures = c("Tumor_Morphology", "Tumor_Grade",
                       "Tumor_Location"),
    annotationColor = ann_colors,
    colors = varcall_colors[
        c(levels(maf_nogistic@data$Variant_Classification),
        "Multi_Hit")
    ],
    logColBar = FALSE,
    gene_mar = 8,
    bgCol = "#ffffff",
    borderCol = "#CCCCCC")

dev.off()
```

### Suppl Fig 1B

>Somatic mutational lanscape of COSMIC cancer genes.

```{r}
pdf("SF1B.pdf",
    width = 18, height = 12, pointsize = 12,
    fonts = "Arial", bg = "white",
    compress = FALSE, useDingbats = FALSE)

suppl_fig_1B <- oncoplot(maf = maf_nogistic,
     genes = cancer_genes_maftools,
     additionalFeature = c("dnds_mutsig", "Driver"),
     removeNonMutated = FALSE,
     keepGeneOrder = FALSE,
     writeMatrix = FALSE,
     sortByMutation = TRUE,
     fontSize = 0.7,
     legendFontSize = 1.2,
     annotationFontSize = 1.2,
     clinicalFeatures = c("Tumor_Morphology", "Tumor_Grade",
                          "Tumor_Location"),
     annotationColor = ann_colors,
     colors = varcall_colors[
        c(levels(maf_nogistic@data$Variant_Classification),
        "Multi_Hit")
        ],
     logColBar = FALSE,
     gene_mar = 7,
     bgCol = "#ffffff",
     borderCol = "#CCCCCC")

dev.off()
```

### Suppl Fig 1F

>Somatic mutational lanscape, including copy-number aberrations of COSMIC cancer genes. This figure also contains somatic copy-number data which is detailed under [Fig 2A](/figures/F2A/).

```{r}
pdf("SF1F.pdf",
    width = 18, height = 12, pointsize = 12,
    bg = "white",
    compress = FALSE, useDingbats = FALSE)

suppl_fig_1F <- oncoplot(maf = cgp_maftools_gistic_n81,
    ## adding known SCNA drivers
    genes = c(cancer_genes_maftools, "CDKN2A", "PTEN"),
    additionalFeature = c("dnds_mutsig", "Driver"),
    removeNonMutated = FALSE,
    keepGeneOrder = FALSE,
    writeMatrix = TRUE,
    sortByMutation = TRUE,
    fontSize = 0.7,
    legendFontSize = 1.2,
    annotationFontSize = 1.2,
    clinicalFeatures = c("Tumor_Morphology", "Tumor_Grade",
                       "Tumor_Location", "Tissue_Archival", 
                       "Matched_Normal"),
    annotationColor = ann_colors,
    colors = varcall_colors[
    c(levels(cgp_maftools_gistic_n81@data$Variant_Classification), 
    "Multi_Hit")
    ],
    logColBar = FALSE,
    gene_mar = 7,
    bgCol = "#ffffff",
    borderCol = "#CCCCCC")

dev.off()
```

***

## Fig 1B

*   Alternate code using `lollipopPlot` function in R package, [*maftools*](https://bioconductor.org/packages/release/bioc/html/maftools.html)

```{r}
pdf("F1B.pdf",
    width = 18, height = 12, pointsize = 12,
    bg = "white",
    compress = FALSE, useDingbats = FALSE)

lollipopPlot(maf = maf_nogistic,
            gene = 'PIK3CA',
            AACol = 'HGVSp_Short',
            showMutationRate = TRUE,
            labelPos = 1047, labPosSize = 1.2,
            cBioPortal = FALSE, repel = FALSE,
            collapsePosLabel = TRUE, legendTxtSize = 1.2,
            labPosAngle = 0, domainLabelSize = 1.0, axisTextSize = c(1, 1),
            printCount = TRUE, colors = NULL, domainColors = NULL,
            labelOnlyUniqueDoamins = TRUE, defaultYaxis = TRUE,
            titleSize = c(1.2, 1), pointSize = 1.2)

lollipopPlot(maf = maf_nogistic,
            gene = 'IDH1',
            AACol = 'HGVSp_Short',
            showMutationRate = TRUE,
            labelPos = 132, labPosSize = 1.2,
            cBioPortal = FALSE, repel = FALSE,
            collapsePosLabel = TRUE, legendTxtSize = 1.2,
            labPosAngle = 0, domainLabelSize = 1.0, axisTextSize = c(1, 1),
            printCount = TRUE, colors = NULL, domainColors = NULL,
            labelOnlyUniqueDoamins = TRUE, defaultYaxis = TRUE,
            titleSize = c(1.2, 1), pointSize = 1.2)

lollipopPlot(maf = maf_nogistic,
             gene = 'SPOP',
             AACol = 'HGVSp_Short',
             showMutationRate = TRUE,
             labelPos = 94, labPosSize = 1.2,
             cBioPortal = FALSE, repel = FALSE,
             collapsePosLabel = TRUE, legendTxtSize = 1.2,
             labPosAngle = 0, domainLabelSize = 1.0, axisTextSize = c(1, 1),
             printCount = TRUE, colors = NULL, domainColors = NULL,
             labelOnlyUniqueDoamins = TRUE, defaultYaxis = TRUE,
             titleSize = c(1.2, 1), pointSize = 1.2)

dev.off()
```

***

## Fig 1C

Cancer Hallmarks enrichment across canine, human pediatric and adult gliomas. 
>Genes to hallmark mapping  was primarily derived from SLAPenrich R package. See also [relevant methods](/methods/S07_hallmarks/) for details.

```{r}
table_hallmark <- as.data.frame(table(HALLMARK$hallmark)) %>%
                    dplyr::rename(hallmark = Var1)

## require at least one driver event
Freq_threshold = 1

CG_HALLMARK <- CG_all %>% 
  gather(key = sample, value = value, -gene_symbol) %>%
  filter(gene_symbol %in% HALLMARK$gene_symbol) %>% 
  left_join(HALLMARK, by = "gene_symbol") %>% 
  dplyr::select(-gene_symbol) %>% 
  group_by(hallmark, sample) %>% 
  mutate(sum = sum(value)) %>%  
  dplyr::select(-value) %>% 
  ungroup() %>% 
  distinct() %>%
  left_join(table_hallmark) %>% 
  mutate(alteration = ifelse(sum >= Freq_threshold, 1, 0)) %>%  
  add_count(hallmark, name = "n_hallmark") %>%
  mutate(n_hallmark = n_hallmark + 5) %>% 
  add_count(hallmark, alteration, name = "n_alterations") %>% 
  mutate(proportion_CG = n_alterations/n_hallmark,
         CG_no_alterations = n_hallmark - n_alterations) %>% 
  filter(alteration != 0) %>% 
  dplyr::select(hallmark, proportion_CG,
         CG_alterations = n_alterations,
         CG_no_alterations,
         CG_total = n_hallmark) %>% 
  distinct()
```

*   Similar to `CG_HALLMARK`, enrichment per hallmark tables were created for molecular subtypes across human cancers, and merged in the master table, `ALL_HALLMARK_v1`.

```{r mk_hallmark, eval = FALSE}
## This code block will not run as variant calls for human cancers are not
## stored in public repository. Instead, we provide ALL_HALLMARK_v1 object
## to draw figures.

ALL_HALLMARK_v1 <- AG_IDHWT_HALLMARK %>%
    left_join(AG_IDHMUT_CODEL_HALLMARK, by = "hallmark") %>%
    left_join(AG_IDHMUT_NONCODEL_HALLMARK,  by =  "hallmark") %>%
    left_join(PG_HGG_HALLMARK,  by =  "hallmark") %>%
    left_join(PG_LGG_HALLMARK,  by =  "hallmark") %>%
    left_join(CG_HALLMARK,  by =  "hallmark") %>% 
    dplyr::select(hallmark,
        IDHWT = proportion_AG_IDHWT,
        IDHMUT_CODEL = proportion_AG_IDHMUT_CODEL,
        IDHMUT_NONCODEL = proportion_AG_IDHMUT_NONCODEL,
        PG_H3mut = proportion_PG_HGG,
        PG_H3wt = proportion_PG_LGG,
        CG = proportion_CG) %>%
        gather(key = species, value = proportion, IDHWT:CG) %>%
        mutate(species = factor(species,
            levels = c("CG",
                        "PG_H3wt", "PG_H3mut",
                        "IDHMUT_CODEL", "IDHMUT_NONCODEL", "IDHWT")))
```

### Fig 1C

>Subset of hallmarks as a main figure 1C.

```{r}
pdf(file = "F1C.pdf",
    height = 7, width = 15,
    bg = "transparent",
    useDingbats = FALSE)

ALL_HALLMARK_v1 %>% 
  filter(hallmark %in%
    c("Avoiding Immune Destruction",
        "Deregulating Cellular Energetics",
        "Epigenetic Deregulation",
        "Genome Instability and Mutation",
        "Resisting Cell Death")) %>% 
  unite("hallmark_species", hallmark, species, remove = F) %>% 
  mutate(hallmark_species = factor(hallmark_species, 
   levels = c("Deregulating Cellular Energetics_CG", "Deregulating Cellular Energetics_PG_H3wt", "Deregulating Cellular Energetics_PG_H3mut","Deregulating Cellular Energetics_IDHMUT_CODEL", "Deregulating Cellular Energetics_IDHMUT_NONCODEL", "Deregulating Cellular Energetics_IDHWT",
              "Resisting Cell Death_CG", "Resisting Cell Death_PG_H3wt", "Resisting Cell Death_PG_H3mut", "Resisting Cell Death_IDHMUT_CODEL", "Resisting Cell Death_IDHMUT_NONCODEL", "Resisting Cell Death_IDHWT",
              "Genome Instability and Mutation_CG", "Genome Instability and Mutation_PG_H3wt", "Genome Instability and Mutation_PG_H3mut", "Genome Instability and Mutation_IDHMUT_CODEL", "Genome Instability and Mutation_IDHMUT_NONCODEL", "Genome Instability and Mutation_IDHWT",
              "Epigenetic Deregulation_CG", "Epigenetic Deregulation_PG_H3wt", "Epigenetic Deregulation_PG_H3mut", "Epigenetic Deregulation_IDHMUT_CODEL", "Epigenetic Deregulation_IDHMUT_NONCODEL", "Epigenetic Deregulation_IDHWT",
              "Avoiding Immune Destruction_CG", "Avoiding Immune Destruction_PG_H3wt","Avoiding Immune Destruction_PG_H3mut", "Avoiding Immune Destruction_IDHMUT_CODEL", "Avoiding Immune Destruction_IDHMUT_NONCODEL", "Avoiding Immune Destruction_IDHWT")),
    hallmark = factor(hallmark, levels = c("Deregulating Cellular Energetics", "Resisting Cell Death", "Genome Instability and Mutation", "Epigenetic Deregulation", "Avoiding Immune Destruction"), 
                   labels = c("Deregulating Cellular Energetics" = "Deregulating  \nCellular Energetics" , "Resisting Cell Death" = "Resisting Cell  \nDeath", "Genome Instability and Mutation" = "Genome  \nInstability", "Epigenetic  \nDeregulation", "Avoiding Immune Destruction" = "Avoiding Immune  \nDestruction"))) %>% 
    ggplot(aes(x = hallmark_species, y = proportion, fill = hallmark)) +
    geom_bar(stat = "identity", position = "dodge", width = 0.975) +
    scale_fill_brewer(palette = "Dark2", " ") +
    labs(x = " ", y = "Proportion of patients") + 
    cowplot::theme_cowplot(font_size = 18) +
    theme(legend.position = "bottom", legend.text = element_text(size=22), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
    coord_cartesian(ylim = c(0,1)) +
    scale_x_discrete(labels=c(
        "Resisting Cell Death_IDHWT" = "IDHwt",
        "Resisting Cell Death_IDHMUT_CODEL" = "IDHmut\ncodel",
        "Resisting Cell Death_IDHMUT_NONCODEL" = "IDHmut\nnoncodel",
        "Resisting Cell Death_PG_H3mut" = "H3mut",
        "Resisting Cell Death_PG_H3wt" = "H3wt",
        "Resisting Cell Death_CG" = "CG", 
        "Avoiding Immune Destruction_IDHWT" = "IDHwt",
        "Avoiding Immune Destruction_IDHMUT_CODEL" = "IDHmut\ncodel",
        "Avoiding Immune Destruction_IDHMUT_NONCODEL" = "IDHmut\nnoncodel",
        "Avoiding Immune Destruction_PG_H3mut" = "H3mut", "Avoiding Immune Destruction_PG_H3wt" = "H3wt",
        "Avoiding Immune Destruction_CG" = "CG", 
        "Deregulating Cellular Energetics_IDHWT" = "IDHwt",
        "Deregulating Cellular Energetics_IDHMUT_CODEL" = "IDHmut\ncodel","Deregulating Cellular Energetics_IDHMUT_NONCODEL" = "IDHmut\nnoncodel",
        "Deregulating Cellular Energetics_PG_H3mut" = "H3mut",
        "Deregulating Cellular Energetics_PG_H3wt" = "H3wt",
        "Deregulating Cellular Energetics_CG" = "CG", 
        "Epigenetic Deregulation_IDHWT" = "IDHwt",
        "Epigenetic Deregulation_IDHMUT_CODEL" = "IDHmut\ncodel",
        "Epigenetic Deregulation_IDHMUT_NONCODEL" = "IDHmut\nnoncodel",
        "Epigenetic Deregulation_PG_H3mut" = "H3mut",
        "Epigenetic Deregulation_PG_H3wt" = "H3wt",
        "Epigenetic Deregulation_CG" = "CG",
        "Genome Instability and Mutation_IDHWT" = "IDHwt",
        "Genome Instability and Mutation_IDHMUT_CODEL" = "IDHmut\ncodel",
        "Genome Instability and Mutation_IDHMUT_NONCODEL" = "IDHmut\nnoncodel",
        "Genome Instability and Mutation_PG_H3mut" = "H3mut",
        "Genome Instability and Mutation_PG_H3wt" = "H3wt",
        "Genome Instability and Mutation_CG" = "CG")) + 
    geom_vline(xintercept = c(1.5, 3.5, 7.5, 9.5, 13.5, 15.5, 19.5, 21.5, 25.5, 27.5),
    color="white", linetype="solid", size=3.5) +
    geom_vline(xintercept = c(6.5, 12.5, 18.5, 24.5),
    color="black", linetype="solid", size=0.5)

dev.off()
```

### Suppl Fig 1C

>Visualize all hallmarks.

```{r}
barplot_all <- ALL_HALLMARK_v1 %>%
    ggplot(aes(x = species, y = proportion, fill = hallmark)) +
    geom_bar(stat = "identity", width = 0.975) + cowplot::theme_cowplot(font_size = 22) +
    geom_vline(xintercept = c(1.5, 3.5), color="white", linetype="solid", size=4) + 
    scale_fill_manual(values = c("Evading Growth Suppressors" = "#754200", 
                               "Avoiding Immune Destruction" = "#D12690", 
                               "Enabling Replicative Immortality" = "#007DB1", 
                               "Tumor-Promoting Inflammation" = "#E17A1D", 
                               "Activating Invasion and Metastasis" = "#221E1F", 
                               "Inducing Angiogenesis"  = "#EF3B34", 
                               "Genome Instability and Mutation" = "#1A3D8F", 
                               "Resisting Cell Death" = "#839098", 
                               "Deregulating Cellular Energetics" = "#7F3F98", 
                               "Sustaining Proliferative Signaling" = "#019E59", 
                               "Epigenetic Deregulation" = "#777AB1",
                               "#777AB1", 
                               "Other" = "#cdcdc1")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  labs(x = " ", y = "Proportion of patients", fill = "") +
  facet_wrap(~hallmark) + 
  theme(strip.background = element_blank(),strip.text.x = element_blank()) +
  scale_x_discrete(labels=c("CG", "H3wt", "H3mut", "IDHmut-codel", "IDHmut-noncodel", "IDHwt"))

pdf(file = "SF1C.pdf",
    height = 10, width = 15,
    bg = "transparent",
    useDingbats = FALSE)

barplot_all

dev.off()
```

***

## Fig 1D

>Plot CNS tumors.

```{r}
master_mutrate_table_cns_cg_who <- mutrate_table %>%
  mutate(who_subtype = case_when(cohort == "Canine Glioma" ~ "Canine Glioma",
                                 TRUE ~ who_subtype)) %>%
  filter(tumor_location == "Brain", total_per_mb < 10) %>%
  mutate(ctype = reorder(x = who_subtype, X = total_per_mb, FUN = median)) %>%
  dplyr::select(-who_subtype) %>%
  dplyr::rename(who_subtype = ctype)

log1py_breaks_cns = c(0, 0.5, 1, 2, 5, 10)

mutrate_plot1_cns_cg_who <- ggplot(master_mutrate_table_cns_cg_who,
                                   aes(x = who_subtype,
                                       y = total_per_mb))

## color x-axis based on cohort
mutrate_plot1_cns_axis_colour_cg_who <- master_mutrate_table_cns_cg_who[
    match(levels(master_mutrate_table_cns_cg_who$who_subtype), master_mutrate_table_cns_cg_who$who_subtype, nomatch = 0),
    "cohort"
    ] %>%
  mutate(axis_colour = case_when(grepl("Adult", cohort) ~ "#8da0cb",
                                 grepl("Pedia", cohort) ~ "#fc8d62",
                                 grepl("Canine", cohort) ~ "#66c2a5",
                                 TRUE ~ "#ffffff")) %>%
  pull(axis_colour)

fig_1D <- mutrate_plot1_cns_cg_who +
  geom_boxplot(aes(fill = cohort)) +
  geom_jitter(width = 0.2) +
  scale_y_continuous(trans = "log1p", breaks = log1py_breaks_cns) +
  cowplot::theme_cowplot(font_size = 22) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, colour = mutrate_plot1_cns_axis_colour_cg_who),
        legend.position = "bottom", legend.key.size = unit(1, "cm")) +
  labs(x = "CNS cancer types across 1,443 patients",
       y = "Total coding snvs+indels per MB in log(x+1) scale") +
  scale_fill_manual(values = c("#66c2a5", "#8da0cb", "#fc8d62"),
                    name="CNS Tumors",
                    labels=c("Canine", "Adult GBM", "Pediatric"))

cowplot::save_plot("F1D.pdf", fig_1D,
                   base_height = 12, base_width = 16,
                   dpi = "retina", useDingbats=FALSE)
```

### Suppl Fig 1D

>Plot pancancer tumors

??? error "unexpected `,` error"
    Running code by copy-paste may result in error related to R syntax, e.g., `Error: unexpected ',' in "`. If so, try to copy code first in text editor or RStudio app with lint feature. Please [report bugs](https://github.com/TheJacksonLaboratory/canineglioma/issues) by submitting a GitHub issue.

```{r}
master_mutrate_table_cg_who <- mutrate_table %>%
  mutate(who_subtype = case_when(cohort == "Canine Glioma" ~ "Canine Glioma",
                                 TRUE ~ who_subtype)) %>%
  mutate(ctype = reorder(x = who_subtype, X = total_per_mb, FUN = median)) %>%
  dplyr::select(-who_subtype) %>%
  dplyr::rename(who_subtype = ctype)

log1py_breaks = c(0, 0.5, 1, 2, 5, 10,20,50,100,300,500)

mutrate_plot1_all_cg_who <- ggplot(master_mutrate_table_cg_who,
                                   aes(x = who_subtype,
                                       y = total_per_mb))

## color x-axis based on cohort
mutrate_plot1_axis_colour_cg_who <- master_mutrate_table_cg_who[
    match(levels(master_mutrate_table_cg_who$who_subtype),
    master_mutrate_table_cg_who$who_subtype, nomatch = 0),
    "cohort"
    ] %>%
  mutate(axis_colour = case_when(grepl("Adult", cohort) ~ "#8da0cb",
                                 grepl("Pedia", cohort) ~ "#fc8d62",
                                 grepl("Canine", cohort) ~ "#66c2a5",
                                 TRUE ~ "#ffffff")) %>%
  pull(axis_colour)

suppl_fig_1D <- mutrate_plot1_all_cg_who +
  geom_boxplot(aes(fill = tumor_location)) +
  geom_jitter(width = 0.2) +
  scale_y_continuous(trans = "log1p", breaks = log1py_breaks) +
  cowplot::theme_cowplot(font_size = 22) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1,
    colour = mutrate_plot1_axis_colour_cg_who),
        legend.position = "bottom", legend.key.size = unit(1, "cm")) +
  labs(x = "Cancer type across 4,840 patients",
       y = "Total coding snvs+indels per MB in log(x+1) scale")

cowplot::save_plot("SF1D.pdf", suppl_fig_1D,
                   base_height = 12, base_width = 16,
                   dpi = "retina", useDingbats=FALSE)
```

### Suppl Fig 1E

>Comparing mutational burden across molecular subtypes.

```{r}
high_grade_gliomas <- cgp_suppl_sample_info_master %>%
  filter(Tumor_Grade == "High") %>%
  pull(Tumor_Sample_Barcode)

mutrate_table_cns_hgg_idh_classes_cg_who <- master_mutrate_table_cns_cg_who %>%
  filter(grepl("IDH|H3", who_subtype) | 
           (cohort == "Canine Glioma" & sample %in% high_grade_gliomas)) %>%
  dplyr::rename(who_subtype_orig = who_subtype) %>%
  mutate(who_subtype_orig = if_else(cohort == "Canine Glioma", "Canine HGG", as.character(who_subtype_orig))) %>%
  mutate(who_subtype = reorder(x = who_subtype_orig, X = total_per_mb, FUN = median))

mutrate_comparisons_idh_classes_cg_who <- list(c("Canine HGG", "H3wt"),
                                               c("Canine HGG", "H3mut"),
                                               c("H3wt", "IDHmut-codel"),
                                               c("H3mut", "IDHmut-codel"),
                                               c("H3mut", "H3wt"),
                                               c("Canine HGG", "IDHwt"),
                                               c("Canine HGG", "IDHmut-non-codel"),
                                               c("Canine HGG", "IDHmut-codel"))

suppl_fig_1E <- ggpubr::ggboxplot(mutrate_table_cns_hgg_idh_classes_cg_who,
    x = "who_subtype", y = "total_per_mb",
    add = "jitter", bxp.errorbar = TRUE, notch = TRUE,
    font.label = list(size = 18, color = "black"),
    ylab = "Somatic Mutation Rate\nper MB",
    xlab = "Molecular Subtype",
    fill = c("#fc8d62", "#fc8d62", "#66c2a5",
        "#8da0cb", "#8da0cb", "#8da0cb")) +
  theme(axis.text = element_text(size = 18),
        axis.title = element_text(size = 20)) +
  ggpubr::stat_compare_means(comparisons = mutrate_comparisons_idh_classes_cg_who,
    method = "wilcox.test")

cowplot::save_plot("SF1E.pdf",
                   suppl_fig_1E,
                   base_height = 12, base_width = 16,
                   dpi = "retina",
                   useDingbats = FALSE)
```

***

## F2A

>Somatic copy-number aberrations (SCNA) lanscape of glioma driver genes. Please run required code from [Fig 1A](/figures/F1A/) prior to running code from this page.

```{r}
pdf("F2A.pdf",
    width = 16, height = 12, pointsize = 14, useDingbats = FALSE)

mafplot_2_scna <- oncoplot(maf = cgp_maftools_gistic_matched,
    genes = oncoprint_genes_gistic,
    additionalFeature = c("dnds_mutsig", "Driver"),
    removeNonMutated = TRUE,
    keepGeneOrder = FALSE,
    writeMatrix = FALSE,
    sortByMutation = FALSE,
    fontSize = 1.6,
    legendFontSize = 2,
    annotationFontSize = 1.6,
    clinicalFeatures = c("Tumor_Morphology", "Tumor_Grade",
                        "Tumor_Location"),
    annotationColor = ann_colors,
    colors = varcall_colors[
    c(levels(cgp_maftools_gistic_matched@data$Variant_Classification),
    "Multi_Hit")
    ],
    logColBar = FALSE,
    gene_mar = 10,
    bgCol = "#ffffff",
    borderCol = "#CCCCCC")

dev.off()
```

>Somatic copy numbers were classified as follows based on GISTIC2 values.

| GISTIC2 value | Estimated copy number | Classification |
|---|---|---|
| $< -1.29$ | -2 | Deep deletion |
| $-1.29 <> -.65$ | -1 | Loss of Heterozygosity (LOH) |
| $-.65 <> 2.0$ | 0-1 | Copy-neutral or low-level amplification |
| $>2.0$ | 2 | High-level amplification |

***

## Fig 2B

>Comparative aneuploidy burden. Related code is available at the GLASS consortium [marker paper code repository](https://github.com/TheJacksonLaboratory/GLASS/tree/master/sql/cnv), namely, `recapseg_postgres.sql` and `taylor_aneuploidy.sql`.

```{r}
fig_2B <- ggpubr::ggboxplot(merged_anp_scores %>%
    filter(!grepl("_LGG", who_subtype_alt2)) %>%
    mutate(cohort = ordered(who_subtype_alt2,
    levels = c("Canine Glioma",
         "H3wt", "H3mut",
         "IDHmut-codel",
         "IDHmut-non-codel",
         "IDHwt"))),
    x = "cohort", y = "aneuploidy",
    add = "jitter", notch = TRUE,
    bxp.errorbar = TRUE,
    xlab = "",
    ylab = "Fraction of genome\nwith aneuploidy",
    fill = c("#66c2a5",
           "#fc8d62", "#fc8d62",
           "#8da0cb", "#8da0cb" , "#8da0cb")) +
    ggpubr::stat_compare_means(ref.group = "Canine Glioma",
    method = "wilcox.test",
                 label = "p.signif", hide.ns = TRUE,
                 size = 6) +
    cowplot::theme_cowplot(font_size = 24) +
    ## make sure that breaks and labels matches order
    ## defined using levels in ordered function above
    scale_x_discrete(breaks = c("Canine Glioma",
                  "H3wt", "H3mut",
                  "IDHmut-codel",
                  "IDHmut-non-codel",
                  "IDHwt"),
       labels = c("Canine\nGlioma",
                  "H3wt", "H3mut",
                  "IDHmut\ncodel",
                  "IDHmut\nnon-codel",
                  "IDHwt")) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title.x = element_blank())

cowplot::save_plot("F2B.pdf",
          fig_2B,
          base_height = 4, base_width = 5,
          dpi = "retina", useDingbats = FALSE)
```

### Suppl Fig 2E

>Aneuploidy vs. somatic mutation burden.

```{r}
p3_cgp_anp_mutrate_table <- cgp_aneuploidy_metrics_master %>%
    filter(!is.na(adusted_coding_mutrate), !is.na(aneuploidy)) %>%
    mutate_if(bit64::is.integer64, as.integer) %>%
    mutate(mutation_load = if_else(
        ntile(adusted_coding_mutrate, 2) > 1,
        "HIGH", "LOW"))

suppl_fig_2E <- ggpubr::ggboxplot(p3_cgp_anp_mutrate_table,
    "mutation_load", "aneuploidy_score",
    add = "jitter", notch = TRUE,
    xlab = "Coding Mutational Rate",
    ylab = "Anueploidy Score",
    bxp.errorbar = TRUE,
    font.label = list(size = 16, color = "black", face = "bold")) +
    ggpubr::stat_compare_means(method = "wilcox.test", size = 6) +
    cowplot::theme_cowplot(font_size = 24)

save_plot("SF2E.pdf",
          suppl_fig_2E,
          base_height = 12, base_width = 16,
          dpi = "retina",
          useDingbats = FALSE)
```

***

## Fig 2C

`mat_aneuploidy_summary_mostvar_cgAll` matrix contains proportion of patients per cohort with arm-level aneuploidy, and was based on [arm-level aneuploidy metrics](/methods/S11_aneuploidy/). Related code is available at the GLASS consortium [marker paper code repository](https://github.com/TheJacksonLaboratory/GLASS/tree/master/sql/cnv), namely, `recapseg_postgres.sql` and `taylor_aneuploidy.sql`.

```{r}
pheatmap::pheatmap(mat_aneuploidy_summary_mostvar_cgAll,
    cluster_rows = FALSE, cluster_cols = TRUE,
    cellwidth = 12, cellheight = 12,
    fontsize_col = 8, fontsize_row = 12,
    labels_col = colnames(mat_aneuploidy_summary_mostvar_cgAll),
    filename = "F2C.pdf",
    width = 12, height = 4, angle_col = 45,
    gaps_row = c(1,4),
    useDingbats = FALSE)
```

***

## Fig 2D

>Clonality of potential driver genes with respect to intra-tumor heterogeneity as measured with Shannon entropy.

??? error "Error: unexpected symbol in...`who_subtype`..."
    Prefer runnning code in RStudio. This error does not appear when running code in RStudio but pops up when running on command-line R. Likely due to font ligatures and/or use of of line breaks.

```{r}
fig_2D <- ggplot(merged_ccf_vaf_df %>%
    ## Pediatric LGG has minimal somatic
    ## drivers, i.e., ~ quiet genome
    filter(who_subtype != "PG_LGG"), 
    aes(x = Shannon_entropy,
        y = Cellular_Prevalence,
     label = Hugo_Symbol)) +
    geom_point(alpha=0.3,
        aes(colour = clone_type, size = VAF)) +
    scale_size(range = c(.1, 10), name = "VAF") +
    ggrepel::geom_label_repel(
        data = . %>%
                filter(Hugo_Symbol %in%
                    c(oncoprint_genes_snv_scna, "CIC")) %>%
                group_by(who_subtype, clone_type) %>%
                add_count(Hugo_Symbol, sort = T) %>%
                mutate(Hugo_Symbol = sprintf("%s (%s)", Hugo_Symbol, n)) %>%
                filter((who_subtype != "PG_H3wt" & n >= 1) | 
                       (who_subtype == "PG_H3wt" & n >= 2)) %>%
                ungroup() %>%
                distinct(Hugo_Symbol, .keep_all = TRUE),
    nudge_x = 0.25,
    direction = "y",
    hjust = 0,
    force = 2,
    segment.size = 0.2,
    size = 6,
    seed = 1234) +
    facet_wrap(~ who_subtype) +
    cowplot::theme_cowplot(font_size = 24) +
    theme(legend.position = "bottom",
    legend.box = "vertical",
    legend.text = element_text(size = 14)) +
    guides(colour = guide_legend(override.aes = list(size = 10, alpha = 1)))

cowplot::save_plot("F2D.pdf",
          fig_2D,
          base_height = 12, base_width = 16,
          dpi = "retina", useDingbats = FALSE)
```

### Suppl Fig 2D

>Comparative intra-tumor heterogeneity measured using Shannon entropy.

```{r}
suppl_fig_2D <- ggpubr::ggboxplot(updated_merged_entropy_df_ridges,
    x = "who_subtype", y = "Shannon_entropy",
    add = "jitter",
    bxp.errorbar = TRUE, notch = FALSE,
    font.label = list(size = 18, color = "black"),
    ylab = "Shannon Diversity Index",
    xlab = "Molecular Subtype",
    fill = c("#8da0cb", "#8da0cb",
          "#66c2a5",
          "#fc8d62", "#fc8d62", "#fc8d62")) +
    theme(axis.text = element_text(size = 18),
    axis.title = element_text(size = 20)) +
    ggpubr::stat_compare_means(ref.group = "Canine Glioma",
    method = "wilcox.test",
    label = "p.format", hide.ns = TRUE,
    size = 5, label.y = 1.1)

save_plot("SF2D.pdf",
          suppl_fig_2D,
          base_height = 12, base_width = 16,
          dpi = "retina",
          useDingbats = FALSE)
```

***

## Fig 3A

### Required metadata

*   Canine glioma patients with [outlier mutation profile](/methods/S13_mut_sigs/).

```{r}
## filter out missing values in mutation rate column
cgp_sample_info_maftools_hypermuts <- cgp_suppl_sample_info_master %>%
  filter(!is.na(adusted_coding_mutrate)) %>%
  arrange(desc(adusted_coding_mutrate))

outlier_model_hypermut <- lm(adusted_coding_mutrate ~ Matched_Normal +
    Tissue_Archival, data = cgp_sample_info_maftools_hypermuts)

summary(outlier_model_hypermut)

## Hypermut outliers after correcting for archival and paired vs tonly mode
hypermut_true_outliers <- car::outlierTest(outlier_model_hypermut,
    cutoff = 0.05,
    labels = cgp_sample_info_maftools_hypermuts$Tumor_Sample_Barcode)

sprintf("hypermut case: %s", names(hypermut_true_outliers$rstudent))

cgp_sample_info_non_hypermut <- cgp_sample_info_maftools_hypermuts %>% 
    filter(!Tumor_Sample_Barcode %in%
    names(hypermut_true_outliers$rstudent))
         
#### outliers in non-hypermut cases
## cases with mut rate exceeding chi sq distribution by > 95% 
outlier_in_non_hypermut_chisq_95 <- cgp_sample_info_non_hypermut[
    outliers::scores(cgp_sample_info_non_hypermut$adusted_coding_mutrate,
    type = "chisq", prob = 0.95),] %>%
    pull(Tumor_Sample_Barcode)

sprintf("outlier among non_hypermut: %s", outlier_in_non_hypermut_chisq_95)

cgp_outlier_mutant_cases <- c(names(hypermut_true_outliers$rstudent),
    outlier_in_non_hypermut_chisq_95)

cgp_outlier_mutant_cases
```

>"i_8228" "i_95FC" "i_BD58" "i_1C36" "i_502F" "i_D732" "i_6B06" "i_56B5"

*   Pick random 9/73 cases without outlier mutation profile.

```{r}
## nonhm_sampled <- sample(ncol(fit_res_nonhm$contribution),
##     ncol(fit_res_hm$contribution), replace = FALSE)

## following cases were used in the right-side panel of fig 3A
nonhm_sampled <- c(28,48,30,57,54,40,37,7,62)
```

*  Deconvolute known human mutational signatures

>Fitted mutational signatures, i.e., `fit_res`, `fit_res_hm` and `fit_res_nonhm` were derived using `fit_to_signatures` function from *MutationalPatterns* R package.

```{r}
## mut_mat object is within /data/cgp_base_objects_v1.0.RData
## mut_mat <- MutationalPattern::mut_matrix(vcf_list = vcfs,
##    ref_genome = ref_genome)

fit_res <- fit_to_signatures(mut_mat, cancer_signatures)

hypermutators <- c("i_8228", "i_95FC", "i_BD58", "i_1C36",
    "i_502F", "i_D732", "i_6B06", "i_56B5")

non_hypermutators <- colnames(mut_mat)[!colnames(mut_mat) %in% hypermutators]

## non-hypermutators or one without outlier mutation profile
fit_res_nonhm <- fit_to_signatures(mut_mat[, colnames(mut_mat) %in%
    non_hypermutators], cancer_signatures)

## hypermutators or one with outlier mutation profile
fit_res_hm <- fit_to_signatures(mut_mat[, colnames(mut_mat) %in%
    hypermutators], cancer_signatures)
```

*   Merge signatures by proposed mechanism

```{r}
hm_sigs <- hm_nonhm_topsigs_tbl %>% filter(sig_enrichment != "NON_HM") %>%
    pull(sig_id)

nonhm_sigs <- hm_nonhm_topsigs_tbl %>% filter(sig_enrichment != "HM") %>%
    pull(sig_id)

fit_res_hm_merged <- as_tibble(fit_res_hm$contribution[hm_sigs, ],
    rownames = "sig_id") %>%
    left_join(hm_nonhm_topsigs_tbl %>% dplyr::select(sig_id, sig_group)) %>%
    gather(patient_id, sig_score, starts_with("i_")) %>%
    group_by(sig_group, patient_id) %>%
    mutate(group_score = sum(sig_score)) %>%
    dplyr::select(sig_group, group_score, patient_id, sig_id) %>%
    spread(patient_id, group_score) %>%
    distinct(sig_group, .keep_all = TRUE) %>%
    ungroup() %>%
    dplyr::select(-sig_id) %>%
    column_to_rownames("sig_group")

fit_res_nonhm_merged <- as_tibble(fit_res_nonhm$contribution[nonhm_sigs, ],
    rownames = "sig_id") %>%
    left_join(hm_nonhm_topsigs_tbl %>% dplyr::select(sig_id, sig_group)) %>%
    gather(patient_id, sig_score, starts_with("i_")) %>%
    group_by(sig_group, patient_id) %>%
    mutate(group_score = sum(sig_score)) %>%
    dplyr::select(sig_group, group_score, patient_id, sig_id) %>%
    spread(patient_id, group_score) %>%
    distinct(sig_group, .keep_all = TRUE) %>%
    ungroup() %>%
    dplyr::select(-sig_id) %>%
    column_to_rownames("sig_group")

## group cancer signatures context proportion for HM

cancer_signatures_merged <- as_tibble(cancer_signatures,
    rownames = "context") %>%
    gather(sig_id, sig_score, -context) %>%
    left_join(hm_nonhm_topsigs_tbl %>% dplyr::select(sig_id, sig_group)) %>%
    filter(!is.na(sig_group)) %>%
    group_by(sig_group, context) %>%
    mutate(group_score = sum(sig_score)) %>%
    filter(!is.na(group_score)) %>%
    ungroup() %>%
    dplyr::select(sig_group, group_score, context, sig_id) %>%
    distinct(sig_group, context, group_score, .keep_all = FALSE) %>%
    spread(sig_group, group_score) %>%
    distinct(context, .keep_all = TRUE) %>%
    column_to_rownames("context")

hm_nonhm_sig_group_colours <- hm_nonhm_topsigs_tbl %>%
    distinct(sig_group, .keep_all = TRUE) %>%
    pull(colour_hex)

names(hm_nonhm_sig_group_colours) <- hm_nonhm_topsigs_tbl %>%
    distinct(sig_group, .keep_all = TRUE) %>%
    pull(sig_group)
```


*   Function to plot signature contribution

>Rewrite of `plot_contribution` function from *MutationalPatterns* R package.

```{r}
cgp_plot_contribution <- function (contribution,
    signatures, index = c(), coord_flip = FALSE,
    mode = "relative", palette = c())
    {
        if (!(mode == "relative" | mode == "absolute"))
          stop("mode parameter should be either 'relative' or 'absolute'")

        if (length(index > 0)) {
          contribution = contribution[, index]
        }
        
        Sample = NULL
        Contribution = NULL
        Signature = NULL
        
        if (mode == "relative") {
            m_contribution = data.table::melt(contribution, "sig_id")
            colnames(m_contribution) = c("Signature",
                "Sample", "Contribution")

            plot = ggplot(m_contribution, aes(x = factor(Sample),
              y = Contribution, fill = factor(Signature), order = Sample)) +
              geom_bar(position = "fill", stat = "identity",
                colour = "black") +
              labs(x = "", y = "Relative contribution") + theme_bw() +
              theme(panel.grid.minor.x = element_blank(),
                panel.grid.major.x = element_blank()) +
              theme(panel.grid.minor.y = element_blank(),
                panel.grid.major.y = element_blank())
        } else {
            if (missing(signatures))
              stop(paste("For contribution plotting in mode 'absolute':",
                  "also provide signatures matrix"))

            total_signatures = colSums(signatures)
            abs_contribution = contribution[,-1] * total_signatures
            rownames(abs_contribution) = contribution %>%
                pull(sig_id)

            m_contribution = data.table::melt(abs_contribution, "sig_id")
            colnames(m_contribution) = c("Signature",
                "Sample", "Contribution")

            plot = ggplot(m_contribution, aes(x = factor(Sample),
              y = Contribution, fill = factor(Signature), order = Sample)) +
              geom_bar(stat = "identity", colour = "black") +
              labs(x = "", y = "Absolute contribution \n (no. mutations)") +
              theme_bw() +
              theme(panel.grid.minor.x = element_blank(),
                panel.grid.major.x = element_blank()) +
              theme(panel.grid.minor.y = element_blank(),
                panel.grid.major.y = element_blank())
        }

        if (length(palette) > 0) {
            plot = plot +
                scale_fill_manual(name = "Signature", values = palette)
        } else {
            plot = plot +
                scale_fill_discrete(name = "Signature")
        }

        if (coord_flip) {
            plot = plot +
                coord_flip() +
                xlim(rev(levels(factor(m_contribution$Sample))))
        } else {
            plot = plot + xlim(levels(factor(m_contribution$Sample)))
        }

        return(plot)
    }
```

### Fig 3A

>Plot deconvoluted signature contribution after grouping signatures based on proposed mechanism.

```{r}
## plot canine glioma patients with outlier mutation profile

plotsig_hm_rel_merged <- cgp_plot_contribution(as_tibble(
    fit_res_hm_merged, rownames = "sig_id"),
    cancer_signatures_merged,
    coord_flip = FALSE,
    mode = "relative", palette = hm_nonhm_sig_group_colours) +
    theme_cowplot(font_size = 20) +
    theme(axis.text.x = element_blank(),
        legend.position = "bottom",
        legend.text = element_text(size = 14)) +
    labs(x = "Canine Gliomas with outlier mutation profile\n(n=8)") +
    guides(fill = guide_legend(
        title = "Known signatures grouped by\ntheir proposed mechanism",
    title.position = "bottom",
    label.theme = element_text(size = 14),
    keywidth=0.8,
    keyheight=0.8,
    default.unit="cm", nrow = 2,byrow=TRUE))

## plot randomly sampled 9 canine patients without outlier mutation profile

plotsig_nonhm_rel_sampled_merged <- cgp_plot_contribution(as_tibble(
    fit_res_nonhm_merged[ ,nonhm_sampled], rownames = "sig_id"),
    cancer_signatures_merged,
    coord_flip = FALSE,
    mode = "relative", palette = hm_nonhm_sig_group_colours) +
    theme_cowplot(font_size = 20) +
    theme(axis.text.x = element_blank(),
        legend.position = "bottom",
        legend.text = element_text(size = 14)) +
        labs(x = "Canine Gliomas with no outlier mutation profile\n(9/73)") +
    guides(fill = guide_legend(
        title = "Signatures grouped by\ntheir proposed mechanism",
    title.position = "bottom",
    label.theme = element_text(size = 14),
    keywidth=0.8,
    keyheight=0.8,
    default.unit="cm", nrow = 2,byrow=TRUE))

ggsave("F3A.pdf",
    grid.arrange(plotsig_hm_rel_merged,
                 plotsig_nonhm_rel_sampled_merged,
        ncol = 2, widths = c(5, 5)),
        width = 18,
        height = 12,
        useDingbats = FALSE)
```

### Suppl Fig 3A

>Plot absolute and relative contribution of dominant signatures.

```{r}
## Select signatures with major contribution > 3rd Qu (2.75)
summary(rowMedians(fit_res$contribution))

select_high <- rownames(fit_res$contribution)[
    which(rowMedians(fit_res$contribution) > 2.75)
    ]

select_high

color_lib <- c('#8e0152','#c51b7d','#de77ae','#f1b6da','#fde0ef',
  '#f7f7f7','#e6f5d0','#b8e186','#7fbc41','#4d9221',
  '#276419')

## Plot contribution
ggsave("SF3A_top_panel.pdf",
    plot_contribution(fit_res$contribution[select_high, ],
         cancer_signatures[ , select_high],
         coord_flip = FALSE,
         mode = "absolute",
         palette = color_lib) +
    theme(axis.text.x = element_text(angle = 90)),
    width = 9,
    height = 5,
    useDingbats = FALSE)

ggsave("SF3A_bottom_panel.pdf",
    plot_contribution(fit_res$contribution[select_high, ],
        cancer_signatures[ , select_high],
        coord_flip = FALSE,
        mode = "relative",
        palette = color_lib) +
    theme(axis.text.x = element_text(angle = 90)),
    width = 9,
    height = 5,
    useDingbats = FALSE)
```

### Suppl Fig 3B

>Boxplot comparison of genomic instability related signatures between canine glioma patients separated by outlier mutation profile.

```{r}
suppl_fig_3B <- ggpubr::ggboxplot(instability_sigs_df,
        x = "hypermut", y = "sig_contrib",
        add = "jitter",
        facet.by = "sig_group",
        scales = "free_y",
        bxp.errorbar = TRUE, notch = TRUE,
        font.label = list(size = 7, color = "black"),
        ylab = "Signature contribution",
        xlab = "Canine Gliomas separated by\noutlier mutation profile") +
    theme(axis.text = element_text(size = 11),
        axis.title = element_text(size = 11)) +
    ggpubr::stat_compare_means(method = "wilcox.test",
    label.y.npc = 0.9, na.rm = TRUE)

ggsave("SF3B.pdf",
       suppl_fig_3B,
       width=12,
       height=8,
       useDingbats = FALSE)
```

***

## Fig 3B

>Compare de-novo mutational signatures in canine and human pediatric and adult glioma with deconvoluted ones from Human Cancers (n=42).

*   For each of three cohorts (canine, human pediatric and adult glioma), de-novo signatures were constructed as follows:

```{r nmf_denovo, eval = FALSE}
nmf_res <- MutationalPatterns::extract_signatures(mut_mat, rank = 3,
    nrun = 100)
denovo_cg_sig <- as.data.frame(t(nmf_res$signatures))
```

*   Resulting de-novo signatures from three cohorts were then merged into a single dataframe.

```{r, merged_denovo_sigs, eval = FALSE}
## merged_ag_pg_cg_denovo_sig is already given in cgp_base_objects_*.RData 
merged_ag_pg_cg_denovo_sig <- as_tibble(ag_denovo_sig, rownames = "sigs") %>%
    bind_rows(as_tibble(pg_denovo_sig, rownames = "sigs")) %>%
    bind_rows(as_tibble(transposed_denovo_cg_sig, rownames = "sigs")) %>%
    column_to_rownames("sigs")
```

*   Calculate cosine similarity using `deconvolution_compare` function in [*Palimpsest*](https://github.com/FunGeST/Palimpsest) R package. Author of Palimpsest package has provided [a detailed documentation](https://github.com/FunGeST/Palimpsest/raw/master/Files/vignette_palimpsest_2.0.pdf) on how to run Palimpsest workflow.

```{r}
## Deconvoluted signatures from human cancers
alt_cancer_sig_transposed <- as.data.frame(t(cancer_signatures))

## rename trinucleotide context pattern to match in de-novo dataframe
colnames(alt_cancer_sig_transposed) <- 
    gsub("([AGCT]{1})(\\[)([AGCT]{1})(\\>)([AGCT]{1})(\\])([AGCT]{1})",
        "\\3\\5_\\1.\\7", perl = TRUE,
        colnames(alt_cancer_sig_transposed))

## calculate cosine similarity
pdf("tmp_SF3D.pdf",
    width = 18, height = 12, pointsize = 12,
    bg = "white",
    compress = FALSE, useDingbats = FALSE)

cos_sim_denovo_cancer_sig_ag_pg_cg <- Palimpsest::deconvolution_compare(
    merged_ag_pg_cg_denovo_sig,
    alt_cancer_sig_transposed)

dev.off()

## Plot Suppl Fig 3D
## Fig3B is a cropped version of Suppl Fig 3D
pheatmap::pheatmap(cos_sim_denovo_cancer_sig_ag_pg_cg,
  filename = "SF3D.pdf",
  width = 12, height = 12,
  useDingbats = FALSE)
```

***

## Fig 3C

>Plot estimated mutational signature contribution per driver gene.

```{r}
## Function to plot each of three panels of Fig 3C
plot_driver_gene_signatures <- function(df, xtitle) {

  tmp_driver_vcf <- df %>%
    filter(Gene_Name %in% oncoprint_genes_snv_scna) %>%
    ## extract columns that have signature wise probability per somatic variant
    dplyr::select(Gene_Name, ends_with("prob")) %>%
    distinct() %>%
    gather(sigs, prop, -Gene_Name) %>%
    mutate(prop = if_else(is.na(prop), 0, prop)) %>%
    group_by(Gene_Name) %>%
    mutate(total_prop = sum(prop),
           prop_scaled = scales::rescale(prop, to = c(0, 1))) %>%
    ungroup() %>%
    filter(!total_prop == 0) %>%
    mutate(sigs = sprintf("S%s", gsub("(Signature\\.)([0-9]{1,2})(\\.prob)", "\\2", sigs))) %>%
    left_join(driver_sigs_groups %>% dplyr::rename(sigs = signature))

    cols_sig_group <- c("Aging" = "#b2182b",
                        "MMR" = "#fddbc7",
                        "HR defect" = "#ffffbf",
                        "S18_Neuroblastoma" = "#542788",
                        "APOBEC_AID" = "#ef8a62",
                        "UV" = "#998ec3",
                        "S16_S25" = "lightgray")

  ggplot(tmp_driver_vcf, aes(Gene_Name, prop_scaled, fill = factor(sig_group))) +
    geom_bar(position = "fill",stat = "identity") +
    scale_y_continuous(labels = scales::percent_format()) +
    coord_flip() +
    scale_fill_manual(values = cols_sig_group) +
    theme_cowplot(font_size = 22) +
    labs(x = "Percentage contribution from signatures",
         y = xtitle,
         fill = "Signatures") +
    theme(legend.title = element_text(color = "blue", size = 14),
          legend.text = element_text(size = 12))
}

## Plot Fig 3C panels
plot_driver_genes_sigs_cg <- plot_driver_gene_signatures(trimmed_cg_vcf,
    "Canine")
plot_driver_genes_sigs_pg <- plot_driver_gene_signatures(trimmed_pg_vcf,
    "Pediatric")
plot_driver_genes_sigs_ag <- plot_driver_gene_signatures(trimmed_ag_vcf,
    "Adult")

ggsave("F3C.pdf",
grid.arrange(plot_driver_genes_sigs_cg + theme(legend.position="bottom"),
            plot_driver_genes_sigs_pg + theme(legend.position="bottom"),
            plot_driver_genes_sigs_ag + theme(legend.position="bottom"),
            ncol = 3, widths=c(4,4,4), heights = c(8,8,8)),
width=20,
height=24,
useDingbats = FALSE)
```

***

### Winning Tables

>Make winning matrix of somatic variants in cancer driver genes (n=73) based on inferred clonality using [*Palimpsest*](https://github.com/FunGeST/Palimpsest) R package. Author of Palimpsest package has provided [a detailed documentation](https://github.com/FunGeST/Palimpsest/raw/master/Files/vignette_palimpsest_2.0.pdf) on how to run Palimpsest workflow.

```{r}
## function to trim vcfs and make winning tables

mk_win_table <- function(win_table_vcf) {

  vcf_timing <- win_table_vcf %>%
    add_count(Gene_Name, timing_state) %>%
    mutate(result = case_when(grepl("early_clonal", timing_state) ~ 5,
                              grepl("early_subclonal", timing_state) ~ 1,
                              grepl("late_clonal", timing_state) ~ -5,
                              grepl("late_subclonal", timing_state) ~ -1,
                              TRUE ~ 0)) %>%
    dplyr::select(Gene_Name, result, n) %>%
    distinct() %>%
    dplyr::select(Gene_Name, result, n) %>%
    ## get weighted counts by multiplying events and occurrence
    mutate(weighted_result = result * n) %>%
    dplyr::rename(freq = n) %>%
    group_by(Gene_Name) %>%
    mutate(mean_result = mean(weighted_result)) %>%
    ungroup() %>%
    arrange(Gene_Name, mean_result) %>%
    distinct(Gene_Name, mean_result) %>%
    ## decide early vs late per driver event using
    ## majority vote
    mutate(timing = case_when(mean_result < 0 ~ "LATE",
                              mean_result > 0 ~ "EARLY",
                              mean_result == 0 ~ "DRAW",
                              TRUE ~ "NONE"))
  
  ## let driver gene compete with one another: combination matrix
  wintbl_combn <- choose(nrow(vcf_timing), 2)
  
  snv_df_wintbl <- as.data.frame(matrix(apply(vcf_timing %>%
                            dplyr::select(Gene_Name,
                            mean_result),
                        2, combn, m = 2),
                    nrow = wintbl_combn),
                    stringsAsFactors = F) %>%
    tbl_df() %>%
    ## rename V1 and V2 to P1 and P2, respectively
    dplyr::rename_at(vars(V1, V2), ~gsub("V", "P", .)) %>%
    ## where _vaf is clonality based estimate if event
    ## is ~clonal (early) or ~subclonal
    dplyr::rename(P1_vaf = V3,
           P2_vaf = V4) %>%
    ## take a win/loss/draw call
    mutate(outcome = case_when(P1_vaf > P2_vaf ~ "P1",
                               P1_vaf < P2_vaf ~ "P2",
                               P1_vaf == P2_vaf ~ "D",
                               TRUE ~ NA_character_)) %>%
    dplyr::select(P1, P2, outcome)
  
  return(snv_df_wintbl)
}

## Get winning tables from vcfs
cg_win_table <- mk_win_table(win_table_cg_vcf)
pg_win_table <- mk_win_table(win_table_pg_vcf)
ag_win_table <- mk_win_table(win_table_ag_vcf)
```

## Fig 3D

>Run *Bradley-Terry model* to get [estimated relative timing of somatic driver events](/methods/S14_mut_timing/).

```{r}
#### Run and plot BT model ####
## default to Canine Glioma
plot_timing_btmodel <- function(win_table = cg_win_table,
                                plot_title = "Canine Glioma",
                                genes2plot = oncoprint_genes_snv_scna) {

  library(tidyverse)
  library(BradleyTerryScalable)
  
  library(igraph)
  library(ggridges)
  
  set.seed(1989)
    
  #### Import somatic winning table ####
  win_table_raw <- win_table
  
  #### create btdata ####       
  toy_data_4col <- codes_to_counts(win_table_raw, c("P1", "P2", "D"))
  toy_btdata <- btdata(toy_data_4col, return_graph = TRUE)
  
  #### fit model: btfit #### 
  ## Prefer using MAP estimate if summary(toy_btdata) returns
  ## not-fully connected network: Needs to be checked manually
  ## and to follow warning flags
  print(summary(toy_btdata))
  
  toy_fit_MLE <- btfit(toy_btdata, 1)
  toy_fit_MAP <- btfit(toy_btdata, 1.1)
  
  summary(toy_fit_MLE, SE = TRUE)
  summary(toy_fit_MAP, SE = TRUE)
  
  coef(toy_fit_MLE)
  coef(toy_fit_MAP)

  vcov(toy_fit_MLE, ref = "home")
  vcov(toy_fit_MAP, ref = "home")
    
  #### get winning prob ####
  
  fitted(toy_fit_MLE, as_df = TRUE)
  fitted(toy_fit_MAP, as_df = TRUE)
  
  btprob(toy_fit_MAP, as_df = TRUE)
  btprob(toy_fit_MLE, as_df = TRUE)
  
  toy_fit_MAP_prob <- btprob(toy_fit_MAP)
  toy_fit_map_df <- btprob(toy_fit_MAP, as_df = TRUE)
  
  toy_fit_map_mat <- as.data.frame(as.matrix(toy_fit_MAP_prob))
  
  ## convert to rank-order df
  toy_fit_map_mat_ranked <- toy_fit_map_mat %>%
    rownames_to_column("driver") %>%
    mutate_at(vars(-driver), ~(1 - cume_dist(.)))
  
  ## probability long table
  toy_fit_map_mat_long_tbl <- toy_fit_map_mat %>%
    rownames_to_column() %>%
    gather(p2, prob, -rowname) %>%
    distinct()
  
  ## rank order long table
  toy_fit_map_mat_ranked_long_tbl <- toy_fit_map_mat_ranked %>%
    gather(p2, rank, -driver) %>%
    distinct()
  
  #### Figure 3D: Timing Plots ####
  
  toy_fit_map_mat_long_tbl <- toy_fit_map_mat %>%
    rownames_to_column() %>%
    ## select rows to plot matching genes found in
    ## known somatic driver genes
    filter(rowname %in% genes2plot) %>%
    gather(p2, prob, -rowname) %>%
    distinct()
  
  #### export timing plots ####
  ggplot(toy_fit_map_mat_long_tbl,
    aes(1-prob, rowname, fill = ..x..), group = p2) +
    geom_density_ridges_gradient(scale = 1, rel_min_height = 0.01) +
    viridis::scale_fill_viridis(name = "Probability", option = "C") +
    labs(title = plot_title,
         x = "Probability being a late event",
         y = "Somatic Drivers") +
    guides(fill = guide_colorbar(barwidth = 15,
                                 barheight = 1.5)) +
    theme_ridges(font_size = 22) +
    theme(legend.position="bottom")

}

## plot per-cohort relative timing probabilities of driver genes
timing_plot_canine <- plot_timing_btmodel(win_table = cg_win_table,
                                plot_title = "Canine Glioma",
                                genes2plot = oncoprint_genes_snv_scna)

timing_plot_peds <- plot_timing_btmodel(win_table = pg_win_table,
                                plot_title = "Human Pediatric Glioma",
                                genes2plot = oncoprint_genes_snv_scna)

timing_plot_adult <- plot_timing_btmodel(win_table = ag_win_table,
                                plot_title = "human Adult Glioma",
                                genes2plot = oncoprint_genes_snv_scna)

library(gridExtra)

ggsave("F3D.pdf",
    grid.arrange(timing_plot_canine,
                timing_plot_peds,
                timing_plot_adult,
                ncol = 3,
                widths = c(4,4,4), heights = c(4,4,4)),
    width = 24,
    height = 24,
    useDingbats = FALSE)
```

***

## Fig 4

>DNA methylation based classification of canine glioma.

### Preprocessing

#### Format data for plotting

```{r}
library(ggplot2)
library(egg)

# Cluster probabilities to determine sample plot order
cgp_RRBS.noNormals_ord <- hclust( dist(cgp_RRBS.noNormals_class_prediction$probabilities, method = "euclidean"), method = "ward.D" )$order

cgp_RRBS.noNormals_class_prediction$probabilities <- cgp_RRBS.noNormals_class_prediction$probabilities[cgp_RRBS.noNormals_ord,]
cgp_RRBS.noNormals_class_prediction$predictions <- factor(cgp_RRBS.noNormals_class_prediction$predictions[cgp_RRBS.noNormals_ord], levels = c("Adult-IDH_mut","Adult-IDH_wt","Pediatric-Tumor"))

# Reformat probabilities table for plotting
cgp_RRBS.noNormals_class_probabilities <- reshape2::melt(cgp_RRBS.noNormals_class_prediction$probabilities)
colnames(cgp_RRBS.noNormals_class_probabilities) <- c("Sample","Predicted_Class","Prediction_Probability")

cgp_RRBS.noNormals_class_probabilities$Sample <- factor(cgp_RRBS.noNormals_class_probabilities$Sample, levels = rownames(cgp_RRBS.noNormals_class_prediction$probabilities)[cgp_RRBS.noNormals_ord])


# Create shorthand barcode for sample text on x-axis (on mutationsFig) consisting of 4 character caseID only
SampleLabel <- unlist(lapply(strsplit(rownames(cgp_RRBS.noNormals_class_prediction$probabilities)[cgp_RRBS.noNormals_ord],"-"), `[`, 3))
SampleLabel <- paste0("i_",SampleLabel)

# Create label for sample text on x-axis (on mutationsFig) to color IDHmut samples
SampleLabelStyle <- rep("plain",length(SampleLabel))
IDHmutsamples <- which(rownames(cgp_RRBS.noNormals_class_prediction$probabilities)[cgp_RRBS.noNormals_ord] %in% c("CGP-S03-E7AB-8AD9ACF7-T1-A2-J05","CGP-S03-1165-177480CC-T1-A2-J05","CGP-S04-1166-3C6CFE5F-T1-A1-J05"))
SampleLabelStyle[IDHmutsamples] <- "bold"
```

#### Create table for plotting annotations

```{r}
cgp_RRBS.noNormals_class_predictions <- data.frame(Sample=rownames(cgp_RRBS.noNormals_class_prediction$probabilities),
                                                Class=cgp_RRBS.noNormals_class_prediction$predictions,
                                                Location=cgp_RRBS.noNormals_sample_info$Tumor_Location,
                                                Tumor_Grade=cgp_RRBS.noNormals_sample_info$Tumor_Grade,
                                                Histology=cgp_RRBS.noNormals_sample_info$Tumor_Morphology)
cgp_RRBS.noNormals_class_predictions$Location <- factor(cgp_RRBS.noNormals_class_predictions$Location, levels = c("Hemispheric","Cerebellar","Midline"))
```

#### Labels/colors for plot

```{r}
class_probabilities_labels <- c("Glioma","Normal","IDHmut","IDHmut\n1p/19q codel","IDHwt","Normal")
class_probabilities_limits <- c("Pediatric-Tumor","Pediatric-Normal","Adult-IDH_mut","Adult-IDH_mut,1p/19q_codel","Adult-IDH_wt","Adult-Normal")
class_probabilities_colors <- c(rep("#fc8d62",2),rep("#8da0cb",4))

# NOTE: 
# Label spacing for centering "Pediatric" over class categories:
## labs(y = "Predicted Class\nAdult                                                Pediatric")
# Label spacing for centering "Adult" over class categories
## labs(y = "Predicted Class\nAdult          Pediatric")
```

#### Plot probabilities and predictions data frame simultaneously

```{r}
cgp_RRBS.noNormals_class_probabilitiesFig.reformatted <- ggplot() +
  geom_tile(data = cgp_RRBS.noNormals_class_probabilities, mapping = aes(x = Sample, y = Predicted_Class, fill = Prediction_Probability), colour = "grey50") +
  geom_point(data = cgp_RRBS.noNormals_class_predictions, mapping = aes(x = Sample, y = Class, shape = Location, size = Tumor_Grade, colour = Histology)) +
  scale_colour_manual(name='Histology', values=c('Astro'='#d95f02', 'Oligo'='#7570b3', 'Undefined'='#1b9e77'), labels = c('Astro','Oligo','Undefined')) +
  scale_size_manual(name='Tumor Grade', values=c('High'=8, 'Low'=4), labels = c('High','Low')) +
  guides(colour = guide_legend("Histology",override.aes=list(size = 4))) +
  guides(shape = guide_legend("Location",override.aes=list(size = 4))) +
  scale_fill_gradient(name = "Prediction\nProbability",
                      low = "white",
                      high = "deepskyblue3",
                      limits = c(0,1)) +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle=90, face=SampleLabelStyle),
        axis.text.y = element_text(color = rev(class_probabilities_colors)),
        axis.title = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(color = "black"),
        legend.title = element_text(size = 10, face = "bold"),
        legend.text = element_text(size = 8),
        plot.margin = margin(1, 1, 0, 1, "cm")) +
  scale_x_discrete(labels=SampleLabel) +
  scale_y_discrete(limits = rev(class_probabilities_limits), 
    labels = rev(class_probabilities_labels)) +
    labs(x = "Sample", y = "Predicted Class\nAdult")
```

#### Add annotations for mutation rate, purity, and age

```{r}
# Age annotation
ageFig <- ggplot(cgp_RRBS.noNormals_sample_info) + geom_point(aes(x = Tumor_Sample_Barcode, y = age_updated)) + 
  geom_hline(yintercept = 2, linetype="dashed", color = "dimgray", cex = 1) + 
  labs(x = "Sample") +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle=90, face=SampleLabelStyle),
        axis.title = element_text(size = 12, face = "bold"),
        plot.margin = margin(0, 1, 1, 1, "cm")) + 
  scale_x_discrete(labels=SampleLabel) +
  scale_y_continuous(name = "Age", breaks=seq(0,12,2), limits = c(0,13))

# Purity annotation
purityFig <- ggplot(data = cgp_RRBS.noNormals_sample_info, mapping = aes(x = Tumor_Sample_Barcode, y = 1, fill = purity)) + 
  geom_tile() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.margin = margin(0, 1, 0, 1, "cm")) +
  scale_fill_gradient(name = "Purity",
                      low = "white",
                      high = "seagreen",
                      limits = c(0,1)) + 
  theme(legend.title=element_text(size=9, face = "bold"),legend.text = element_text(size=8),legend.key.height = unit(0.08, "in"),legend.direction = "horizontal") +
  guides(fill = guide_colourbar(title.position="top", title.hjust = 0.5))

# Mutation annotation
mutationsFig <- ggplot(data = cgp_RRBS.noNormals_sample_info, mapping = aes(x = Tumor_Sample_Barcode, y = 1, fill = adusted_coding_mutrate)) + 
  geom_tile() + 
  # scale_x_discrete(labels=SampleLabel) +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.margin = margin(0, 1, 0, 1, "cm")) +
  scale_fill_gradient(name = "Mutations/Mb",
                      low = "white",
                      high = "red",
                      limits = c(0,2.6)) + 
  theme(legend.title=element_text(size=9, face = "bold"),legend.text = element_text(size=8),legend.key.height = unit(0.08, "in"),legend.direction = "horizontal") +
  guides(fill = guide_colourbar(title.position="top", title.hjust = 0.5))
```

### Fig 4

```{r}
## Combine plots
library(ggpubr)

cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.combinedVer <- cgp_RRBS.noNormals_class_probabilitiesFig.reformatted +
theme(axis.title.x=element_blank(),
 axis.text.x=element_blank(),
 axis.ticks.x=element_blank())

cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.w_purity_mutations_age <- ggarrange(cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.combinedVer, purityFig, mutationsFig, ageFig,nrow = 4,ncol = 1, heights = c(3,0.25,0.25,0.5), padding = unit(0,"line"))

pdf("F4.pdf",
    width = 18, height = 12, pointsize = 12,
    bg = "white",
    compress = FALSE, useDingbats = FALSE)

cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.w_purity_mutations_age

dev.off()
```

### Suppl Fig 4A

>Somatic mutational lanscape, including copy-number aberrations of COSMIC cancer genes, grouped by [DNA methylation classification](/methods/S15_class_predict_methy/). 

```{r}
## Keep sample info for cases with RRBS data
cgp_methylation_cases <- cgp_suppl_sample_info_master %>%
  filter(!is.na(Methylation_Predicted_Class), WGS == "YES") %>%
  tbl_df() %>% 
  arrange(Methylation_Predicted_Class) %>%
  dplyr::select(Tumor_Sample_Barcode,
                Methylation_Predicted_Class,
                Methylation_Prediction_Probability,
                everything())

cgp_methylation_cases_peds <- cgp_methylation_cases %>%
  filter(Methylation_Predicted_Class == "Pediatric-Tumor")

cgp_methylation_cases_adult <- cgp_methylation_cases %>%
  filter(grepl("IDH", Methylation_Predicted_Class))

## subset MAF object based on methylation classification
cgp_maftools_gistic_methy_peds <- subsetMaf(cgp_maftools_gistic_methy,
    tsb = cgp_methylation_cases_peds$Tumor_Sample_Barcode)

cgp_maftools_gistic_methy_adult <- subsetMaf(cgp_maftools_gistic_methy,
    tsb = cgp_methylation_cases_adult$Tumor_Sample_Barcode)

## Specify colors for sample annotations
ann_colors_methy <- list(Tumor_Morphology = c(Astro = "#d95f02",
    Oligo = "#7570b3",
    Undefined = "#1b9e77"),
    Tumor_Grade = c(High = "#FEB24C", Low = "#FFEDA0"),
    Tumor_Location = c(Hemispheric = "#a6cee3", Cerebellar = "#1f78b4",
                    midline = "#e31a1c", UN = "#fb9a99"),
    Methylation_Predicted_Class = c(`Adult-IDH_mut` = "#a6cee3",
                                 `Adult-IDH_wt` = "#1f78b4",
                                 `Pediatric-Tumor` = "#e31a1c"))

## Suppl Fig 4A
pdf("SF4A.pdf",
    width = 18, height = 12, pointsize = 12,
    fonts = "Arial", bg = "white",
    compress = FALSE, useDingbats = FALSE)

suppl_fig_4A <- coOncoplot(m1 = cgp_maftools_gistic_methy_peds,
    m2 = cgp_maftools_gistic_methy_adult,
    m1Name = 'Pediatric-like',
    m2Name = 'Adult-like',
    genes = oncoprint_genes_snv_scna,
    removeNonMutated = FALSE,
    additionalFeature1 = c("dnds_mutsig", "Driver"),
    additionalFeature2 = c("dnds_mutsig", "Driver"),
    clinicalFeatures1 = c("Methylation_Predicted_Class",
                       "Tumor_Morphology", "Tumor_Grade",
                       "Tumor_Location"),
    clinicalFeatures2 = c("Methylation_Predicted_Class",
                       "Tumor_Morphology", "Tumor_Grade",
                       "Tumor_Location"),
    sortByAnnotation2 = TRUE,
    legendFontSize = 1.2,
    annotationFontSize = 1.2,
    annotationColor1 = ann_colors_methy,
    annotationColor2 = ann_colors_methy,
    colors = varcall_colors[
    c(levels(cgp_maftools_gistic_methy_peds@data$Variant_Classification),
    "Multi_Hit")
    ],
    bgCol = "#ffffff",
    borderCol = "#CCCCCC",
    sepwd_genes1 = 0.8,
    sepwd_genes2 = 0.8)

dev.off()
```

### Suppl Fig 4B

>DNA methylation age of human and canine glioma.

Brewing...

***

## Fig 5

>IHC Panels

### Required metadata

```{r}
library(ggplot2)
library(tidyr)

dat <- fig5b_dat

type_map <- c(Ped = "Pediatric", Dog = "Canine", Adult = "Adult")
dat$type <- type_map[dat$type]

## rename Low -> Low Grade, High -> High Grade
grade_rename <- c("Low"="Low Grade","High"="High Grade")
dat$Grade <- grade_rename[dat$Grade]

## create mapping from sample name to type
temp <- table(dat$type,dat$Sample.Name)
samp_type <- apply(temp,2,function(x){which(x>0)})
samp_type[] <- rownames(temp)[samp_type]

## create mapping from sample name to grade
temp <- table(dat$Grade,dat$Sample.Name)
samp_grade <- apply(temp,2,function(x){which(x>0)})
samp_grade[] <- rownames(temp)[samp_grade]

## Summary of Data Set
n <- nrow(dat)
row_miss <- rowSums(is.na(dat[,5:8])) 
dat <- dat[row_miss < 0.5,]
```

>There were `r n` total fields measured and `r sum(row_miss > 0.5)` fields with missing data. Fields with missing positivies were removed from further analysis.

*   Number of fields by marker

```{r}
kable(table(dat$marker),col.names=c("Marker","# Fields"))
```

*   Counts of number of fields by marker for each sample:

```{r}
tab <- table(dat$Sample.Name,dat$marker)
tab <- cbind(Type=samp_type[rownames(tab)],
    Grade=samp_grade[rownames(tab)],tab)
tab <- tab[order(tab[,1]),]
kable(tab,align="c")
```

*   Counts of different types of samples (ped, adult, or breed):

```{r}
temp <- table(dat$Breed,dat$Sample.Name)
samp_breed <- apply(temp,2,function(x){which(x>0)})
samp_breed[] <- rownames(temp)[samp_breed]
kable(table(samp_breed),col.names=c("Sample Type","Number"))
```

*   Marker Distributions

>For each field, the "1+", "2+", and "3+" columns positivity are summed to create a new variable Positivity. The mean of the Positivity is computed across all fields for a given sample and marker. The means are shown in the table below.

```{r}
## create response, sum of percentage positivities
dat$percent_sum <- dat$X1. + dat$X2. + dat$X3.
dat$percent_sum <- 100*dat$percent_sum

dat_sum <- aggregate(dat$percent_sum,
                     by=list(dat$Sample.Name,dat$marker),
                     FUN=mean)
colnames(dat_sum) <- c("Sample.Name","marker","percent_sum")
dat_sum <- data.frame(dat_sum,
                      "type"=samp_type[dat_sum$Sample.Name],
                      "Grade"=samp_grade[dat_sum$Sample.Name],
                      stringsAsFactors=FALSE)
dat_sum_wide <- pivot_wider(dat_sum,
                            names_from=marker,
                            values_from=percent_sum)
ords <- order(dat_sum_wide$type,dat_sum_wide$Sample.Name)
kable(dat_sum_wide[ords,],digits=1)
```

### Fig 5B

>The violin plots below show the distribution of percentage positivity. The violins are constructed using all the fields. The points are subject means. The markers CD14, CD3, and CD79A are low across grade and subject type.

```{r}
cols <- c(Canine="#66c2a5",Pediatric="#fc8d62",Adult="#8da0cb")

ps <- ggplot(dat, aes(x=Grade, y=percent_sum)) +
  geom_violin(aes(fill=type),
               position=position_dodge(width=0.75),
               alpha=.3,color='grey') +
  geom_dotplot(aes(fill=type),
    data=dat_sum,binaxis='y', 
               stackdir='center',
               dotsize=1,
               position=position_dodge(width=0.75),
               col="black") +
  ylab("Percentage Positivity") +
  scale_fill_manual(values=cols) + 
  xlab("") +
  theme_classic() +
  labs(fill="") + 
  coord_cartesian(xlim=c(1.15,1.85))

library(cowplot)

ps <-  ps + facet_wrap( ~ marker,nrow = 2)
ps <- ps + theme(legend.position = c(0.75, 0.15),
                 legend.background = element_rect(fill = "white",
                    colour = NA),
                 axis.text.x=element_text(size=15, 
                                          angle = 0,
                                          vjust=.5),
                 axis.text.y=element_text(size=15, 
                                          angle = 0,
                                          vjust=.5),
                 axis.title.y=element_text(size=20),
                 legend.title=element_text(size = 20),
                 legend.text = element_text(size = 20),
                 strip.text = element_text(size = 20))

ps + theme_cowplot(font_size = 24) +
    theme(legend.position = c(0.75, 0.15))

plot(ps)

pdf(file = "F5B.pdf",
    useDingbats = FALSE,
    width = 20, colormodel="srgb",
    compress = FALSE)

plot(ps)

dev.off()
```

### p-values

>For CD163 and IBA1, mean percentage positivities are compared across sample types (Pediatric, Adult, Canine) within high grade patients. Specifically a Wilcoxon two sample p--value is computed for Pediatric vs. Canine subject means and Pediatric vs. Adult subject means.

```{r}
dat_sum <- dat_sum[dat_sum$Grade=="High Grade",]
out <- data.frame(Marker=rep(NA_character_,4),
                  "Group 1"=rep("Pediatric",4),
                  "Group 2"=rep(NA_character_,4),
                  "p-value"=rep(NA_real_,4),
                  stringsAsFactors=FALSE,
                  check.names=FALSE)

marker <- "IBA1"
comp <- "Canine"
dat_sum_sub <- dat_sum[dat_sum$marker==marker,]
p <- wilcox.test(dat_sum_sub[dat_sum_sub$type=="Pediatric","percent_sum"],
                 dat_sum_sub[dat_sum_sub$type==comp,"percent_sum"])$p.value
out[1,1] <- marker
out[1,3] <- comp
out[1,4] <- p
comp <- "Adult"
p <- wilcox.test(dat_sum_sub[dat_sum_sub$type=="Pediatric","percent_sum"],
            dat_sum_sub[dat_sum_sub$type==comp,"percent_sum"])$p.value
out[2,1] <- marker
out[2,3] <- comp
out[2,4] <- p

marker <- "CD163"
comp <- "Canine"
dat_sum_sub <- dat_sum[dat_sum$marker==marker,]
p <- wilcox.test(dat_sum_sub[dat_sum_sub$type=="Pediatric","percent_sum"],
                 dat_sum_sub[dat_sum_sub$type==comp,"percent_sum"])$p.value
out[3,1] <- marker
out[3,3] <- comp
out[3,4] <- p
comp <- "Adult"
p <- wilcox.test(dat_sum_sub[dat_sum_sub$type=="Pediatric","percent_sum"],
            dat_sum_sub[dat_sum_sub$type==comp,"percent_sum"])$p.value
out[4,1] <- marker
out[4,3] <- comp
out[4,4] <- p

kable(out,digit=4)
```

***

## Fig 5

>[CIBERSORT based deconvolution of immune microenvironment](/methods/S16_expression_immune/#cibersort-based-expression-analysis) in human and canine glioma. 

### Required metadata

```{r}
library(tidyverse)
library(RColorBrewer)

#Clean up cgp info
#Streamline sample barcode, tumor grades (high vs low)
#Only keep 40 cases where we have rnaseq for tumors
#rnaseq
cgp_info[which(cgp_info[,"tumor_grade"]=="GBM"),"tumor_grade"] <- "High"
cgp_info <- cgp_info[-which(is.na(cgp_info[,"tumor_grade"])),]

#Clean up TCGA info
grade_class <- rep("",nrow(tcga_info))
grade_class[which(tcga_info[,"neoplasm_histologic_grade"] == "g2")] <- "Low"
grade_class[which(tcga_info[,"neoplasm_histologic_grade"] == "g3" | tcga_info[,"revised_histology"]=="glioblastoma")] <- "High"
tcga_info[,"grade_class"] <- grade_class
tcga_info <- tcga_info[-which(tcga_info[,"grade_class"]==""),]

cgp_cell_fractions <- cgp_cibersort[,1:22]
tcga_cell_fractions <- tcga_cibersort[,1:22]
ped_cell_fractions <- ped_cibersort[,1:22]
tcga_cell_fractions <- tcga_cell_fractions[grep("-01A-",rownames(tcga_cell_fractions)),]        ## May need to change to A or B

#--------------------------------------
#Histology based splitting 

#Clean data in canines
cgp_fraction_holder <- rownames(cgp_cell_fractions)
cgp_fraction_holder <- strsplit(cgp_fraction_holder,"-")
cgp_fraction_holder <- sapply(cgp_fraction_holder,function(x)paste(x[1],x[2],x[3],sep="-"))
rownames(cgp_cell_fractions) <- cgp_fraction_holder
 
cgp_comxx <- intersect(rownames(cgp_info),rownames(cgp_cell_fractions))
cgp_cell_fractions <- cgp_cell_fractions[cgp_comxx,]
cgp_info <- cgp_info[cgp_comxx,]

cgp_grade <- cgp_info[,"tumor_grade"]
cgp_grade_fractions <- aggregate(cgp_cell_fractions, by = list(Grade = cgp_grade), FUN = mean)

#Clean data in TCGA
tcga_fraction_holder <- rownames(tcga_cell_fractions)
tcga_fraction_holder <- strsplit(tcga_fraction_holder,"__")
tcga_fraction_holder <- sapply(tcga_fraction_holder,function(x)x[2])
tcga_fraction_holder <- strsplit(tcga_fraction_holder,"-")
tcga_fraction_holder <- sapply(tcga_fraction_holder,function(x)paste(x[1],x[2],x[3],sep="-"))
rownames(tcga_cell_fractions) <- tcga_fraction_holder

tcga_comxx <- intersect(rownames(tcga_info),rownames(tcga_cell_fractions))
tcga_cell_fractions <- tcga_cell_fractions[tcga_comxx,]
tcga_info <- tcga_info[tcga_comxx,]

tcga_grade <- tcga_info[,"grade_class"]
tcga_grade_fractions <- aggregate(tcga_cell_fractions, by = list(Grade = tcga_grade), FUN = mean)

#Prepare ped data
ped_grade <- rep("",nrow(ped_cell_fractions))
ped_grade[grep("HGG",rownames(ped_cell_fractions))] <- "High"
ped_grade[grep("LGG",rownames(ped_cell_fractions))] <- "Low"

ped_grade_fractions <- aggregate(ped_cell_fractions, by = list(Grade = ped_grade), FUN = mean)

grade_labels <- c("High","Low")

#What is left: collapse the histology tables and draw the figures
cgp_grade_fractions <- cgp_grade_fractions[,-1]
tcga_grade_fractions <- tcga_grade_fractions[,-1]
ped_grade_fractions  <- ped_grade_fractions[,-1]

cgp_grade_fractions <- t(cgp_grade_fractions)
tcga_grade_fractions <- t(tcga_grade_fractions)
ped_grade_fractions <- t(ped_grade_fractions)

colnames(cgp_grade_fractions) <- paste("cgp_", grade_labels,sep="")
colnames(tcga_grade_fractions) <- paste("TCGA_", grade_labels,sep="")
colnames(ped_grade_fractions) <- paste("ped_", grade_labels,sep="")

grade_fractions_full <- data.frame(cgp_grade_fractions, tcga_grade_fractions, ped_grade_fractions)

collapsed_cells <- c("B cells", "PCs", "CD8", "CD4", "gd T cells", "NKs", "Mono/MP",
                    "DCs","MCs","Eos","PMN cells")
collapsed_grade_fractions <- matrix(0,nrow=length(collapsed_cells),ncol=6)
rownames(collapsed_grade_fractions) <- collapsed_cells
colnames(collapsed_grade_fractions) <- colnames(grade_fractions_full)

collapsed_grade_fractions["B cells",] <- apply(grade_fractions_full[grep("B.cells",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["PCs",] <- apply(grade_fractions_full[grep("Plasma.cells",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["CD8",] <- apply(grade_fractions_full[grep("T.cells.CD8",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["CD4",] <- apply(grade_fractions_full[grep("T.cells.CD4|T.cells.follicular|T.ce..s.regulatory",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["gd T cells",] <- apply(grade_fractions_full[grep("T.cells.gamma",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["NKs",] <- apply(grade_fractions_full[grep("NK.cells",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["Mono/MP",] <- apply(grade_fractions_full[grep("Monocytes|Macrophages",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["DCs",] <- apply(grade_fractions_full[grep("Dendritic.cells",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["MCs",] <- apply(grade_fractions_full[grep("Mast.cells",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["Eos",] <- apply(grade_fractions_full[grep("Eosinophils",rownames(grade_fractions_full)),],2,sum)
collapsed_grade_fractions["PMN cells",] <- apply(grade_fractions_full[grep("Neutrophils",rownames(grade_fractions_full)),],2,sum)

collapsed_grade_label <- as.factor(c(rep("Canine",22),rep("Adult",22),rep("Pediatric",22)))
collapsed_grade <- rep(rep(c("High grade","Low grade"),each=11),3)
collapsed_grade_fraction <- as.numeric(collapsed_grade_fractions[,1:6])
cell <- rep(collapsed_cells,6)
cell <- factor(cell,levels=unique(cell))

collapsed_grade_plot_result <- data.frame(collapsed_grade_fraction,cell,collapsed_grade_label,collapsed_grade)
collapsed_grade_plot_result[,"collapsed_grade_label"] <- factor(collapsed_grade_plot_result[,"collapsed_grade_label"],levels=c("Canine","Adult","Pediatric"))
collapsed_grade_plot_result[,"collapsed_grade"] <- factor(collapsed_grade_plot_result[,"collapsed_grade"],levels=c("Low grade","High grade"))

colors <- c("#4aac8b",
"#a45dcf",
"#64ac48",
"#c563b0",
"#9a963f",
"#686cc5",
"#c78543",
"#6b9ad5",
"#cd5136",
"#c26b7e",
"#cd3c6d")
#colors <- sample(colors,replace=F)
colors <- c("#cd5136","#c78543","#c563b0","#c26b7e","#686cc5","#4aac8b","#6b9ad5","#cd3c6d","#9a963f","#64ac48","#a45dcf")
```

### Fig S5

```{r}
pdf("SF5.pdf",
    width = 5,
    height = 5,
    useDingbats = FALSE)

suppl_fig_S5 <- ggplot(collapsed_grade_plot_result,
    aes(y = collapsed_grade_fraction,
        x = collapsed_grade, fill = cell)) +
    geom_bar(stat="identity",
        width=1, colour="black", size=0.15) +
    facet_wrap(~collapsed_grade_label) +
    scale_fill_manual(values=colors) +
    theme(
    legend.title=element_blank(),
    legend.text=element_text(size=7),
    legend.key.size = unit(0.35, "cm"),
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(),
    plot.background=element_blank(),
    axis.text.x = element_text(angle=45, hjust=1),
    axis.title=element_blank()) +
    scale_x_discrete(expand=c(0,0)) +
    scale_y_continuous(limits=c(0,1.0001),
    expand=c(0,0))

suppl_fig_S5

dev.off()
```
