<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><link href=https://canineglioma.verhaaklab.com/figures/F4/ rel=canonical><meta name=author content="Kevin J. Anderson"><meta name=lang:clipboard.copy content="Copy to clipboard"><meta name=lang:clipboard.copied content="Copied to clipboard"><meta name=lang:search.language content=en><meta name=lang:search.pipeline.stopwords content=True><meta name=lang:search.pipeline.trimmer content=True><meta name=lang:search.result.none content="No matching documents"><meta name=lang:search.result.one content="1 matching document"><meta name=lang:search.result.other content="# matching documents"><meta name=lang:search.tokenizer content=[\s\-\.]+><link rel="shortcut icon" href=../../assets/images/logo.png><meta name=generator content="mkdocs-1.0.4, mkdocs-material-4.4.0"><title>F4 &middot; DNA Methylation based Prediction of Canine Gliomas</title><link rel=stylesheet href=../../assets/stylesheets/application.4031d38b.css><link rel=stylesheet href=../../assets/stylesheets/application-palette.224b79ff.css><meta name=theme-color content=#2196f3><script src=../../assets/javascripts/modernizr.74668098.js></script><link href=https://fonts.gstatic.com rel=preconnect crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,400,400i,700|Fira+Code&display=fallback"><style>body,input{font-family:"IBM Plex Sans","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Fira Code","Courier New",Courier,monospace}</style><link rel=stylesheet href=../../assets/fonts/material-icons.css><link rel=stylesheet href=../../assets/css/katex.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css><script async src="https://www.googletagmanager.com/gtag/js?id=UA-64053990-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-64053990-2');
</script><meta content=https://canineglioma.verhaaklab.com/figures/F4/ property=og:url><meta property=og:title content="F4 · DNA Methylation based Prediction of Canine Gliomas"><meta property=og:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta property=og:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta property=og:image:alt content="Image alt"><meta property=og:image:type content=image/jpeg><meta property=og:image:width content=300><meta property=og:image:height content=300><meta name=twitter:card content=summary><meta name=twitter:site content=@roelverhaak><meta name=twitter:creator content=@roelverhaak><meta property=twitter:title content="F4 · DNA Methylation based Prediction of Canine Gliomas"><meta name=twitter:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta name=twitter:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta name=twitter:image:alt content="Image alt"></head> <body dir=ltr data-md-color-primary=blue data-md-color-accent=blue> <svg class=md-svg> <defs> <!-- <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg> --> <svg xmlns=http://www.w3.org/2000/svg xmlns:xlink=http://www.w3.org/1999/xlink version=1.1 style=width:0;height:0;position:absolute;overflow:hidden;> <defs> <symbol viewbox="0 0 1024 998" aria-labelledby=dpsi-ant-github-title id=__github><title id=dpsi-ant-github-title>icon github</title><path fill=currentColor d="M0 512q0 166 95.5 298.5T343 996q6 1 10 1t6.5-1.5 4-3 2-5 .5-4.5V882q-37 4-66-.5t-45.5-14-29-23.5-17-25.5-9-24T194 780q-9-15-27-27.5t-27-20-2-14.5q50-26 113 66 34 51 119 30 10-41 40-70-116-21-172-86t-56-158q0-87 55-151-22-65 6-137 29-2 65 11.5t50.5 23T384 264q57-16 128.5-16T642 264q13-9 29-19t49-21.5 61-9.5q27 71 6 135 56 64 56 152 0 92-56.5 157.5T615 744q43 43 43 104v129q0 1 1 3 0 6 .5 9t4.5 6 11 3q154-52 251.5-185.5T1024 512q0-104-40.5-199t-109-163.5T711 40.5 512 0 313 40.5t-163.5 109T40.5 313 0 512z"/></symbol> </defs> </svg> </defs> </svg> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay data-md-component=overlay for=__drawer></label> <a href=#preprocessing tabindex=1 class=md-skip> Skip to content </a> <header class=md-header data-md-component=header> <nav class="md-header-nav md-grid"> <div class=md-flex> <div class="md-flex__cell md-flex__cell--shrink"> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-header-nav__button md-logo"> <img src=../../assets/images/logo.png width=24 height=24> </a> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--menu md-header-nav__button" for=__drawer></label> </div> <div class="md-flex__cell md-flex__cell--stretch"> <div class="md-flex__ellipsis md-header-nav__title" data-md-component=title> <span class=md-header-nav__topic> Canine Glioma Project &middot; Verhaak Lab </span> <span class=md-header-nav__topic> F4 &middot; DNA Methylation based Prediction of Canine Gliomas </span> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--search md-header-nav__button" for=__search></label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=query data-md-state=active> <label class="md-icon md-search__icon" for=__search></label> <button type=reset class="md-icon md-search__icon" data-md-component=reset tabindex=-1> &#xE5CD; </button> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=result> <div class=md-search-result__meta> Type to start searching </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <div class=md-header-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> </div> </div> </nav> </header> <div class=md-container> <main class=md-main> <div class="md-main__inner md-grid" data-md-component=container> <div class="md-sidebar md-sidebar--primary" data-md-component=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" data-md-level=0> <label class="md-nav__title md-nav__title--site" for=__drawer> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-nav__button md-logo"> <img src=../../assets/images/logo.png width=48 height=48> </a> Canine Glioma Project &middot; Verhaak Lab </label> <div class=md-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. title=Home class=md-nav__link> Home </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-2 type=checkbox id=nav-2> <label class=md-nav__link for=nav-2> Methods </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-2> Methods </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../methods/S01_samples/ title=Samples class=md-nav__link> Samples </a> </li> <li class=md-nav__item> <a href=../../methods/S02_data_sources/ title="Data Sources" class=md-nav__link> Data Sources </a> </li> <li class=md-nav__item> <a href=../../methods/S03_sample_prep/ title="Sample Preparations" class=md-nav__link> Sample Preparations </a> </li> <li class=md-nav__item> <a href=../../methods/S04_alignments_fprint/ title=Alignments class=md-nav__link> Alignments </a> </li> <li class=md-nav__item> <a href=../../methods/S05_somatic_variants/ title="Somatic SNVs and Indels" class=md-nav__link> Somatic SNVs and Indels </a> </li> <li class=md-nav__item> <a href=../../methods/S06_smgs/ title="Significantly Mutated Genes" class=md-nav__link> Significantly Mutated Genes </a> </li> <li class=md-nav__item> <a href=../../methods/S07_hallmarks/ title="Cancer Hallmarks" class=md-nav__link> Cancer Hallmarks </a> </li> <li class=md-nav__item> <a href=../../methods/S09_mutation_rate/ title="Somatic Mutation Rate" class=md-nav__link> Somatic Mutation Rate </a> </li> <li class=md-nav__item> <a href=../../methods/S10_SCNA/ title="Somatic Copy Number Aberrations" class=md-nav__link> Somatic Copy Number Aberrations </a> </li> <li class=md-nav__item> <a href=../../methods/S11_aneuploidy/ title="Aneuploidy Metrics" class=md-nav__link> Aneuploidy Metrics </a> </li> <li class=md-nav__item> <a href=../../methods/S12_ith/ title="Intratumor Heterogeneity" class=md-nav__link> Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../../methods/S13_mut_sigs/ title="Mutational Signatures" class=md-nav__link> Mutational Signatures </a> </li> <li class=md-nav__item> <a href=../../methods/S14_mut_timing/ title="Molecular Timing of Somatic Drivers" class=md-nav__link> Molecular Timing of Somatic Drivers </a> </li> <li class=md-nav__item> <a href=../../methods/S15_class_predict_methy/ title="Class Prediction using methylation classifier" class=md-nav__link> Class Prediction using methylation classifier </a> </li> <li class=md-nav__item> <a href=../../methods/S16_expression_immune/ title="Immune Microenvironment Analysis" class=md-nav__link> Immune Microenvironment Analysis </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3 type=checkbox id=nav-3 checked> <label class=md-nav__link for=nav-3> Figures </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-3> Figures </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../A1_preload/ title="Start Here" class=md-nav__link> Start Here </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-2 type=checkbox id=nav-3-2> <label class=md-nav__link for=nav-3-2> F1 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-2> F1 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F1A/ title="F1A · Somatic Mutational Landscape" class=md-nav__link> F1A &middot; Somatic Mutational Landscape </a> </li> <li class=md-nav__item> <a href=../F1B/ title="F1B · Somatic Recurrent Hotspot Mutations" class=md-nav__link> F1B &middot; Somatic Recurrent Hotspot Mutations </a> </li> <li class=md-nav__item> <a href=../F1C/ title="F1C · Comparative Pathway Analysis" class=md-nav__link> F1C &middot; Comparative Pathway Analysis </a> </li> <li class=md-nav__item> <a href=../F1D/ title="F1D · Comparative Somatic Mutational Burden" class=md-nav__link> F1D &middot; Comparative Somatic Mutational Burden </a> </li> <li class=md-nav__item> <a href=../S1/ title="S1A · DNA fingerprinting" class=md-nav__link> S1A &middot; DNA fingerprinting </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-3 type=checkbox id=nav-3-3> <label class=md-nav__link for=nav-3-3> F2 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-3> F2 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F2A/ title="F2A · Somatic Copy Number Landscape" class=md-nav__link> F2A &middot; Somatic Copy Number Landscape </a> </li> <li class=md-nav__item> <a href=../F2B/ title="F2B · Comparative Aneuploidy Burden" class=md-nav__link> F2B &middot; Comparative Aneuploidy Burden </a> </li> <li class=md-nav__item> <a href=../F2C/ title="F2C · Comparative Shared Syntenic Aneuploidy" class=md-nav__link> F2C &middot; Comparative Shared Syntenic Aneuploidy </a> </li> <li class=md-nav__item> <a href=../F2D/ title="F2D · Comparative Intratumor Heterogeneity" class=md-nav__link> F2D &middot; Comparative Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../S2/ title="S2 · IGV Plots" class=md-nav__link> S2 &middot; IGV Plots </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-4 type=checkbox id=nav-3-4> <label class=md-nav__link for=nav-3-4> F3 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-4> F3 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F3A/ title="F3A · Somatic Mutational Signatures - Deconvoluted" class=md-nav__link> F3A &middot; Somatic Mutational Signatures - Deconvoluted </a> </li> <li class=md-nav__item> <a href=../F3B/ title="F3B · Somatic Mutational Landscape - de novo" class=md-nav__link> F3B &middot; Somatic Mutational Landscape - de novo </a> </li> <li class=md-nav__item> <a href=../F3C/ title="F3C · Comparative Somatic Mutational Signatures in Driver Genes" class=md-nav__link> F3C &middot; Comparative Somatic Mutational Signatures in Driver Genes </a> </li> <li class=md-nav__item> <a href=../F3D/ title="F3D · Comparative Molecular Timing of Driver Events" class=md-nav__link> F3D &middot; Comparative Molecular Timing of Driver Events </a> </li> <li class=md-nav__item> <a href=../S3/ title="S3 · Suppl Figures" class=md-nav__link> S3 &middot; Suppl Figures </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-5 type=checkbox id=nav-3-5 checked> <label class=md-nav__link for=nav-3-5> F4 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-5> F4 </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-toggle md-nav__toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> F4 &middot; DNA Methylation based Prediction of Canine Gliomas </label> <a href=./ title="F4 · DNA Methylation based Prediction of Canine Gliomas" class="md-nav__link md-nav__link--active"> F4 &middot; DNA Methylation based Prediction of Canine Gliomas </a> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#preprocessing title=Preprocessing class=md-nav__link> Preprocessing </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#format-data-for-plotting title="Format data for plotting" class=md-nav__link> Format data for plotting </a> </li> <li class=md-nav__item> <a href=#create-table-for-plotting-annotations title="Create table for plotting annotations" class=md-nav__link> Create table for plotting annotations </a> </li> <li class=md-nav__item> <a href=#labelscolors-for-plot title="Labels/colors for plot" class=md-nav__link> Labels/colors for plot </a> </li> <li class=md-nav__item> <a href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Plot probabilities and predictions data frame simultaneously" class=md-nav__link> Plot probabilities and predictions data frame simultaneously </a> </li> <li class=md-nav__item> <a href=#add-annotations-for-mutation-rate-purity-and-age title="Add annotations for mutation rate, purity, and age" class=md-nav__link> Add annotations for mutation rate, purity, and age </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-4 title="Fig 4" class=md-nav__link> Fig 4 </a> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../S4/ title="S4 · Suppl Figure 4" class=md-nav__link> S4 &middot; Suppl Figure 4 </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-6 type=checkbox id=nav-3-6> <label class=md-nav__link for=nav-3-6> F5 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-6> F5 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../F5/ title="F5B · Comparative Immunohistochemistry - Quantitative" class=md-nav__link> F5B &middot; Comparative Immunohistochemistry - Quantitative </a> </li> <li class=md-nav__item> <a href=../S5/ title="S5 · Suppl Figure 5" class=md-nav__link> S5 &middot; Suppl Figure 5 </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-4 type=checkbox id=nav-4> <label class=md-nav__link for=nav-4> R Markdown </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-4> R Markdown </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../run_code/ title=How-To class=md-nav__link> How-To </a> </li> <li class=md-nav__item> <a href=../../run_code/run_all/ title=Output class=md-nav__link> Output </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=https://verhaaklab.com/contact title=Contact class=md-nav__link> Contact </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#preprocessing title=Preprocessing class=md-nav__link> Preprocessing </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#format-data-for-plotting title="Format data for plotting" class=md-nav__link> Format data for plotting </a> </li> <li class=md-nav__item> <a href=#create-table-for-plotting-annotations title="Create table for plotting annotations" class=md-nav__link> Create table for plotting annotations </a> </li> <li class=md-nav__item> <a href=#labelscolors-for-plot title="Labels/colors for plot" class=md-nav__link> Labels/colors for plot </a> </li> <li class=md-nav__item> <a href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Plot probabilities and predictions data frame simultaneously" class=md-nav__link> Plot probabilities and predictions data frame simultaneously </a> </li> <li class=md-nav__item> <a href=#add-annotations-for-mutation-rate-purity-and-age title="Add annotations for mutation rate, purity, and age" class=md-nav__link> Add annotations for mutation rate, purity, and age </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-4 title="Fig 4" class=md-nav__link> Fig 4 </a> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content> <article class="md-content__inner md-typeset"> <!-- enable git edit link only if config.repo_edit is set to true --> <h1>F4 &middot; DNA Methylation based Prediction of Canine Gliomas</h1> <div class="admonition cite"> <p class=admonition-title>Contributor(s)</p> <p>Kevin J. Anderson</p> </div> <blockquote> <p>DNA methylation based classification of canine glioma.</p> </blockquote> <h3 id=preprocessing>Preprocessing<a class=headerlink href=#preprocessing title="Permanent link">&para;</a></h3> <h4 id=format-data-for-plotting>Format data for plotting<a class=headerlink href=#format-data-for-plotting title="Permanent link">&para;</a></h4> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=nf>library</span><span class=p>(</span><span class=n>ggplot2</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>egg</span><span class=p>)</span>

<span class=c1># Cluster probabilities to determine sample plot order</span>
<span class=n>cgp_RRBS.noNormals_ord</span> <span class=o>&lt;-</span> <span class=nf>hclust</span><span class=p>(</span> <span class=nf>dist</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>,</span> <span class=n>method</span> <span class=o>=</span> <span class=s>&quot;euclidean&quot;</span><span class=p>),</span> <span class=n>method</span> <span class=o>=</span> <span class=s>&quot;ward.D&quot;</span> <span class=p>)</span><span class=o>$</span><span class=n>order</span>

<span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span> <span class=o>&lt;-</span> <span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities[cgp_RRBS.noNormals_ord</span><span class=p>,</span><span class=n>]</span>
<span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>predictions</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>predictions[cgp_RRBS.noNormals_ord]</span><span class=p>,</span> <span class=n>levels</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;Adult-IDH_mut&quot;</span><span class=p>,</span><span class=s>&quot;Adult-IDH_wt&quot;</span><span class=p>,</span><span class=s>&quot;Pediatric-Tumor&quot;</span><span class=p>))</span>

<span class=c1># Reformat probabilities table for plotting</span>
<span class=n>cgp_RRBS.noNormals_class_probabilities</span> <span class=o>&lt;-</span> <span class=n>reshape2</span><span class=o>::</span><span class=nf>melt</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>)</span>
<span class=nf>colnames</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_probabilities</span><span class=p>)</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;Sample&quot;</span><span class=p>,</span><span class=s>&quot;Predicted_Class&quot;</span><span class=p>,</span><span class=s>&quot;Prediction_Probability&quot;</span><span class=p>)</span>

<span class=n>cgp_RRBS.noNormals_class_probabilities</span><span class=o>$</span><span class=n>Sample</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_probabilities</span><span class=o>$</span><span class=n>Sample</span><span class=p>,</span> <span class=n>levels</span> <span class=o>=</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>)</span><span class=n>[cgp_RRBS.noNormals_ord]</span><span class=p>)</span>


<span class=c1># Create shorthand barcode for sample text on x-axis (on mutationsFig) consisting of 4 character caseID only</span>
<span class=n>SampleLabel</span> <span class=o>&lt;-</span> <span class=nf>unlist</span><span class=p>(</span><span class=nf>lapply</span><span class=p>(</span><span class=nf>strsplit</span><span class=p>(</span><span class=nf>rownames</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>)</span><span class=n>[cgp_RRBS.noNormals_ord]</span><span class=p>,</span><span class=s>&quot;-&quot;</span><span class=p>),</span> <span class=n>`[`</span><span class=p>,</span> <span class=m>3</span><span class=p>))</span>
<span class=n>SampleLabel</span> <span class=o>&lt;-</span> <span class=nf>paste0</span><span class=p>(</span><span class=s>&quot;i_&quot;</span><span class=p>,</span><span class=n>SampleLabel</span><span class=p>)</span>

<span class=c1># Create label for sample text on x-axis (on mutationsFig) to color IDHmut samples</span>
<span class=n>SampleLabelStyle</span> <span class=o>&lt;-</span> <span class=nf>rep</span><span class=p>(</span><span class=s>&quot;plain&quot;</span><span class=p>,</span><span class=nf>length</span><span class=p>(</span><span class=n>SampleLabel</span><span class=p>))</span>
<span class=n>IDHmutsamples</span> <span class=o>&lt;-</span> <span class=nf>which</span><span class=p>(</span><span class=nf>rownames</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>)</span><span class=n>[cgp_RRBS.noNormals_ord]</span> <span class=o>%in%</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;CGP-S03-E7AB-8AD9ACF7-T1-A2-J05&quot;</span><span class=p>,</span><span class=s>&quot;CGP-S03-1165-177480CC-T1-A2-J05&quot;</span><span class=p>,</span><span class=s>&quot;CGP-S04-1166-3C6CFE5F-T1-A1-J05&quot;</span><span class=p>))</span>
<span class=n>SampleLabelStyle[IDHmutsamples]</span> <span class=o>&lt;-</span> <span class=s>&quot;bold&quot;</span>
</pre></div> </td></tr></table> <h4 id=create-table-for-plotting-annotations>Create table for plotting annotations<a class=headerlink href=#create-table-for-plotting-annotations title="Permanent link">&para;</a></h4> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5
6</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=n>cgp_RRBS.noNormals_class_predictions</span> <span class=o>&lt;-</span> <span class=nf>data.frame</span><span class=p>(</span><span class=n>Sample</span><span class=o>=</span><span class=nf>rownames</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>probabilities</span><span class=p>),</span>
                                                <span class=n>Class</span><span class=o>=</span><span class=n>cgp_RRBS.noNormals_class_prediction</span><span class=o>$</span><span class=n>predictions</span><span class=p>,</span>
                                                <span class=n>Location</span><span class=o>=</span><span class=n>cgp_RRBS.noNormals_sample_info</span><span class=o>$</span><span class=n>Tumor_Location</span><span class=p>,</span>
                                                <span class=n>Tumor_Grade</span><span class=o>=</span><span class=n>cgp_RRBS.noNormals_sample_info</span><span class=o>$</span><span class=n>Tumor_Grade</span><span class=p>,</span>
                                                <span class=n>Histology</span><span class=o>=</span><span class=n>cgp_RRBS.noNormals_sample_info</span><span class=o>$</span><span class=n>Tumor_Morphology</span><span class=p>)</span>
<span class=n>cgp_RRBS.noNormals_class_predictions</span><span class=o>$</span><span class=n>Location</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_predictions</span><span class=o>$</span><span class=n>Location</span><span class=p>,</span> <span class=n>levels</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;Hemispheric&quot;</span><span class=p>,</span><span class=s>&quot;Cerebellar&quot;</span><span class=p>,</span><span class=s>&quot;Midline&quot;</span><span class=p>))</span>
</pre></div> </td></tr></table> <h4 id=labelscolors-for-plot>Labels/colors for plot<a class=headerlink href=#labelscolors-for-plot title="Permanent link">&para;</a></h4> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5
6
7
8
9</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=n>class_probabilities_labels</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;Glioma&quot;</span><span class=p>,</span><span class=s>&quot;Normal&quot;</span><span class=p>,</span><span class=s>&quot;IDHmut&quot;</span><span class=p>,</span><span class=s>&quot;IDHmut\n1p/19q codel&quot;</span><span class=p>,</span><span class=s>&quot;IDHwt&quot;</span><span class=p>,</span><span class=s>&quot;Normal&quot;</span><span class=p>)</span>
<span class=n>class_probabilities_limits</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;Pediatric-Tumor&quot;</span><span class=p>,</span><span class=s>&quot;Pediatric-Normal&quot;</span><span class=p>,</span><span class=s>&quot;Adult-IDH_mut&quot;</span><span class=p>,</span><span class=s>&quot;Adult-IDH_mut,1p/19q_codel&quot;</span><span class=p>,</span><span class=s>&quot;Adult-IDH_wt&quot;</span><span class=p>,</span><span class=s>&quot;Adult-Normal&quot;</span><span class=p>)</span>
<span class=n>class_probabilities_colors</span> <span class=o>&lt;-</span> <span class=nf>c</span><span class=p>(</span><span class=nf>rep</span><span class=p>(</span><span class=s>&quot;#fc8d62&quot;</span><span class=p>,</span><span class=m>2</span><span class=p>),</span><span class=nf>rep</span><span class=p>(</span><span class=s>&quot;#8da0cb&quot;</span><span class=p>,</span><span class=m>4</span><span class=p>))</span>

<span class=c1># NOTE: </span>
<span class=c1># Label spacing for centering &quot;Pediatric&quot; over class categories:</span>
<span class=c1>## labs(y = &quot;Predicted Class\nAdult                                                Pediatric&quot;)</span>
<span class=c1># Label spacing for centering &quot;Adult&quot; over class categories</span>
<span class=c1>## labs(y = &quot;Predicted Class\nAdult          Pediatric&quot;)</span>
</pre></div> </td></tr></table> <h4 id=plot-probabilities-and-predictions-data-frame-simultaneously>Plot probabilities and predictions data frame simultaneously<a class=headerlink href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Permanent link">&para;</a></h4> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</pre></div></td><td class=code><div class=codehilite><pre><span></span>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted &lt;- ggplot() +
  geom_tile(data = cgp_RRBS.noNormals_class_probabilities, mapping = aes(x = Sample, y = Predicted_Class, fill = Prediction_Probability), colour = &quot;grey50&quot;) +
  geom_point(data = cgp_RRBS.noNormals_class_predictions, mapping = aes(x = Sample, y = Class, shape = Location, size = Tumor_Grade, colour = Histology)) +
  scale_colour_manual(name=&#39;Histology&#39;, values=c(&#39;Astro&#39;=&#39;#d95f02&#39;, &#39;Oligo&#39;=&#39;#7570b3&#39;, &#39;Undefined&#39;=&#39;#1b9e77&#39;), labels = c(&#39;Astro&#39;,&#39;Oligo&#39;,&#39;Undefined&#39;)) +
  scale_size_manual(name=&#39;Tumor Grade&#39;, values=c(&#39;High&#39;=8, &#39;Low&#39;=4), labels = c(&#39;High&#39;,&#39;Low&#39;)) +
  guides(colour = guide_legend(&quot;Histology&quot;,override.aes=list(size = 4))) +
  guides(shape = guide_legend(&quot;Location&quot;,override.aes=list(size = 4))) +
  scale_fill_gradient(name = &quot;Prediction\nProbability&quot;,
                      low = &quot;white&quot;,
                      high = &quot;deepskyblue3&quot;,
                      limits = c(0,1)) +
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle=90, face=SampleLabelStyle),
        axis.text.y = element_text(color = rev(class_probabilities_colors)),
        axis.title = element_text(size = 12, face = &quot;bold&quot;),
        axis.title.y = element_text(color = &quot;black&quot;),
        legend.title = element_text(size = 10, face = &quot;bold&quot;),
        legend.text = element_text(size = 8),
        plot.margin = margin(1, 1, 0, 1, &quot;cm&quot;)) +
  scale_x_discrete(labels=SampleLabel) +
  scale_y_discrete(limits = rev(class_probabilities_limits), 
    labels = rev(class_probabilities_labels)) +
    labs(x = &quot;Sample&quot;, y = &quot;Predicted Class\nAdult&quot;)
</pre></div> </td></tr></table> <h4 id=add-annotations-for-mutation-rate-purity-and-age>Add annotations for mutation rate, purity, and age<a class=headerlink href=#add-annotations-for-mutation-rate-purity-and-age title="Permanent link">&para;</a></h4> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=c1># Age annotation</span>
<span class=n>ageFig</span> <span class=o>&lt;-</span> <span class=nf>ggplot</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_sample_info</span><span class=p>)</span> <span class=o>+</span> <span class=nf>geom_point</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=n>Tumor_Sample_Barcode</span><span class=p>,</span> <span class=n>y</span> <span class=o>=</span> <span class=n>age_updated</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>geom_hline</span><span class=p>(</span><span class=n>yintercept</span> <span class=o>=</span> <span class=m>2</span><span class=p>,</span> <span class=n>linetype</span><span class=o>=</span><span class=s>&quot;dashed&quot;</span><span class=p>,</span> <span class=n>color</span> <span class=o>=</span> <span class=s>&quot;dimgray&quot;</span><span class=p>,</span> <span class=n>cex</span> <span class=o>=</span> <span class=m>1</span><span class=p>)</span> <span class=o>+</span> 
  <span class=nf>labs</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=s>&quot;Sample&quot;</span><span class=p>)</span> <span class=o>+</span>
  <span class=nf>theme</span><span class=p>(</span><span class=n>axis.text</span> <span class=o>=</span> <span class=nf>element_text</span><span class=p>(</span><span class=n>size</span> <span class=o>=</span> <span class=m>10</span><span class=p>),</span>
        <span class=n>axis.text.x</span> <span class=o>=</span> <span class=nf>element_text</span><span class=p>(</span><span class=n>angle</span><span class=o>=</span><span class=m>90</span><span class=p>,</span> <span class=n>face</span><span class=o>=</span><span class=n>SampleLabelStyle</span><span class=p>),</span>
        <span class=n>axis.title</span> <span class=o>=</span> <span class=nf>element_text</span><span class=p>(</span><span class=n>size</span> <span class=o>=</span> <span class=m>12</span><span class=p>,</span> <span class=n>face</span> <span class=o>=</span> <span class=s>&quot;bold&quot;</span><span class=p>),</span>
        <span class=n>plot.margin</span> <span class=o>=</span> <span class=nf>margin</span><span class=p>(</span><span class=m>0</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=s>&quot;cm&quot;</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>scale_x_discrete</span><span class=p>(</span><span class=n>labels</span><span class=o>=</span><span class=n>SampleLabel</span><span class=p>)</span> <span class=o>+</span>
  <span class=nf>scale_y_continuous</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;Age&quot;</span><span class=p>,</span> <span class=n>breaks</span><span class=o>=</span><span class=nf>seq</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>12</span><span class=p>,</span><span class=m>2</span><span class=p>),</span> <span class=n>limits</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>13</span><span class=p>))</span>

<span class=c1># Purity annotation</span>
<span class=n>purityFig</span> <span class=o>&lt;-</span> <span class=nf>ggplot</span><span class=p>(</span><span class=n>data</span> <span class=o>=</span> <span class=n>cgp_RRBS.noNormals_sample_info</span><span class=p>,</span> <span class=n>mapping</span> <span class=o>=</span> <span class=nf>aes</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=n>Tumor_Sample_Barcode</span><span class=p>,</span> <span class=n>y</span> <span class=o>=</span> <span class=m>1</span><span class=p>,</span> <span class=n>fill</span> <span class=o>=</span> <span class=n>purity</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>geom_tile</span><span class=p>()</span> <span class=o>+</span> 
  <span class=nf>theme</span><span class=p>(</span><span class=n>axis.title.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.text.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.ticks.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.title.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.text.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.ticks.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>plot.margin</span> <span class=o>=</span> <span class=nf>margin</span><span class=p>(</span><span class=m>0</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=m>0</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=s>&quot;cm&quot;</span><span class=p>))</span> <span class=o>+</span>
  <span class=nf>scale_fill_gradient</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;Purity&quot;</span><span class=p>,</span>
                      <span class=n>low</span> <span class=o>=</span> <span class=s>&quot;white&quot;</span><span class=p>,</span>
                      <span class=n>high</span> <span class=o>=</span> <span class=s>&quot;seagreen&quot;</span><span class=p>,</span>
                      <span class=n>limits</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>1</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>theme</span><span class=p>(</span><span class=n>legend.title</span><span class=o>=</span><span class=nf>element_text</span><span class=p>(</span><span class=n>size</span><span class=o>=</span><span class=m>9</span><span class=p>,</span> <span class=n>face</span> <span class=o>=</span> <span class=s>&quot;bold&quot;</span><span class=p>),</span><span class=n>legend.text</span> <span class=o>=</span> <span class=nf>element_text</span><span class=p>(</span><span class=n>size</span><span class=o>=</span><span class=m>8</span><span class=p>),</span><span class=n>legend.key.height</span> <span class=o>=</span> <span class=nf>unit</span><span class=p>(</span><span class=m>0.08</span><span class=p>,</span> <span class=s>&quot;in&quot;</span><span class=p>),</span><span class=n>legend.direction</span> <span class=o>=</span> <span class=s>&quot;horizontal&quot;</span><span class=p>)</span> <span class=o>+</span>
  <span class=nf>guides</span><span class=p>(</span><span class=n>fill</span> <span class=o>=</span> <span class=nf>guide_colourbar</span><span class=p>(</span><span class=n>title.position</span><span class=o>=</span><span class=s>&quot;top&quot;</span><span class=p>,</span> <span class=n>title.hjust</span> <span class=o>=</span> <span class=m>0.5</span><span class=p>))</span>

<span class=c1># Mutation annotation</span>
<span class=n>mutationsFig</span> <span class=o>&lt;-</span> <span class=nf>ggplot</span><span class=p>(</span><span class=n>data</span> <span class=o>=</span> <span class=n>cgp_RRBS.noNormals_sample_info</span><span class=p>,</span> <span class=n>mapping</span> <span class=o>=</span> <span class=nf>aes</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=n>Tumor_Sample_Barcode</span><span class=p>,</span> <span class=n>y</span> <span class=o>=</span> <span class=m>1</span><span class=p>,</span> <span class=n>fill</span> <span class=o>=</span> <span class=n>adusted_coding_mutrate</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>geom_tile</span><span class=p>()</span> <span class=o>+</span> 
  <span class=c1># scale_x_discrete(labels=SampleLabel) +</span>
  <span class=nf>theme</span><span class=p>(</span><span class=n>axis.title.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.text.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.ticks.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.title.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.text.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>axis.ticks.y</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
        <span class=n>plot.margin</span> <span class=o>=</span> <span class=nf>margin</span><span class=p>(</span><span class=m>0</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=m>0</span><span class=p>,</span> <span class=m>1</span><span class=p>,</span> <span class=s>&quot;cm&quot;</span><span class=p>))</span> <span class=o>+</span>
  <span class=nf>scale_fill_gradient</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;Mutations/Mb&quot;</span><span class=p>,</span>
                      <span class=n>low</span> <span class=o>=</span> <span class=s>&quot;white&quot;</span><span class=p>,</span>
                      <span class=n>high</span> <span class=o>=</span> <span class=s>&quot;red&quot;</span><span class=p>,</span>
                      <span class=n>limits</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>2.6</span><span class=p>))</span> <span class=o>+</span> 
  <span class=nf>theme</span><span class=p>(</span><span class=n>legend.title</span><span class=o>=</span><span class=nf>element_text</span><span class=p>(</span><span class=n>size</span><span class=o>=</span><span class=m>9</span><span class=p>,</span> <span class=n>face</span> <span class=o>=</span> <span class=s>&quot;bold&quot;</span><span class=p>),</span><span class=n>legend.text</span> <span class=o>=</span> <span class=nf>element_text</span><span class=p>(</span><span class=n>size</span><span class=o>=</span><span class=m>8</span><span class=p>),</span><span class=n>legend.key.height</span> <span class=o>=</span> <span class=nf>unit</span><span class=p>(</span><span class=m>0.08</span><span class=p>,</span> <span class=s>&quot;in&quot;</span><span class=p>),</span><span class=n>legend.direction</span> <span class=o>=</span> <span class=s>&quot;horizontal&quot;</span><span class=p>)</span> <span class=o>+</span>
  <span class=nf>guides</span><span class=p>(</span><span class=n>fill</span> <span class=o>=</span> <span class=nf>guide_colourbar</span><span class=p>(</span><span class=n>title.position</span><span class=o>=</span><span class=s>&quot;top&quot;</span><span class=p>,</span> <span class=n>title.hjust</span> <span class=o>=</span> <span class=m>0.5</span><span class=p>))</span>
</pre></div> </td></tr></table> <h3 id=fig-4>Fig 4<a class=headerlink href=#fig-4 title="Permanent link">&para;</a></h3> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=c1>## Combine plots</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggpubr</span><span class=p>)</span>

<span class=n>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.combinedVer</span> <span class=o>&lt;-</span> <span class=n>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted</span> <span class=o>+</span>
<span class=nf>theme</span><span class=p>(</span><span class=n>axis.title.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
 <span class=n>axis.text.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>(),</span>
 <span class=n>axis.ticks.x</span><span class=o>=</span><span class=nf>element_blank</span><span class=p>())</span>

<span class=n>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.w_purity_mutations_age</span> <span class=o>&lt;-</span> <span class=nf>ggarrange</span><span class=p>(</span><span class=n>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.combinedVer</span><span class=p>,</span> <span class=n>purityFig</span><span class=p>,</span> <span class=n>mutationsFig</span><span class=p>,</span> <span class=n>ageFig</span><span class=p>,</span><span class=n>nrow</span> <span class=o>=</span> <span class=m>4</span><span class=p>,</span><span class=n>ncol</span> <span class=o>=</span> <span class=m>1</span><span class=p>,</span> <span class=n>heights</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>3</span><span class=p>,</span><span class=m>0.25</span><span class=p>,</span><span class=m>0.25</span><span class=p>,</span><span class=m>0.5</span><span class=p>),</span> <span class=n>padding</span> <span class=o>=</span> <span class=nf>unit</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=s>&quot;line&quot;</span><span class=p>))</span>


<span class=nf>pdf</span><span class=p>(</span><span class=s>&quot;F4.pdf&quot;</span><span class=p>,</span>
    <span class=n>width</span> <span class=o>=</span> <span class=m>18</span><span class=p>,</span> <span class=n>height</span> <span class=o>=</span> <span class=m>12</span><span class=p>,</span> <span class=n>pointsize</span> <span class=o>=</span> <span class=m>12</span><span class=p>,</span>
    <span class=n>bg</span> <span class=o>=</span> <span class=s>&quot;white&quot;</span><span class=p>,</span>
    <span class=n>compress</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span> <span class=n>useDingbats</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>

<span class=n>cgp_RRBS.noNormals_class_probabilitiesFig.reformatted.w_purity_mutations_age</span>

<span class=nf>dev.off</span><span class=p>()</span>
</pre></div> </td></tr></table> <p>Last updated: February 8, 2020</p> <h2 id=__comments>Comments</h2> <div class="admonition bug"> <p class=admonition-title>Issues with code ?</p> <p>Please limit comments below for general questions. For code-related query, prefer <a href=https://github.com/TheJacksonLaboratory/canineglioma/issues>submitting a GitHub issue</a>.</p> </div> <div id=disqus_thread></div> <script>
    var disqus_config = function () {
      this.page.url = "https://canineglioma.verhaaklab.com/figures/F4/";
      this.page.identifier =
        "/figures/F4/";
    };
    (function() {
      var d = document, s = d.createElement("script");
      s.src = "//verhaaklab.disqus.com/embed.js";
      s.setAttribute("data-timestamp", +new Date());
      (d.head || d.body).appendChild(s);
    })();
  </script> </article> </div> </div> </main> <footer class=md-footer> <div class=md-footer-nav> <nav class="md-footer-nav__inner md-grid"> <a href=../S3/ title="S3 · Suppl Figures" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel=prev> <div class="md-flex__cell md-flex__cell--shrink"> <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i> </div> <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"> <span class=md-flex__ellipsis> <span class=md-footer-nav__direction> Previous </span> S3 &middot; Suppl Figures </span> </div> </a> <a href=../S4/ title="S4 · Suppl Figure 4" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel=next> <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"> <span class=md-flex__ellipsis> <span class=md-footer-nav__direction> Next </span> S4 &middot; Suppl Figure 4 </span> </div> <div class="md-flex__cell md-flex__cell--shrink"> <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i> </div> </a> </nav> </div> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-footer-copyright> <div class=md-footer-copyright__highlight> Copyright &copy; 2017 - 2020 <a href=https://verhaaklab.com>Verhaak Lab at the Jackson Laboratory for Genomic Medicine</a> </div> powered by <a href=https://www.mkdocs.org>MkDocs</a> and <a href=https://squidfunk.github.io/mkdocs-material/ > Material for MkDocs</a> </div> <div class=md-footer-social> <link rel=stylesheet href=../../assets/fonts/font-awesome.css> <a href=https://verhaaklab.com/contact class="md-footer-social__link fa fa-envelope"></a> <a href=https://twitter.com/roelverhaak class="md-footer-social__link fa fa-twitter"></a> <a href=https://github.com/TheJacksonLaboratory/canineglioma class="md-footer-social__link fa fa-github"></a> <a href=https://twitter.com/sbamin class="md-footer-social__link fa fa-twitter"></a> </div> </div> </div> </footer> </div> <script src=../../assets/javascripts/application.ffb616e8.js></script> <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script> <script src=../../assets/js/katex.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js></script> <script src=../../search/main.js></script> </body> </html>