



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Canine Glioma Project &middot; Summary and Extended Methods">
      
      
        <link rel="canonical" href="https://canineglioma.verhaaklab.com/figures/F3A/">
      
      
        <meta name="author" content="Samir B. Amin">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-\.]+">
      
      <link rel="shortcut icon" href="../../assets/images/logo.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>F3A &middot; Somatic Mutational Signatures - Deconvoluted</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.4031d38b.css">
      
        <link rel="stylesheet" href="../../assets/stylesheets/application-palette.224b79ff.css">
      
      
        
        
        <meta name="theme-color" content="#2196f3">
      
    
    
      <script src="../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,400,400i,700|Fira+Code&display=fallback">
        <style>body,input{font-family:"IBM Plex Sans","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Fira Code","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../assets/css/katex.css">
    
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css">
    

    
      
        
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-64053990-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-64053990-2');
</script>
      
    

    
  
      

      

      <meta property="og:url" content="https://canineglioma.verhaaklab.com/figures/F3A/">

      
        <meta property="og:title" content="F3A &middot; Somatic Mutational Signatures - Deconvoluted">
      

      <meta property="og:description" content="Canine Glioma Project &middot; Summary and Extended Methods">
      <meta property="og:image" content="../../assets/images/lab_logo_sq-md.jpg">
      <meta property="og:image:alt" content="Image alt">
      <meta property="og:image:type" content="image/jpeg">
      <meta property="og:image:width" content="300">
      <meta property="og:image:height" content="300">

      <meta name="twitter:card" content="summary">
      <meta name="twitter:site" content="@roelverhaak">
      <meta name="twitter:creator" content="@roelverhaak">

      
        <meta property="twitter:title" content="F3A &middot; Somatic Mutational Signatures - Deconvoluted">
      

      <meta name="twitter:description" content="Canine Glioma Project &middot; Summary and Extended Methods">

      <meta name="twitter:image" content="../../assets/images/lab_logo_sq-md.jpg">
      <meta name="twitter:image:alt" content="Image alt">


  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="blue" data-md-color-accent="blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#required-metadata" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://canineglioma.verhaaklab.com" title="Canine Glioma Project &middot; Verhaak Lab" class="md-header-nav__button md-logo">
          
            <img src="../../assets/images/logo.png" width="24" height="24">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Canine Glioma Project &middot; Verhaak Lab
            </span>
            <span class="md-header-nav__topic">
              
                F3A &middot; Somatic Mutational Signatures - Deconvoluted
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://canineglioma.verhaaklab.com" title="Canine Glioma Project &middot; Verhaak Lab" class="md-nav__button md-logo">
      
        <img src="../../assets/images/logo.png" width="48" height="48">
      
    </a>
    Canine Glioma Project &middot; Verhaak Lab
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Methods
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Methods
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S01_samples/" title="Samples" class="md-nav__link">
      Samples
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S02_data_sources/" title="Data Sources" class="md-nav__link">
      Data Sources
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S03_sample_prep/" title="Sample Preparations" class="md-nav__link">
      Sample Preparations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S04_alignments_fprint/" title="Alignments" class="md-nav__link">
      Alignments
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S05_somatic_variants/" title="Somatic SNVs and Indels" class="md-nav__link">
      Somatic SNVs and Indels
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S06_smgs/" title="Significantly Mutated Genes" class="md-nav__link">
      Significantly Mutated Genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S07_hallmarks/" title="Cancer Hallmarks" class="md-nav__link">
      Cancer Hallmarks
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S09_mutation_rate/" title="Somatic Mutation Rate" class="md-nav__link">
      Somatic Mutation Rate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S10_SCNA/" title="Somatic Copy Number Aberrations" class="md-nav__link">
      Somatic Copy Number Aberrations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S11_aneuploidy/" title="Aneuploidy Metrics" class="md-nav__link">
      Aneuploidy Metrics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S12_ith/" title="Intratumor Heterogeneity" class="md-nav__link">
      Intratumor Heterogeneity
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S13_mut_sigs/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S14_mut_timing/" title="Molecular Timing of Somatic Drivers" class="md-nav__link">
      Molecular Timing of Somatic Drivers
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S15_class_predict_methy/" title="Class Prediction using methylation classifier" class="md-nav__link">
      Class Prediction using methylation classifier
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../methods/S16_expression_immune/" title="Immune Microenvironment Analysis" class="md-nav__link">
      Immune Microenvironment Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Figures
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Figures
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../A1_preload/" title="Preload" class="md-nav__link">
      Preload
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2">
    
    <label class="md-nav__link" for="nav-3-2">
      F1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-2">
        F1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../F1A/" title="F1A · Somatic Mutational Landscape" class="md-nav__link">
      F1A &middot; Somatic Mutational Landscape
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F1B/" title="F1B · Somatic Recurrent Hotspot Mutations" class="md-nav__link">
      F1B &middot; Somatic Recurrent Hotspot Mutations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F1C/" title="F1C · Comparative Pathway Analysis" class="md-nav__link">
      F1C &middot; Comparative Pathway Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F1D/" title="F1D · Comparative Somatic Mutational Burden" class="md-nav__link">
      F1D &middot; Comparative Somatic Mutational Burden
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../S1/" title="S1A · DNA fingerprinting" class="md-nav__link">
      S1A &middot; DNA fingerprinting
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-3" type="checkbox" id="nav-3-3">
    
    <label class="md-nav__link" for="nav-3-3">
      F2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-3">
        F2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../F2A/" title="F2A · Somatic Copy Number Landscape" class="md-nav__link">
      F2A &middot; Somatic Copy Number Landscape
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F2B/" title="F2B · Comparative Aneuploidy Burden" class="md-nav__link">
      F2B &middot; Comparative Aneuploidy Burden
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F2C/" title="F2C · Comparative Shared Syntenic Aneuploidy" class="md-nav__link">
      F2C &middot; Comparative Shared Syntenic Aneuploidy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F2D/" title="F2D · Comparative Intratumor Heterogeneity" class="md-nav__link">
      F2D &middot; Comparative Intratumor Heterogeneity
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../S2/" title="S2 · IGV Plots" class="md-nav__link">
      S2 &middot; IGV Plots
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-4" type="checkbox" id="nav-3-4" checked>
    
    <label class="md-nav__link" for="nav-3-4">
      F3
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-4">
        F3
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        F3A &middot; Somatic Mutational Signatures - Deconvoluted
      </label>
    
    <a href="./" title="F3A · Somatic Mutational Signatures - Deconvoluted" class="md-nav__link md-nav__link--active">
      F3A &middot; Somatic Mutational Signatures - Deconvoluted
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#required-metadata" title="Required metadata" class="md-nav__link">
    Required metadata
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fig-3a" title="Fig 3A" class="md-nav__link">
    Fig 3A
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#suppl-fig-3a" title="Suppl Fig 3A" class="md-nav__link">
    Suppl Fig 3A
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#suppl-fig-3b" title="Suppl Fig 3B" class="md-nav__link">
    Suppl Fig 3B
  </a>
  
</li>
      
      
      
      
      
        <li class="md-nav__item">
          <a href="#__comments" title="Comments" class="md-nav__link md-nav__link--active">
            Comments
          </a>
        </li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F3B/" title="F3B · Somatic Mutational Landscape - de novo" class="md-nav__link">
      F3B &middot; Somatic Mutational Landscape - de novo
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F3C/" title="F3C · Comparative Somatic Mutational Signatures in Driver Genes" class="md-nav__link">
      F3C &middot; Comparative Somatic Mutational Signatures in Driver Genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../F3D/" title="F3D · Comparative Molecular Timing of Driver Events" class="md-nav__link">
      F3D &middot; Comparative Molecular Timing of Driver Events
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../S3/" title="S3 · Suppl Figure 3" class="md-nav__link">
      S3 &middot; Suppl Figure 3
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-5" type="checkbox" id="nav-3-5">
    
    <label class="md-nav__link" for="nav-3-5">
      F4
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-5">
        F4
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../F4/" title="F4 · DNA Methylation based Prediction of Canine Gliomas" class="md-nav__link">
      F4 &middot; DNA Methylation based Prediction of Canine Gliomas
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../S4/" title="S4 · Suppl Figure 4" class="md-nav__link">
      S4 &middot; Suppl Figure 4
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-6" type="checkbox" id="nav-3-6">
    
    <label class="md-nav__link" for="nav-3-6">
      F5
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-6">
        F5
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../F5/" title="F5B · Comparative Immunohistochemistry - Quantitative" class="md-nav__link">
      F5B &middot; Comparative Immunohistochemistry - Quantitative
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../S5/" title="S5 · Suppl Figure 5" class="md-nav__link">
      S5 &middot; Suppl Figure 5
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="https://verhaaklab.com/contact" title="Contact" class="md-nav__link">
      Contact
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#required-metadata" title="Required metadata" class="md-nav__link">
    Required metadata
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fig-3a" title="Fig 3A" class="md-nav__link">
    Fig 3A
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#suppl-fig-3a" title="Suppl Fig 3A" class="md-nav__link">
    Suppl Fig 3A
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#suppl-fig-3b" title="Suppl Fig 3B" class="md-nav__link">
    Suppl Fig 3B
  </a>
  
</li>
      
      
      
      
      
        <li class="md-nav__item">
          <a href="#__comments" title="Comments" class="md-nav__link md-nav__link--active">
            Comments
          </a>
        </li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>F3A &middot; Somatic Mutational Signatures - Deconvoluted</h1>
                
                <h3 id="required-metadata">Required metadata<a class="headerlink" href="#required-metadata" title="Permanent link">&para;</a></h3>
<ul>
<li>Canine glioma patients with <a href="/methods/S13_mut_sigs/">outlier mutation profile</a>.</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="c1">## filter out missing values in mutation rate column</span>
<span class="n">cgp_sample_info_maftools_hypermuts</span> <span class="o">&lt;-</span> <span class="n">cgp_suppl_sample_info_master</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">adusted_coding_mutrate</span><span class="p">))</span> <span class="o">%&gt;%</span>
  <span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">adusted_coding_mutrate</span><span class="p">))</span>

<span class="n">outlier_model_hypermut</span> <span class="o">&lt;-</span> <span class="nf">lm</span><span class="p">(</span><span class="n">adusted_coding_mutrate</span> <span class="o">~</span> <span class="n">Matched_Normal</span> <span class="o">+</span>
    <span class="n">Tissue_Archival</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">cgp_sample_info_maftools_hypermuts</span><span class="p">)</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">outlier_model_hypermut</span><span class="p">)</span>

<span class="c1">## Hypermut outliers after correcting for archival and paired vs tonly mode</span>
<span class="n">hypermut_true_outliers</span> <span class="o">&lt;-</span> <span class="n">car</span><span class="o">::</span><span class="nf">outlierTest</span><span class="p">(</span><span class="n">outlier_model_hypermut</span><span class="p">,</span>
    <span class="n">cutoff</span> <span class="o">=</span> <span class="m">0.05</span><span class="p">,</span>
    <span class="n">labels</span> <span class="o">=</span> <span class="n">cgp_sample_info_maftools_hypermuts</span><span class="o">$</span><span class="n">Tumor_Sample_Barcode</span><span class="p">)</span>

<span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;hypermut case: %s&quot;</span><span class="p">,</span> <span class="nf">names</span><span class="p">(</span><span class="n">hypermut_true_outliers</span><span class="o">$</span><span class="n">rstudent</span><span class="p">))</span>

<span class="n">cgp_sample_info_non_hypermut</span> <span class="o">&lt;-</span> <span class="n">cgp_sample_info_maftools_hypermuts</span> <span class="o">%&gt;%</span> 
    <span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="n">Tumor_Sample_Barcode</span> <span class="o">%in%</span>
    <span class="nf">names</span><span class="p">(</span><span class="n">hypermut_true_outliers</span><span class="o">$</span><span class="n">rstudent</span><span class="p">))</span>

<span class="c1">#### outliers in non-hypermut cases</span>
<span class="c1">## cases with mut rate exceeding chi sq distribution by &gt; 95% </span>
<span class="n">outlier_in_non_hypermut_chisq_95</span> <span class="o">&lt;-</span> <span class="n">cgp_sample_info_non_hypermut[</span>
    <span class="n">outliers</span><span class="o">::</span><span class="nf">scores</span><span class="p">(</span><span class="n">cgp_sample_info_non_hypermut</span><span class="o">$</span><span class="n">adusted_coding_mutrate</span><span class="p">,</span>
    <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;chisq&quot;</span><span class="p">,</span> <span class="n">prob</span> <span class="o">=</span> <span class="m">0.95</span><span class="p">),</span><span class="n">]</span> <span class="o">%&gt;%</span>
    <span class="nf">pull</span><span class="p">(</span><span class="n">Tumor_Sample_Barcode</span><span class="p">)</span>

<span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;outlier among non_hypermut: %s&quot;</span><span class="p">,</span> <span class="n">outlier_in_non_hypermut_chisq_95</span><span class="p">)</span>

<span class="n">cgp_outlier_mutant_cases</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">hypermut_true_outliers</span><span class="o">$</span><span class="n">rstudent</span><span class="p">),</span>
    <span class="n">outlier_in_non_hypermut_chisq_95</span><span class="p">)</span>

<span class="n">cgp_outlier_mutant_cases</span>
</pre></div>
</td></tr></table>

<blockquote>
<p>"i_8228" "i_95FC" "i_BD58" "i_1C36" "i_502F" "i_D732" "i_6B06" "i_56B5"</p>
</blockquote>
<ul>
<li>Pick random 9/73 cases<sup id="fnref:1"><a class="footnote-ref" href="#fn:1">1</a></sup> without outlier mutation profile.</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="c1">## nonhm_sampled &lt;- sample(ncol(fit_res_nonhm$contribution),</span>
    <span class="nf">ncol</span><span class="p">(</span><span class="n">fit_res_hm</span><span class="o">$</span><span class="n">contribution</span><span class="p">),</span> <span class="n">replace</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="c1">## following cases were used in the right-side panel of fig 3A</span>
<span class="n">nonhm_sampled</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="m">28</span><span class="p">,</span><span class="m">48</span><span class="p">,</span><span class="m">30</span><span class="p">,</span><span class="m">57</span><span class="p">,</span><span class="m">54</span><span class="p">,</span><span class="m">40</span><span class="p">,</span><span class="m">37</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">62</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<ul>
<li>Deconvolute known human mutational signatures</li>
</ul>
<blockquote>
<p>Fitted mutational signatures, i.e., <code>fit_res</code>, <code>fit_res_hm</code> and <code>fit_res_nonhm</code> were derived using <code>fit_to_signatures</code> function from <em>MutationalPatterns</em> R package<sup id="fnref:2"><a class="footnote-ref" href="#fn:2">2</a></sup>.</p>
</blockquote>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>## mut_mat object is within /data/cgp_base_objects_v1.0.RData
## mut_mat &lt;- MutationalPattern::mut_matrix(vcf_list = vcfs,
    ref_genome = ref_genome)

fit_res &lt;- fit_to_signatures(mut_mat, cancer_signatures)

hypermutators &lt;- c(&quot;i_8228&quot;, &quot;i_95FC&quot;, &quot;i_BD58&quot;, &quot;i_1C36&quot;,
    &quot;i_502F&quot;, &quot;i_D732&quot;, &quot;i_6B06&quot;, &quot;i_56B5&quot;)

non_hypermutators &lt;- colnames(mut_mat)[!colnames(mut_mat) %in% hypermutators]

## non-hypermutators or one without outlier mutation profile
fit_res_nonhm &lt;- fit_to_signatures(mut_mat[, colnames(mut_mat) %in%
    non_hypermutators], cancer_signatures)

## hypermutators or one with outlier mutation profile
fit_res_hm &lt;- fit_to_signatures(mut_mat[, colnames(mut_mat) %in%
    hypermutators], cancer_signatures)
</pre></div>
</td></tr></table>

<ul>
<li>Merge signatures by proposed mechanism</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">hm_sigs</span> <span class="o">&lt;-</span> <span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">sig_enrichment</span> <span class="o">!=</span> <span class="s">&quot;NON_HM&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">pull</span><span class="p">(</span><span class="n">sig_id</span><span class="p">)</span>

<span class="n">nonhm_sigs</span> <span class="o">&lt;-</span> <span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">sig_enrichment</span> <span class="o">!=</span> <span class="s">&quot;HM&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">pull</span><span class="p">(</span><span class="n">sig_id</span><span class="p">)</span>

<span class="n">fit_res_hm_merged</span> <span class="o">&lt;-</span> <span class="nf">as_tibble</span><span class="p">(</span><span class="n">fit_res_hm</span><span class="o">$</span><span class="n">contribution[hm_sigs</span><span class="p">,</span> <span class="n">]</span><span class="p">,</span>
    <span class="n">rownames</span> <span class="o">=</span> <span class="s">&quot;sig_id&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">left_join</span><span class="p">(</span><span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_id</span><span class="p">,</span> <span class="n">sig_group</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">gather</span><span class="p">(</span><span class="n">patient_id</span><span class="p">,</span> <span class="n">sig_score</span><span class="p">,</span> <span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;i_&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">group_by</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">patient_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">mutate</span><span class="p">(</span><span class="n">group_score</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="n">sig_score</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">group_score</span><span class="p">,</span> <span class="n">patient_id</span><span class="p">,</span> <span class="n">sig_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">spread</span><span class="p">(</span><span class="n">patient_id</span><span class="p">,</span> <span class="n">group_score</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">ungroup</span><span class="p">()</span> <span class="o">%&gt;%</span>
    <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">sig_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">column_to_rownames</span><span class="p">(</span><span class="s">&quot;sig_group&quot;</span><span class="p">)</span>

<span class="n">fit_res_nonhm_merged</span> <span class="o">&lt;-</span> <span class="nf">as_tibble</span><span class="p">(</span><span class="n">fit_res_nonhm</span><span class="o">$</span><span class="n">contribution[nonhm_sigs</span><span class="p">,</span> <span class="n">]</span><span class="p">,</span>
    <span class="n">rownames</span> <span class="o">=</span> <span class="s">&quot;sig_id&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">left_join</span><span class="p">(</span><span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_id</span><span class="p">,</span> <span class="n">sig_group</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">gather</span><span class="p">(</span><span class="n">patient_id</span><span class="p">,</span> <span class="n">sig_score</span><span class="p">,</span> <span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;i_&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">group_by</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">patient_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">mutate</span><span class="p">(</span><span class="n">group_score</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="n">sig_score</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">group_score</span><span class="p">,</span> <span class="n">patient_id</span><span class="p">,</span> <span class="n">sig_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">spread</span><span class="p">(</span><span class="n">patient_id</span><span class="p">,</span> <span class="n">group_score</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">ungroup</span><span class="p">()</span> <span class="o">%&gt;%</span>
    <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">sig_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">column_to_rownames</span><span class="p">(</span><span class="s">&quot;sig_group&quot;</span><span class="p">)</span>

<span class="c1">## group cancer signatures context proportion for HM</span>

<span class="n">cancer_signatures_merged</span> <span class="o">&lt;-</span> <span class="nf">as_tibble</span><span class="p">(</span><span class="n">cancer_signatures</span><span class="p">,</span>
    <span class="n">rownames</span> <span class="o">=</span> <span class="s">&quot;context&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">gather</span><span class="p">(</span><span class="n">sig_id</span><span class="p">,</span> <span class="n">sig_score</span><span class="p">,</span> <span class="o">-</span><span class="n">context</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">left_join</span><span class="p">(</span><span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_id</span><span class="p">,</span> <span class="n">sig_group</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">sig_group</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">group_by</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">context</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">mutate</span><span class="p">(</span><span class="n">group_score</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="n">sig_score</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">group_score</span><span class="p">))</span> <span class="o">%&gt;%</span>
    <span class="nf">ungroup</span><span class="p">()</span> <span class="o">%&gt;%</span>
    <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">group_score</span><span class="p">,</span> <span class="n">context</span><span class="p">,</span> <span class="n">sig_id</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">context</span><span class="p">,</span> <span class="n">group_score</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">spread</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">group_score</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">context</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">column_to_rownames</span><span class="p">(</span><span class="s">&quot;context&quot;</span><span class="p">)</span>

<span class="n">hm_nonhm_sig_group_colours</span> <span class="o">&lt;-</span> <span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">pull</span><span class="p">(</span><span class="n">colour_hex</span><span class="p">)</span>

<span class="nf">names</span><span class="p">(</span><span class="n">hm_nonhm_sig_group_colours</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">hm_nonhm_topsigs_tbl</span> <span class="o">%&gt;%</span>
    <span class="nf">distinct</span><span class="p">(</span><span class="n">sig_group</span><span class="p">,</span> <span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">pull</span><span class="p">(</span><span class="n">sig_group</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<ul>
<li>Function to plot signature contribution</li>
</ul>
<blockquote>
<p>Rewrite of <code>plot_contribution</code> function from <em>MutationalPatterns</em> R package.</p>
</blockquote>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cgp_plot_contribution</span> <span class="o">&lt;-</span> <span class="nf">function </span><span class="p">(</span><span class="n">contribution</span><span class="p">,</span>
    <span class="n">signatures</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(),</span> <span class="n">coord_flip</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s">&quot;relative&quot;</span><span class="p">,</span> <span class="n">palette</span> <span class="o">=</span> <span class="nf">c</span><span class="p">())</span>
    <span class="p">{</span>
        <span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">mode</span> <span class="o">==</span> <span class="s">&quot;relative&quot;</span> <span class="o">|</span> <span class="n">mode</span> <span class="o">==</span> <span class="s">&quot;absolute&quot;</span><span class="p">))</span>
          <span class="nf">stop</span><span class="p">(</span><span class="s">&quot;mode parameter should be either &#39;relative&#39; or &#39;absolute&#39;&quot;</span><span class="p">)</span>

        <span class="nf">if </span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">index</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">))</span> <span class="p">{</span>
          <span class="n">contribution</span> <span class="o">=</span> <span class="n">contribution[</span><span class="p">,</span> <span class="n">index]</span>
        <span class="p">}</span>

        <span class="n">Sample</span> <span class="o">=</span> <span class="kc">NULL</span>
        <span class="n">Contribution</span> <span class="o">=</span> <span class="kc">NULL</span>
        <span class="n">Signature</span> <span class="o">=</span> <span class="kc">NULL</span>

        <span class="nf">if </span><span class="p">(</span><span class="n">mode</span> <span class="o">==</span> <span class="s">&quot;relative&quot;</span><span class="p">)</span> <span class="p">{</span>
            <span class="n">m_contribution</span> <span class="o">=</span> <span class="n">data.table</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">contribution</span><span class="p">,</span> <span class="s">&quot;sig_id&quot;</span><span class="p">)</span>
            <span class="nf">colnames</span><span class="p">(</span><span class="n">m_contribution</span><span class="p">)</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Signature&quot;</span><span class="p">,</span>
                <span class="s">&quot;Sample&quot;</span><span class="p">,</span> <span class="s">&quot;Contribution&quot;</span><span class="p">)</span>

            <span class="n">plot</span> <span class="o">=</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">m_contribution</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="n">Sample</span><span class="p">),</span>
              <span class="n">y</span> <span class="o">=</span> <span class="n">Contribution</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="n">Signature</span><span class="p">),</span> <span class="n">order</span> <span class="o">=</span> <span class="n">Sample</span><span class="p">))</span> <span class="o">+</span>
              <span class="nf">geom_bar</span><span class="p">(</span><span class="n">position</span> <span class="o">=</span> <span class="s">&quot;fill&quot;</span><span class="p">,</span> <span class="n">stat</span> <span class="o">=</span> <span class="s">&quot;identity&quot;</span><span class="p">,</span>
                <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)</span> <span class="o">+</span>
              <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&quot;Relative contribution&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span>
              <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid.minor.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
                <span class="n">panel.grid.major.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
              <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid.minor.y</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
                <span class="n">panel.grid.major.y</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span>
        <span class="p">}</span> <span class="n">else</span> <span class="p">{</span>
            <span class="nf">if </span><span class="p">(</span><span class="nf">missing</span><span class="p">(</span><span class="n">signatures</span><span class="p">))</span>
              <span class="nf">stop</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;For contribution plotting in mode &#39;absolute&#39;:&quot;</span><span class="p">,</span>
                  <span class="s">&quot;also provide signatures matrix&quot;</span><span class="p">))</span>

            <span class="n">total_signatures</span> <span class="o">=</span> <span class="nf">colSums</span><span class="p">(</span><span class="n">signatures</span><span class="p">)</span>
            <span class="n">abs_contribution</span> <span class="o">=</span> <span class="n">contribution[</span><span class="p">,</span><span class="m">-1</span><span class="n">]</span> <span class="o">*</span> <span class="n">total_signatures</span>
            <span class="nf">rownames</span><span class="p">(</span><span class="n">abs_contribution</span><span class="p">)</span> <span class="o">=</span> <span class="n">contribution</span> <span class="o">%&gt;%</span>
                <span class="nf">pull</span><span class="p">(</span><span class="n">sig_id</span><span class="p">)</span>

            <span class="n">m_contribution</span> <span class="o">=</span> <span class="n">data.table</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">abs_contribution</span><span class="p">,</span> <span class="s">&quot;sig_id&quot;</span><span class="p">)</span>
            <span class="nf">colnames</span><span class="p">(</span><span class="n">m_contribution</span><span class="p">)</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Signature&quot;</span><span class="p">,</span>
                <span class="s">&quot;Sample&quot;</span><span class="p">,</span> <span class="s">&quot;Contribution&quot;</span><span class="p">)</span>

            <span class="n">plot</span> <span class="o">=</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">m_contribution</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="n">Sample</span><span class="p">),</span>
              <span class="n">y</span> <span class="o">=</span> <span class="n">Contribution</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="n">Signature</span><span class="p">),</span> <span class="n">order</span> <span class="o">=</span> <span class="n">Sample</span><span class="p">))</span> <span class="o">+</span>
              <span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span> <span class="o">=</span> <span class="s">&quot;identity&quot;</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)</span> <span class="o">+</span>
              <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&quot;Absolute contribution \n (no. mutations)&quot;</span><span class="p">)</span> <span class="o">+</span>
              <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span>
              <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid.minor.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
                <span class="n">panel.grid.major.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
              <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid.minor.y</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
                <span class="n">panel.grid.major.y</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span>
        <span class="p">}</span>

        <span class="nf">if </span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">palette</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span> <span class="p">{</span>
            <span class="n">plot</span> <span class="o">=</span> <span class="n">plot</span> <span class="o">+</span>
                <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&quot;Signature&quot;</span><span class="p">,</span> <span class="n">values</span> <span class="o">=</span> <span class="n">palette</span><span class="p">)</span>
        <span class="p">}</span> <span class="n">else</span> <span class="p">{</span>
            <span class="n">plot</span> <span class="o">=</span> <span class="n">plot</span> <span class="o">+</span>
                <span class="nf">scale_fill_discrete</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&quot;Signature&quot;</span><span class="p">)</span>
        <span class="p">}</span>

        <span class="nf">if </span><span class="p">(</span><span class="n">coord_flip</span><span class="p">)</span> <span class="p">{</span>
            <span class="n">plot</span> <span class="o">=</span> <span class="n">plot</span> <span class="o">+</span>
                <span class="nf">coord_flip</span><span class="p">()</span> <span class="o">+</span>
                <span class="nf">xlim</span><span class="p">(</span><span class="nf">rev</span><span class="p">(</span><span class="nf">levels</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="n">m_contribution</span><span class="o">$</span><span class="n">Sample</span><span class="p">))))</span>
        <span class="p">}</span> <span class="n">else</span> <span class="p">{</span>
            <span class="n">plot</span> <span class="o">=</span> <span class="n">plot</span> <span class="o">+</span> <span class="nf">xlim</span><span class="p">(</span><span class="nf">levels</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="n">m_contribution</span><span class="o">$</span><span class="n">Sample</span><span class="p">)))</span>
        <span class="p">}</span>

        <span class="nf">return</span><span class="p">(</span><span class="n">plot</span><span class="p">)</span>
    <span class="p">}</span>
</pre></div>
</td></tr></table>

<h3 id="fig-3a">Fig 3A<a class="headerlink" href="#fig-3a" title="Permanent link">&para;</a></h3>
<blockquote>
<p>Plot deconvoluted signature contribution after grouping signatures based on proposed mechanism.</p>
</blockquote>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="c1">## plot canine glioma patients with outlier mutation profile</span>

<span class="n">plotsig_hm_rel_merged</span> <span class="o">&lt;-</span> <span class="nf">cgp_plot_contribution</span><span class="p">(</span><span class="nf">as_tibble</span><span class="p">(</span>
    <span class="n">fit_res_hm_merged</span><span class="p">,</span> <span class="n">rownames</span> <span class="o">=</span> <span class="s">&quot;sig_id&quot;</span><span class="p">),</span>
    <span class="n">cancer_signatures_merged</span><span class="p">,</span>
    <span class="n">coord_flip</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s">&quot;relative&quot;</span><span class="p">,</span> <span class="n">palette</span> <span class="o">=</span> <span class="n">hm_nonhm_sig_group_colours</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span> <span class="o">=</span> <span class="m">20</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">legend.position</span> <span class="o">=</span> <span class="s">&quot;bottom&quot;</span><span class="p">,</span>
        <span class="n">legend.text</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">14</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&quot;Canine Gliomas with outlier mutation profile\n(n=8)&quot;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span>
        <span class="n">title</span> <span class="o">=</span> <span class="s">&quot;Known signatures grouped by\ntheir proposed mechanism&quot;</span><span class="p">,</span>
    <span class="n">title.position</span> <span class="o">=</span> <span class="s">&quot;bottom&quot;</span><span class="p">,</span>
    <span class="n">label.theme</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">14</span><span class="p">),</span>
    <span class="n">keywidth</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span>
    <span class="n">keyheight</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span>
    <span class="n">default.unit</span><span class="o">=</span><span class="s">&quot;cm&quot;</span><span class="p">,</span> <span class="n">nrow</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span>

<span class="c1">## plot randomly sampled 9 canine patients without outlier mutation profile</span>

<span class="n">plotsig_nonhm_rel_sampled_merged</span> <span class="o">&lt;-</span> <span class="nf">cgp_plot_contribution</span><span class="p">(</span><span class="nf">as_tibble</span><span class="p">(</span>
    <span class="n">fit_res_nonhm_merged[</span> <span class="p">,</span><span class="n">nonhm_sampled]</span><span class="p">,</span> <span class="n">rownames</span> <span class="o">=</span> <span class="s">&quot;sig_id&quot;</span><span class="p">),</span>
    <span class="n">cancer_signatures_merged</span><span class="p">,</span>
    <span class="n">coord_flip</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s">&quot;relative&quot;</span><span class="p">,</span> <span class="n">palette</span> <span class="o">=</span> <span class="n">hm_nonhm_sig_group_colours</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span> <span class="o">=</span> <span class="m">20</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">legend.position</span> <span class="o">=</span> <span class="s">&quot;bottom&quot;</span><span class="p">,</span>
        <span class="n">legend.text</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">14</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&quot;Canine Gliomas with no outlier mutation profile\n(9/73)&quot;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span>
        <span class="n">title</span> <span class="o">=</span> <span class="s">&quot;Signatures grouped by\ntheir proposed mechanism&quot;</span><span class="p">,</span>
    <span class="n">title.position</span> <span class="o">=</span> <span class="s">&quot;bottom&quot;</span><span class="p">,</span>
    <span class="n">label.theme</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">14</span><span class="p">),</span>
    <span class="n">keywidth</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span>
    <span class="n">keyheight</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span>
    <span class="n">default.unit</span><span class="o">=</span><span class="s">&quot;cm&quot;</span><span class="p">,</span> <span class="n">nrow</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;F3A.pdf&quot;</span><span class="p">,</span>
    <span class="nf">grid.arrange</span><span class="p">(</span><span class="n">plotsig_hm_rel_merged</span><span class="p">,</span>
                 <span class="n">plotsig_nonhm_rel_sampled_merged</span><span class="p">,</span>
        <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">widths</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span> <span class="m">5</span><span class="p">)),</span>
        <span class="n">width</span> <span class="o">=</span> <span class="m">18</span><span class="p">,</span>
        <span class="n">height</span> <span class="o">=</span> <span class="m">12</span><span class="p">,</span>
        <span class="n">useDingbats</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<h3 id="suppl-fig-3a">Suppl Fig 3A<a class="headerlink" href="#suppl-fig-3a" title="Permanent link">&para;</a></h3>
<blockquote>
<p>Plot absolute and relative contribution of dominant signatures.</p>
</blockquote>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="c1">## Select signatures with major contribution &gt; 3rd Qu (2.75)</span>
<span class="nf">summary</span><span class="p">(</span><span class="nf">rowMedians</span><span class="p">(</span><span class="n">fit_res</span><span class="o">$</span><span class="n">contribution</span><span class="p">))</span>

<span class="n">select_high</span> <span class="o">&lt;-</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">fit_res</span><span class="o">$</span><span class="n">contribution</span><span class="p">)</span><span class="n">[</span>
    <span class="nf">which</span><span class="p">(</span><span class="nf">rowMedians</span><span class="p">(</span><span class="n">fit_res</span><span class="o">$</span><span class="n">contribution</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">2.75</span><span class="p">)</span>
    <span class="n">]</span>

<span class="n">select_high</span>

<span class="n">color_lib</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;#8e0152&#39;</span><span class="p">,</span><span class="s">&#39;#c51b7d&#39;</span><span class="p">,</span><span class="s">&#39;#de77ae&#39;</span><span class="p">,</span><span class="s">&#39;#f1b6da&#39;</span><span class="p">,</span><span class="s">&#39;#fde0ef&#39;</span><span class="p">,</span>
  <span class="s">&#39;#f7f7f7&#39;</span><span class="p">,</span><span class="s">&#39;#e6f5d0&#39;</span><span class="p">,</span><span class="s">&#39;#b8e186&#39;</span><span class="p">,</span><span class="s">&#39;#7fbc41&#39;</span><span class="p">,</span><span class="s">&#39;#4d9221&#39;</span><span class="p">,</span>
  <span class="s">&#39;#276419&#39;</span><span class="p">)</span>

<span class="c1">## Plot contribution</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;SF3A_top_panel.pdf&quot;</span><span class="p">,</span>
    <span class="nf">plot_contribution</span><span class="p">(</span><span class="n">fit_res</span><span class="o">$</span><span class="n">contribution[select_high</span><span class="p">,</span> <span class="n">]</span><span class="p">,</span>
         <span class="n">cancer_signatures[</span> <span class="p">,</span> <span class="n">select_high]</span><span class="p">,</span>
         <span class="n">coord_flip</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
         <span class="n">mode</span> <span class="o">=</span> <span class="s">&quot;absolute&quot;</span><span class="p">,</span>
         <span class="n">palette</span> <span class="o">=</span> <span class="n">color_lib</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span> <span class="o">=</span> <span class="m">90</span><span class="p">)),</span>
    <span class="n">width</span> <span class="o">=</span> <span class="m">9</span><span class="p">,</span>
    <span class="n">height</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span>
    <span class="n">useDingbats</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;SF3A_bottom_panel.pdf&quot;</span><span class="p">,</span>
    <span class="nf">plot_contribution</span><span class="p">(</span><span class="n">fit_res</span><span class="o">$</span><span class="n">contribution[select_high</span><span class="p">,</span> <span class="n">]</span><span class="p">,</span>
        <span class="n">cancer_signatures[</span> <span class="p">,</span> <span class="n">select_high]</span><span class="p">,</span>
        <span class="n">coord_flip</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
        <span class="n">mode</span> <span class="o">=</span> <span class="s">&quot;relative&quot;</span><span class="p">,</span>
        <span class="n">palette</span> <span class="o">=</span> <span class="n">color_lib</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span> <span class="o">=</span> <span class="m">90</span><span class="p">)),</span>
    <span class="n">width</span> <span class="o">=</span> <span class="m">9</span><span class="p">,</span>
    <span class="n">height</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span>
    <span class="n">useDingbats</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<h3 id="suppl-fig-3b">Suppl Fig 3B<a class="headerlink" href="#suppl-fig-3b" title="Permanent link">&para;</a></h3>
<blockquote>
<p>Boxplot comparison of genomic instability related signatures between canine glioma patients separated by outlier mutation profile.</p>
</blockquote>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">suppl_fig_3B</span> <span class="o">&lt;-</span> <span class="n">ggpubr</span><span class="o">::</span><span class="nf">ggboxplot</span><span class="p">(</span><span class="n">instability_sigs_df</span><span class="p">,</span>
        <span class="n">x</span> <span class="o">=</span> <span class="s">&quot;hypermut&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&quot;sig_contrib&quot;</span><span class="p">,</span>
        <span class="n">add</span> <span class="o">=</span> <span class="s">&quot;jitter&quot;</span><span class="p">,</span>
        <span class="n">facet.by</span> <span class="o">=</span> <span class="s">&quot;sig_group&quot;</span><span class="p">,</span>
        <span class="n">scales</span> <span class="o">=</span> <span class="s">&quot;free_y&quot;</span><span class="p">,</span>
        <span class="n">bxp.errorbar</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">notch</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
        <span class="n">font.label</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">7</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">),</span>
        <span class="n">ylab</span> <span class="o">=</span> <span class="s">&quot;Signature contribution&quot;</span><span class="p">,</span>
        <span class="n">xlab</span> <span class="o">=</span> <span class="s">&quot;Canine Gliomas separated by\noutlier mutation profile&quot;</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">11</span><span class="p">),</span>
        <span class="n">axis.title</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">11</span><span class="p">))</span> <span class="o">+</span>
    <span class="n">ggpubr</span><span class="o">::</span><span class="nf">stat_compare_means</span><span class="p">(</span><span class="n">method</span> <span class="o">=</span> <span class="s">&quot;wilcox.test&quot;</span><span class="p">,</span>
    <span class="n">label.y.npc</span> <span class="o">=</span> <span class="m">0.9</span><span class="p">,</span> <span class="n">na.rm</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;SF3B.pdf&quot;</span><span class="p">,</span>
       <span class="n">suppl_fig_3B</span><span class="p">,</span>
       <span class="n">width</span><span class="o">=</span><span class="m">12</span><span class="p">,</span>
       <span class="n">height</span><span class="o">=</span><span class="m">8</span><span class="p">,</span>
       <span class="n">useDingbats</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<div class="footnote">
<hr />
<ol>
<li id="fn:1">
<p>Earlier version of fit_res_hm had 9 and not 8 canine glioma patients with outlier mutation profile. Therefore, 9 cases were selected at random from remaining patients without outlier mutation profile.&#160;<a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:2">
<p>Blokzijl F, Janssen R, van Boxtel R, Cuppen E. MutationalPatterns: comprehensive genome-wide analysis of mutational processes. Genome Med 2018;10:33. doi: <a href="https://doi.org/10.1186/s13073-018-0539-0">10.1186/s13073-018-0539-0</a>.&#160;<a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
</ol>
</div>
                
                  
                
              
              
                


  <h2 id="__comments">Comments</h2>
  <div id="disqus_thread"></div>
  <script>
    var disqus_config = function () {
      this.page.url = "https://canineglioma.verhaaklab.com/figures/F3A/";
      this.page.identifier =
        "/figures/F3A/";
    };
    (function() {
      var d = document, s = d.createElement("script");
      s.src = "//verhaaklab.disqus.com/embed.js";
      s.setAttribute("data-timestamp", +new Date());
      (d.head || d.body).appendChild(s);
    })();
  </script>

              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../S2/" title="S2 &middot; IGV Plots" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                S2 &middot; IGV Plots
              </span>
            </div>
          </a>
        
        
          <a href="../F3B/" title="F3B &middot; Somatic Mutational Landscape - de novo" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                F3B &middot; Somatic Mutational Landscape - de novo
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2017 - 2020 <a href="https://verhaaklab.com">Verhaak Lab at the Jackson Laboratory for Genomic Medicine</a>
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="https://verhaaklab.com/contact" class="md-footer-social__link fa fa-envelope"></a>
    
      <a href="https://twitter.com/roelverhaak" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://github.com/TheJacksonLaboratory/canineglioma" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/sbamin" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.ffb616e8.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
        <script src="../../assets/js/katex.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js"></script>
      
        <script src="../../search/main.js"></script>
      
    
  </body>
</html>