<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><link href=https://canineglioma.verhaaklab.com/run_code/run_all/ rel=canonical><meta name=author content="Samir B. Amin"><meta name=lang:clipboard.copy content="Copy to clipboard"><meta name=lang:clipboard.copied content="Copied to clipboard"><meta name=lang:search.language content=en><meta name=lang:search.pipeline.stopwords content=True><meta name=lang:search.pipeline.trimmer content=True><meta name=lang:search.result.none content="No matching documents"><meta name=lang:search.result.one content="1 matching document"><meta name=lang:search.result.other content="# matching documents"><meta name=lang:search.tokenizer content=[\s\-\.]+><link rel="shortcut icon" href=../../assets/images/logo.png><meta name=generator content="mkdocs-1.0.4, mkdocs-material-4.4.0"><title>cgp_figures</title><link rel=stylesheet href=../../assets/stylesheets/application.4031d38b.css><link rel=stylesheet href=../../assets/stylesheets/application-palette.224b79ff.css><meta name=theme-color content=#2196f3><script src=../../assets/javascripts/modernizr.74668098.js></script><link href=https://fonts.gstatic.com rel=preconnect crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,400,400i,700|Fira+Code&display=fallback"><style>body,input{font-family:"IBM Plex Sans","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Fira Code","Courier New",Courier,monospace}</style><link rel=stylesheet href=../../assets/fonts/material-icons.css><link rel=stylesheet href=../../assets/css/katex.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css><script async src="https://www.googletagmanager.com/gtag/js?id=UA-64053990-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-64053990-2');
</script><meta content=https://canineglioma.verhaaklab.com/run_code/run_all/ property=og:url><meta property=og:title content=cgp_figures><meta property=og:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta property=og:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta property=og:image:alt content="Image alt"><meta property=og:image:type content=image/jpeg><meta property=og:image:width content=300><meta property=og:image:height content=300><meta name=twitter:card content=summary><meta name=twitter:site content=@roelverhaak><meta name=twitter:creator content=@roelverhaak><meta property=twitter:title content=cgp_figures><meta name=twitter:description content="Canine Glioma Project · Extended Methods, Data, and Code for Figures"><meta name=twitter:image content=https://canineglioma.verhaaklab.com/assets/images/lab_logo_sq-md.jpg><meta name=twitter:image:alt content="Image alt"></head> <body dir=ltr data-md-color-primary=blue data-md-color-accent=blue> <svg class=md-svg> <defs> <!-- <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg> --> <svg xmlns=http://www.w3.org/2000/svg xmlns:xlink=http://www.w3.org/1999/xlink version=1.1 style=width:0;height:0;position:absolute;overflow:hidden;> <defs> <symbol viewbox="0 0 1024 998" aria-labelledby=dpsi-ant-github-title id=__github><title id=dpsi-ant-github-title>icon github</title><path fill=currentColor d="M0 512q0 166 95.5 298.5T343 996q6 1 10 1t6.5-1.5 4-3 2-5 .5-4.5V882q-37 4-66-.5t-45.5-14-29-23.5-17-25.5-9-24T194 780q-9-15-27-27.5t-27-20-2-14.5q50-26 113 66 34 51 119 30 10-41 40-70-116-21-172-86t-56-158q0-87 55-151-22-65 6-137 29-2 65 11.5t50.5 23T384 264q57-16 128.5-16T642 264q13-9 29-19t49-21.5 61-9.5q27 71 6 135 56 64 56 152 0 92-56.5 157.5T615 744q43 43 43 104v129q0 1 1 3 0 6 .5 9t4.5 6 11 3q154-52 251.5-185.5T1024 512q0-104-40.5-199t-109-163.5T711 40.5 512 0 313 40.5t-163.5 109T40.5 313 0 512z"/></symbol> </defs> </svg> </defs> </svg> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay data-md-component=overlay for=__drawer></label> <a href=#comparative-molecular-life-history-of-spontaneous-canine-and-human-gliomas tabindex=1 class=md-skip> Skip to content </a> <header class=md-header data-md-component=header> <nav class="md-header-nav md-grid"> <div class=md-flex> <div class="md-flex__cell md-flex__cell--shrink"> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-header-nav__button md-logo"> <img src=../../assets/images/logo.png width=24 height=24> </a> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--menu md-header-nav__button" for=__drawer></label> </div> <div class="md-flex__cell md-flex__cell--stretch"> <div class="md-flex__ellipsis md-header-nav__title" data-md-component=title> <span class=md-header-nav__topic> Canine Glioma Project &middot; Verhaak Lab </span> <span class=md-header-nav__topic> cgp_figures </span> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <label class="md-icon md-icon--search md-header-nav__button" for=__search></label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=query data-md-state=active> <label class="md-icon md-search__icon" for=__search></label> <button type=reset class="md-icon md-search__icon" data-md-component=reset tabindex=-1> &#xE5CD; </button> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=result> <div class=md-search-result__meta> Type to start searching </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> </div> <div class="md-flex__cell md-flex__cell--shrink"> <div class=md-header-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> </div> </div> </nav> </header> <div class=md-container> <main class=md-main> <div class="md-main__inner md-grid" data-md-component=container> <div class="md-sidebar md-sidebar--primary" data-md-component=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" data-md-level=0> <label class="md-nav__title md-nav__title--site" for=__drawer> <a href=https://canineglioma.verhaaklab.com title="Canine Glioma Project · Verhaak Lab" class="md-nav__button md-logo"> <img src=../../assets/images/logo.png width=48 height=48> </a> Canine Glioma Project &middot; Verhaak Lab </label> <div class=md-nav__source> <a href=https://github.com/TheJacksonLaboratory/canineglioma/ title="Go to repository" class=md-source data-md-source=github> <div class=md-source__icon> <svg viewbox="0 0 24 24" width=24 height=24> <use xlink:href=#__github width=24 height=24></use> </svg> </div> <div class=md-source__repository> TheJacksonLaboratory/canineglioma </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. title=Home class=md-nav__link> Home </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-2 type=checkbox id=nav-2> <label class=md-nav__link for=nav-2> Methods </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-2> Methods </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../methods/S01_samples/ title=Samples class=md-nav__link> Samples </a> </li> <li class=md-nav__item> <a href=../../methods/S02_data_sources/ title="Data Sources" class=md-nav__link> Data Sources </a> </li> <li class=md-nav__item> <a href=../../methods/S03_sample_prep/ title="Sample Preparations" class=md-nav__link> Sample Preparations </a> </li> <li class=md-nav__item> <a href=../../methods/S04_alignments_fprint/ title=Alignments class=md-nav__link> Alignments </a> </li> <li class=md-nav__item> <a href=../../methods/S05_somatic_variants/ title="Somatic SNVs and Indels" class=md-nav__link> Somatic SNVs and Indels </a> </li> <li class=md-nav__item> <a href=../../methods/S06_smgs/ title="Significantly Mutated Genes" class=md-nav__link> Significantly Mutated Genes </a> </li> <li class=md-nav__item> <a href=../../methods/S07_hallmarks/ title="Cancer Hallmarks" class=md-nav__link> Cancer Hallmarks </a> </li> <li class=md-nav__item> <a href=../../methods/S09_mutation_rate/ title="Somatic Mutation Rate" class=md-nav__link> Somatic Mutation Rate </a> </li> <li class=md-nav__item> <a href=../../methods/S10_SCNA/ title="Somatic Copy Number Aberrations" class=md-nav__link> Somatic Copy Number Aberrations </a> </li> <li class=md-nav__item> <a href=../../methods/S11_aneuploidy/ title="Aneuploidy Metrics" class=md-nav__link> Aneuploidy Metrics </a> </li> <li class=md-nav__item> <a href=../../methods/S12_ith/ title="Intratumor Heterogeneity" class=md-nav__link> Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../../methods/S13_mut_sigs/ title="Mutational Signatures" class=md-nav__link> Mutational Signatures </a> </li> <li class=md-nav__item> <a href=../../methods/S14_mut_timing/ title="Molecular Timing of Somatic Drivers" class=md-nav__link> Molecular Timing of Somatic Drivers </a> </li> <li class=md-nav__item> <a href=../../methods/S15_class_predict_methy/ title="Class Prediction using methylation classifier" class=md-nav__link> Class Prediction using methylation classifier </a> </li> <li class=md-nav__item> <a href=../../methods/S16_expression_immune/ title="Immune Microenvironment Analysis" class=md-nav__link> Immune Microenvironment Analysis </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3 type=checkbox id=nav-3> <label class=md-nav__link for=nav-3> Figures </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-3> Figures </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/A1_preload/ title="Start Here" class=md-nav__link> Start Here </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-2 type=checkbox id=nav-3-2> <label class=md-nav__link for=nav-3-2> F1 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-2> F1 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/F1A/ title="F1A · Somatic Mutational Landscape" class=md-nav__link> F1A &middot; Somatic Mutational Landscape </a> </li> <li class=md-nav__item> <a href=../../figures/F1B/ title="F1B · Somatic Recurrent Hotspot Mutations" class=md-nav__link> F1B &middot; Somatic Recurrent Hotspot Mutations </a> </li> <li class=md-nav__item> <a href=../../figures/F1C/ title="F1C · Comparative Pathway Analysis" class=md-nav__link> F1C &middot; Comparative Pathway Analysis </a> </li> <li class=md-nav__item> <a href=../../figures/F1D/ title="F1D · Comparative Somatic Mutational Burden" class=md-nav__link> F1D &middot; Comparative Somatic Mutational Burden </a> </li> <li class=md-nav__item> <a href=../../figures/S1/ title="S1A · DNA fingerprinting" class=md-nav__link> S1A &middot; DNA fingerprinting </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-3 type=checkbox id=nav-3-3> <label class=md-nav__link for=nav-3-3> F2 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-3> F2 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/F2A/ title="F2A · Somatic Copy Number Landscape" class=md-nav__link> F2A &middot; Somatic Copy Number Landscape </a> </li> <li class=md-nav__item> <a href=../../figures/F2B/ title="F2B · Comparative Aneuploidy Burden" class=md-nav__link> F2B &middot; Comparative Aneuploidy Burden </a> </li> <li class=md-nav__item> <a href=../../figures/F2C/ title="F2C · Comparative Shared Syntenic Aneuploidy" class=md-nav__link> F2C &middot; Comparative Shared Syntenic Aneuploidy </a> </li> <li class=md-nav__item> <a href=../../figures/F2D/ title="F2D · Comparative Intratumor Heterogeneity" class=md-nav__link> F2D &middot; Comparative Intratumor Heterogeneity </a> </li> <li class=md-nav__item> <a href=../../figures/S2/ title="S2 · IGV Plots" class=md-nav__link> S2 &middot; IGV Plots </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-4 type=checkbox id=nav-3-4> <label class=md-nav__link for=nav-3-4> F3 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-4> F3 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/F3A/ title="F3A · Somatic Mutational Signatures - Deconvoluted" class=md-nav__link> F3A &middot; Somatic Mutational Signatures - Deconvoluted </a> </li> <li class=md-nav__item> <a href=../../figures/F3B/ title="F3B · Somatic Mutational Landscape - de novo" class=md-nav__link> F3B &middot; Somatic Mutational Landscape - de novo </a> </li> <li class=md-nav__item> <a href=../../figures/F3C/ title="F3C · Comparative Somatic Mutational Signatures in Driver Genes" class=md-nav__link> F3C &middot; Comparative Somatic Mutational Signatures in Driver Genes </a> </li> <li class=md-nav__item> <a href=../../figures/F3D/ title="F3D · Comparative Molecular Timing of Driver Events" class=md-nav__link> F3D &middot; Comparative Molecular Timing of Driver Events </a> </li> <li class=md-nav__item> <a href=../../figures/S3/ title="S3 · Suppl Figures" class=md-nav__link> S3 &middot; Suppl Figures </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-5 type=checkbox id=nav-3-5> <label class=md-nav__link for=nav-3-5> F4 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-5> F4 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/F4/ title="F4 · DNA Methylation based Prediction of Canine Gliomas" class=md-nav__link> F4 &middot; DNA Methylation based Prediction of Canine Gliomas </a> </li> <li class=md-nav__item> <a href=../../figures/S4/ title="S4 · Suppl Figure 4" class=md-nav__link> S4 &middot; Suppl Figure 4 </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-3-6 type=checkbox id=nav-3-6> <label class=md-nav__link for=nav-3-6> F5 </label> <nav class=md-nav data-md-component=collapsible data-md-level=2> <label class=md-nav__title for=nav-3-6> F5 </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../figures/F5/ title="F5B · Comparative Immunohistochemistry - Quantitative" class=md-nav__link> F5B &middot; Comparative Immunohistochemistry - Quantitative </a> </li> <li class=md-nav__item> <a href=../../figures/S5/ title="S5 · Suppl Figure 5" class=md-nav__link> S5 &middot; Suppl Figure 5 </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-toggle md-nav__toggle" data-md-toggle=nav-4 type=checkbox id=nav-4 checked> <label class=md-nav__link for=nav-4> R Markdown </label> <nav class=md-nav data-md-component=collapsible data-md-level=1> <label class=md-nav__title for=nav-4> R Markdown </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../ title=How-To class=md-nav__link> How-To </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-toggle md-nav__toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Output </label> <a href=./ title=Output class="md-nav__link md-nav__link--active"> Output </a> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#comparative-molecular-life-history-of-spontaneous-canine-and-human-gliomas title="Comparative molecular life history of spontaneous canine and human gliomas" class=md-nav__link> Comparative molecular life history of spontaneous canine and human gliomas </a> </li> <li class=md-nav__item> <a href=#set-env title="Set Env" class=md-nav__link> Set Env </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#load-r-packages title="Load R packages" class=md-nav__link> Load R packages </a> </li> <li class=md-nav__item> <a href=#load-r-objects title="Load R objects" class=md-nav__link> Load R objects </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1a title="Fig 1A" class=md-nav__link> Fig 1A </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#plot-fig-1a title="Plot Fig 1A" class=md-nav__link> Plot Fig 1A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1b title="Suppl Fig 1B" class=md-nav__link> Suppl Fig 1B </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1f title="Suppl Fig 1F" class=md-nav__link> Suppl Fig 1F </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1b title="Fig 1B" class=md-nav__link> Fig 1B </a> </li> <li class=md-nav__item> <a href=#fig-1c title="Fig 1C" class=md-nav__link> Fig 1C </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#fig-1c_1 title="Fig 1C" class=md-nav__link> Fig 1C </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1c title="Suppl Fig 1C" class=md-nav__link> Suppl Fig 1C </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1d title="Fig 1D" class=md-nav__link> Fig 1D </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-1d title="Suppl Fig 1D" class=md-nav__link> Suppl Fig 1D </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1e title="Suppl Fig 1E" class=md-nav__link> Suppl Fig 1E </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#f2a title=F2A class=md-nav__link> F2A </a> </li> <li class=md-nav__item> <a href=#fig-2b title="Fig 2B" class=md-nav__link> Fig 2B </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-2e title="Suppl Fig 2E" class=md-nav__link> Suppl Fig 2E </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-2c title="Fig 2C" class=md-nav__link> Fig 2C </a> </li> <li class=md-nav__item> <a href=#fig-2d title="Fig 2D" class=md-nav__link> Fig 2D </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-2d title="Suppl Fig 2D" class=md-nav__link> Suppl Fig 2D </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3a title="Fig 3A" class=md-nav__link> Fig 3A </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_1 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-3a_1 title="Fig 3A" class=md-nav__link> Fig 3A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-3a title="Suppl Fig 3A" class=md-nav__link> Suppl Fig 3A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-3b title="Suppl Fig 3B" class=md-nav__link> Suppl Fig 3B </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3b title="Fig 3B" class=md-nav__link> Fig 3B </a> </li> <li class=md-nav__item> <a href=#fig-3c title="Fig 3C" class=md-nav__link> Fig 3C </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#winning-tables title="Winning Tables" class=md-nav__link> Winning Tables </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3d title="Fig 3D" class=md-nav__link> Fig 3D </a> </li> <li class=md-nav__item> <a href=#fig-4 title="Fig 4" class=md-nav__link> Fig 4 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#preprocessing title=Preprocessing class=md-nav__link> Preprocessing </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#format-data-for-plotting title="Format data for plotting" class=md-nav__link> Format data for plotting </a> </li> <li class=md-nav__item> <a href=#create-table-for-plotting-annotations title="Create table for plotting annotations" class=md-nav__link> Create table for plotting annotations </a> </li> <li class=md-nav__item> <a href=#labelscolors-for-plot title="Labels/colors for plot" class=md-nav__link> Labels/colors for plot </a> </li> <li class=md-nav__item> <a href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Plot probabilities and predictions data frame simultaneously" class=md-nav__link> Plot probabilities and predictions data frame simultaneously </a> </li> <li class=md-nav__item> <a href=#add-annotations-for-mutation-rate-purity-and-age title="Add annotations for mutation rate, purity, and age" class=md-nav__link> Add annotations for mutation rate, purity, and age </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-4_1 title="Fig 4" class=md-nav__link> Fig 4 </a> </li> <li class=md-nav__item> <a href=#suppl-fig-4a title="Suppl Fig 4A" class=md-nav__link> Suppl Fig 4A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-4b title="Suppl Fig 4B" class=md-nav__link> Suppl Fig 4B </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-5 title="Fig 5" class=md-nav__link> Fig 5 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_2 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-5b title="Fig 5B" class=md-nav__link> Fig 5B </a> </li> <li class=md-nav__item> <a href=#p-values title=p-values class=md-nav__link> p-values </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-5_1 title="Fig 5" class=md-nav__link> Fig 5 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_3 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-s5 title="Fig S5" class=md-nav__link> Fig S5 </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=https://verhaaklab.com/contact title=Contact class=md-nav__link> Contact </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary"> <label class=md-nav__title for=__toc>Table of contents</label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#comparative-molecular-life-history-of-spontaneous-canine-and-human-gliomas title="Comparative molecular life history of spontaneous canine and human gliomas" class=md-nav__link> Comparative molecular life history of spontaneous canine and human gliomas </a> </li> <li class=md-nav__item> <a href=#set-env title="Set Env" class=md-nav__link> Set Env </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#load-r-packages title="Load R packages" class=md-nav__link> Load R packages </a> </li> <li class=md-nav__item> <a href=#load-r-objects title="Load R objects" class=md-nav__link> Load R objects </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1a title="Fig 1A" class=md-nav__link> Fig 1A </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#plot-fig-1a title="Plot Fig 1A" class=md-nav__link> Plot Fig 1A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1b title="Suppl Fig 1B" class=md-nav__link> Suppl Fig 1B </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1f title="Suppl Fig 1F" class=md-nav__link> Suppl Fig 1F </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1b title="Fig 1B" class=md-nav__link> Fig 1B </a> </li> <li class=md-nav__item> <a href=#fig-1c title="Fig 1C" class=md-nav__link> Fig 1C </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#fig-1c_1 title="Fig 1C" class=md-nav__link> Fig 1C </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1c title="Suppl Fig 1C" class=md-nav__link> Suppl Fig 1C </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-1d title="Fig 1D" class=md-nav__link> Fig 1D </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-1d title="Suppl Fig 1D" class=md-nav__link> Suppl Fig 1D </a> </li> <li class=md-nav__item> <a href=#suppl-fig-1e title="Suppl Fig 1E" class=md-nav__link> Suppl Fig 1E </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#f2a title=F2A class=md-nav__link> F2A </a> </li> <li class=md-nav__item> <a href=#fig-2b title="Fig 2B" class=md-nav__link> Fig 2B </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-2e title="Suppl Fig 2E" class=md-nav__link> Suppl Fig 2E </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-2c title="Fig 2C" class=md-nav__link> Fig 2C </a> </li> <li class=md-nav__item> <a href=#fig-2d title="Fig 2D" class=md-nav__link> Fig 2D </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#suppl-fig-2d title="Suppl Fig 2D" class=md-nav__link> Suppl Fig 2D </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3a title="Fig 3A" class=md-nav__link> Fig 3A </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_1 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-3a_1 title="Fig 3A" class=md-nav__link> Fig 3A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-3a title="Suppl Fig 3A" class=md-nav__link> Suppl Fig 3A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-3b title="Suppl Fig 3B" class=md-nav__link> Suppl Fig 3B </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3b title="Fig 3B" class=md-nav__link> Fig 3B </a> </li> <li class=md-nav__item> <a href=#fig-3c title="Fig 3C" class=md-nav__link> Fig 3C </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#winning-tables title="Winning Tables" class=md-nav__link> Winning Tables </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-3d title="Fig 3D" class=md-nav__link> Fig 3D </a> </li> <li class=md-nav__item> <a href=#fig-4 title="Fig 4" class=md-nav__link> Fig 4 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#preprocessing title=Preprocessing class=md-nav__link> Preprocessing </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#format-data-for-plotting title="Format data for plotting" class=md-nav__link> Format data for plotting </a> </li> <li class=md-nav__item> <a href=#create-table-for-plotting-annotations title="Create table for plotting annotations" class=md-nav__link> Create table for plotting annotations </a> </li> <li class=md-nav__item> <a href=#labelscolors-for-plot title="Labels/colors for plot" class=md-nav__link> Labels/colors for plot </a> </li> <li class=md-nav__item> <a href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Plot probabilities and predictions data frame simultaneously" class=md-nav__link> Plot probabilities and predictions data frame simultaneously </a> </li> <li class=md-nav__item> <a href=#add-annotations-for-mutation-rate-purity-and-age title="Add annotations for mutation rate, purity, and age" class=md-nav__link> Add annotations for mutation rate, purity, and age </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-4_1 title="Fig 4" class=md-nav__link> Fig 4 </a> </li> <li class=md-nav__item> <a href=#suppl-fig-4a title="Suppl Fig 4A" class=md-nav__link> Suppl Fig 4A </a> </li> <li class=md-nav__item> <a href=#suppl-fig-4b title="Suppl Fig 4B" class=md-nav__link> Suppl Fig 4B </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-5 title="Fig 5" class=md-nav__link> Fig 5 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_2 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-5b title="Fig 5B" class=md-nav__link> Fig 5B </a> </li> <li class=md-nav__item> <a href=#p-values title=p-values class=md-nav__link> p-values </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#fig-5_1 title="Fig 5" class=md-nav__link> Fig 5 </a> <nav class=md-nav> <ul class=md-nav__list> <li class=md-nav__item> <a href=#required-metadata_3 title="Required metadata" class=md-nav__link> Required metadata </a> </li> <li class=md-nav__item> <a href=#fig-s5 title="Fig S5" class=md-nav__link> Fig S5 </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#__comments title=Comments class="md-nav__link md-nav__link--active"> Comments </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content> <article class="md-content__inner md-typeset"> <!-- enable git edit link only if config.repo_edit is set to true --> <h1>Output</h1> <h2 id=comparative-molecular-life-history-of-spontaneous-canine-and-human-gliomas>Comparative molecular life history of spontaneous canine and human gliomas<a class=headerlink href=#comparative-molecular-life-history-of-spontaneous-canine-and-human-gliomas title="Permanent link">&para;</a></h2> <blockquote> <p>Amin et al., Comparative Molecular Life History of Spontaneous Canine and Human Gliomas, <em>Cancer Cell</em> (2020). DOI: <a href=https://doi.org/10.1016/j.ccell.2020.01.004>10.1016/j.ccell.2020.01.004</a> Feb 10, 2020.</p> </blockquote> <h2 id=set-env>Set Env<a class=headerlink href=#set-env title="Permanent link">&para;</a></h2> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span><span class=nf>library</span><span class=p>(</span><span class=n>knitr</span><span class=p>)</span>
<span class=n>opts_chunk</span><span class=o>$</span><span class=nf>set</span><span class=p>(</span><span class=n>comment</span> <span class=o>=</span> <span class=kc>NA</span><span class=p>,</span> <span class=n>fig.width</span> <span class=o>=</span> <span class=m>18</span><span class=p>,</span> <span class=n>fig.height</span> <span class=o>=</span> <span class=m>12</span><span class=p>,</span> 
               <span class=n>message</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=n>cache</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span> <span class=n>echo</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
</pre></div> </td></tr></table> <h3 id=load-r-packages>Load R packages<a class=headerlink href=#load-r-packages title="Permanent link">&para;</a></h3> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning: replacing previous import &#39;NMF::entropy&#39; by &#39;lsa::entropy&#39; when loading
&#39;Palimpsest&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning: replacing previous import &#39;NMF::dispersion&#39; by &#39;plotrix::dispersion&#39;
when loading &#39;Palimpsest&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning: replacing previous import &#39;gplots::plotCI&#39; by &#39;plotrix::plotCI&#39; when
loading &#39;Palimpsest&#39;
</pre></div> </td></tr></table> <h3 id=load-r-objects>Load R objects<a class=headerlink href=#load-r-objects title="Permanent link">&para;</a></h3> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Work directory is /Users/samin1/Dropbox (Personal)/acad/scripts/HOURGLASS/cgp_keystone/canineglioma/docs/run_code
Loading base objects...
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>List objects
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</pre></div></td><td class=code><div class=codehilite><pre><span></span> [1] &quot;ALL_HALLMARK_v1&quot;                     
 [2] &quot;cancer_genes_maftools&quot;               
 [3] &quot;cancer_signatures&quot;                   
 [4] &quot;CG_all&quot;                              
 [5] &quot;cgp_aneuploidy_metrics_master&quot;       
 [6] &quot;cgp_cibersort&quot;                       
 [7] &quot;cgp_info&quot;                            
 [8] &quot;cgp_maftools_gistic_matched&quot;         
 [9] &quot;cgp_maftools_gistic_methy&quot;           
[10] &quot;cgp_maftools_gistic_n81&quot;             
[11] &quot;cgp_RRBS.noNormals_class_prediction&quot; 
[12] &quot;cgp_RRBS.noNormals_sample_info&quot;      
[13] &quot;cgp_suppl_sample_info_master&quot;        
[14] &quot;driver_sigs_groups&quot;                  
[15] &quot;fig5b_dat&quot;                           
[16] &quot;fit_res_hm&quot;                          
[17] &quot;fit_res_nonhm&quot;                       
[18] &quot;HALLMARK&quot;                            
[19] &quot;hm_nonhm_topsigs_tbl&quot;                
[20] &quot;instability_sigs_df&quot;                 
[21] &quot;maf_nogistic&quot;                        
[22] &quot;mat_aneuploidy_summary_mostvar_cgAll&quot;
[23] &quot;merged_ag_pg_cg_denovo_sig&quot;          
[24] &quot;merged_anp_scores&quot;                   
[25] &quot;merged_ccf_vaf_df&quot;                   
[26] &quot;mut_mat&quot;                             
[27] &quot;mutrate_table&quot;                       
[28] &quot;ped_cibersort&quot;                       
[29] &quot;tcga_cibersort&quot;                      
[30] &quot;tcga_info&quot;                           
[31] &quot;trimmed_ag_vcf&quot;                      
[32] &quot;trimmed_cg_vcf&quot;                      
[33] &quot;trimmed_pg_vcf&quot;                      
[34] &quot;updated_merged_entropy_df_ridges&quot;    
[35] &quot;win_table_ag_vcf&quot;                    
[36] &quot;win_table_cg_vcf&quot;                    
[37] &quot;win_table_pg_vcf&quot;                    
</pre></div> </td></tr></table> <hr> <h2 id=fig-1a>Fig 1A<a class=headerlink href=#fig-1a title="Permanent link">&para;</a></h2> <h3 id=required-metadata>Required metadata<a class=headerlink href=#required-metadata title="Permanent link">&para;</a></h3> <ul> <li>Genes to plot</li> </ul> <blockquote> <p>Based on <a href=/methods/S06_smgs>significantly mutated genes analysis</a>. See also Table S2.</p> </blockquote> <ul> <li>Sample and Variant Annotations </li> </ul> <h3 id=plot-fig-1a>Plot Fig 1A<a class=headerlink href=#plot-fig-1a title="Permanent link">&para;</a></h3> <blockquote> <p>Somatic mutational lanscape of glioma driver genes. Based on <a href=/methods/S06_smgs>significantly mutated genes analysis</a>.</p> </blockquote> <details class=error><summary>font family 'Arial' not found</summary><p>If you get an error, <code>font family 'Arial' not found in PostScript font database</code>, either remove <em>fonts</em> argument from <code>pdf</code> or import system fonts using <em>extrafont</em> R package. See <a href=https://www.fromthebottomoftheheap.net/2013/09/09/preparing-figures-for-plos-one-with-r/ >documentation by Gavin Simpson</a> for details. Run <code>extrafont::loadfonts()</code> as detailed in <a href=/figures/a1_preload/#load-r-packages>Start Here</a>.</p> </details> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=suppl-fig-1b>Suppl Fig 1B<a class=headerlink href=#suppl-fig-1b title="Permanent link">&para;</a></h3> <blockquote> <p>Somatic mutational lanscape of COSMIC cancer genes.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=suppl-fig-1f>Suppl Fig 1F<a class=headerlink href=#suppl-fig-1f title="Permanent link">&para;</a></h3> <blockquote> <p>Somatic mutational lanscape, including copy-number aberrations of COSMIC cancer genes. This figure also contains somatic copy-number data which is detailed under <a href=/figures/F2A/ >Fig 2A</a>.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <hr> <h2 id=fig-1b>Fig 1B<a class=headerlink href=#fig-1b title="Permanent link">&para;</a></h2> <ul> <li>Alternate code using <code>lollipopPlot</code> function in R package, <a href=https://bioconductor.org/packages/release/bioc/html/maftools.html><em>maftools</em></a></li> </ul> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5
6</pre></div></td><td class=code><div class=codehilite><pre><span></span>    pos   conv count Variant_Classification
1:   88   R88Q     1      Missense_Mutation
2:  542  E542K     1      Missense_Mutation
3:  545  E545K     1      Missense_Mutation
4: 1047 H1047R     7      Missense_Mutation
5: 1047 H1047L     1      Missense_Mutation
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>   pos  conv count Variant_Classification
1: 132 R132C     3      Missense_Mutation
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>6 transcripts available. Use arguments refSeqID or proteinID to manually specify tx name.
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5
6
7</pre></div></td><td class=code><div class=codehilite><pre><span></span>   HGNC    refseq.ID   protein.ID aa.length
1: SPOP NM_001007227 NP_001007228       374
2: SPOP NM_001007228 NP_001007229       374
3: SPOP NM_001007229 NP_001007230       374
4: SPOP NM_001007230 NP_001007231       374
5: SPOP NM_001007226 NP_001007227       374
6: SPOP    NM_003563    NP_003554       374
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Using longer transcript NM_001007227 for now.
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>   pos conv count Variant_Classification
1:  94 P94R     2      Missense_Mutation
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <hr> <h2 id=fig-1c>Fig 1C<a class=headerlink href=#fig-1c title="Permanent link">&para;</a></h2> <p>Cancer Hallmarks enrichment across canine, human pediatric and adult gliomas. </p> <blockquote> <p>Genes to hallmark mapping was primarily derived from SLAPenrich R package. See also <a href=/methods/S07_hallmarks/ >relevant methods</a> for details.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Joining, by = &quot;hallmark&quot;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning: Column `hallmark` joining character vector and factor, coercing into
character vector
</pre></div> </td></tr></table> <ul> <li>Similar to <code>CG_HALLMARK</code>, enrichment per hallmark tables were created for molecular subtypes across human cancers, and merged in the master table, <code>ALL_HALLMARK_v1</code>.</li> </ul> <h3 id=fig-1c_1>Fig 1C<a class=headerlink href=#fig-1c_1 title="Permanent link">&para;</a></h3> <blockquote> <p>Subset of hallmarks as a main figure 1C.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=suppl-fig-1c>Suppl Fig 1C<a class=headerlink href=#suppl-fig-1c title="Permanent link">&para;</a></h3> <blockquote> <p>Visualize all hallmarks.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <hr> <h2 id=fig-1d>Fig 1D<a class=headerlink href=#fig-1d title="Permanent link">&para;</a></h2> <blockquote> <p>Plot CNS tumors.</p> </blockquote> <h3 id=suppl-fig-1d>Suppl Fig 1D<a class=headerlink href=#suppl-fig-1d title="Permanent link">&para;</a></h3> <blockquote> <p>Plot pancancer tumors</p> </blockquote> <details class=error><summary>unexpected <code>,</code> error</summary><p>Running code by copy-paste may result in error related to R syntax, e.g., <code>Error: unexpected ',' in "</code>. If so, try to copy code first in text editor or RStudio app with lint feature. Please <a href=https://github.com/TheJacksonLaboratory/canineglioma/issues>report bugs</a> by submitting a GitHub issue.</p> </details> <h3 id=suppl-fig-1e>Suppl Fig 1E<a class=headerlink href=#suppl-fig-1e title="Permanent link">&para;</a></h3> <blockquote> <p>Comparing mutational burden across molecular subtypes.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in if (fill %in% names(data) &amp; is.null(add.params$fill)) add.params$fill
&lt;- fill: the condition has length &gt; 1 and only the first element will be used
</pre></div> </td></tr></table> <hr> <h2 id=f2a>F2A<a class=headerlink href=#f2a title="Permanent link">&para;</a></h2> <blockquote> <p>Somatic copy-number aberrations (SCNA) lanscape of glioma driver genes. Please run required code from <a href=/figures/F1A/ >Fig 1A</a> prior to running code from this page.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <blockquote> <p>Somatic copy numbers were classified as follows based on GISTIC2 values.</p> </blockquote> <table> <thead> <tr> <th>GISTIC2 value</th> <th>Estimated copy number</th> <th>Classification</th> </tr> </thead> <tbody> <tr> <td>$&lt; -1.29$</td> <td>-2</td> <td>Deep deletion</td> </tr> <tr> <td>$-1.29 &lt;&gt; -.65$</td> <td>-1</td> <td>Loss of Heterozygosity (LOH)</td> </tr> <tr> <td>$-.65 &lt;&gt; 2.0$</td> <td>0-1</td> <td>Copy-neutral or low-level amplification</td> </tr> <tr> <td>$&gt;2.0$</td> <td>2</td> <td>High-level amplification</td> </tr> </tbody> </table> <hr> <h2 id=fig-2b>Fig 2B<a class=headerlink href=#fig-2b title="Permanent link">&para;</a></h2> <blockquote> <p>Comparative aneuploidy burden. Related code is available at the GLASS consortium <a href=https://github.com/TheJacksonLaboratory/GLASS/tree/master/sql/cnv>marker paper code repository</a>, namely, <code>recapseg_postgres.sql</code> and <code>taylor_aneuploidy.sql</code>.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in if (fill %in% names(data) &amp; is.null(add.params$fill)) add.params$fill
&lt;- fill: the condition has length &gt; 1 and only the first element will be used
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>notch went outside hinges. Try setting notch=FALSE.
notch went outside hinges. Try setting notch=FALSE.
</pre></div> </td></tr></table> <h3 id=suppl-fig-2e>Suppl Fig 2E<a class=headerlink href=#suppl-fig-2e title="Permanent link">&para;</a></h3> <blockquote> <p>Aneuploidy vs. somatic mutation burden.</p> </blockquote> <hr> <h2 id=fig-2c>Fig 2C<a class=headerlink href=#fig-2c title="Permanent link">&para;</a></h2> <p><code>mat_aneuploidy_summary_mostvar_cgAll</code> matrix contains proportion of patients per cohort with arm-level aneuploidy, and was based on <a href=/methods/S11_aneuploidy/ >arm-level aneuploidy metrics</a>. Related code is available at the GLASS consortium <a href=https://github.com/TheJacksonLaboratory/GLASS/tree/master/sql/cnv>marker paper code repository</a>, namely, <code>recapseg_postgres.sql</code> and <code>taylor_aneuploidy.sql</code>.</p> <hr> <h2 id=fig-2d>Fig 2D<a class=headerlink href=#fig-2d title="Permanent link">&para;</a></h2> <blockquote> <p>Clonality of potential driver genes with respect to intra-tumor heterogeneity as measured with Shannon entropy.</p> </blockquote> <details class=error><summary>Error: unexpected symbol in...<code>who_subtype</code>...</summary><p>Prefer runnning code in RStudio. This error does not appear when running code in RStudio but pops up when running on command-line R. Likely due to font ligatures and/or use of of line breaks.</p> </details> <h3 id=suppl-fig-2d>Suppl Fig 2D<a class=headerlink href=#suppl-fig-2d title="Permanent link">&para;</a></h3> <blockquote> <p>Comparative intra-tumor heterogeneity measured using Shannon entropy.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in if (fill %in% names(data) &amp; is.null(add.params$fill)) add.params$fill
&lt;- fill: the condition has length &gt; 1 and only the first element will be used
</pre></div> </td></tr></table> <hr> <h2 id=fig-3a>Fig 3A<a class=headerlink href=#fig-3a title="Permanent link">&para;</a></h2> <h3 id=required-metadata_1>Required metadata<a class=headerlink href=#required-metadata_1 title="Permanent link">&para;</a></h3> <ul> <li>Canine glioma patients with <a href=/methods/S13_mut_sigs/ >outlier mutation profile</a>.</li> </ul> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class=code><div class=codehilite><pre><span></span>Call:
lm(formula = adusted_coding_mutrate ~ Matched_Normal + Tissue_Archival, 
    data = cgp_sample_info_maftools_hypermuts)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.4620 -0.2753 -0.1240  0.1106  6.0669 

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                 0.36934    0.14806   2.495   0.0148 *  
Matched_Normaltumor_only    1.49685    0.24842   6.025 5.62e-08 ***
Tissue_Archivalsnap-frozen  0.05463    0.19010   0.287   0.7746    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.8392 on 76 degrees of freedom
Multiple R-squared:  0.3238,    Adjusted R-squared:  0.306 
F-statistic:  18.2 on 2 and 76 DF,  p-value: 3.492e-07
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>[1] &quot;hypermut case: i_8228&quot; &quot;hypermut case: i_95FC&quot;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>[1] &quot;outlier among non_hypermut: i_BD58&quot; &quot;outlier among non_hypermut: i_1C36&quot;
[3] &quot;outlier among non_hypermut: i_502F&quot; &quot;outlier among non_hypermut: i_D732&quot;
[5] &quot;outlier among non_hypermut: i_6B06&quot; &quot;outlier among non_hypermut: i_56B5&quot;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>[1] &quot;i_8228&quot; &quot;i_95FC&quot; &quot;i_BD58&quot; &quot;i_1C36&quot; &quot;i_502F&quot; &quot;i_D732&quot; &quot;i_6B06&quot; &quot;i_56B5&quot;
</pre></div> </td></tr></table> <blockquote> <p>"i_8228" "i_95FC" "i_BD58" "i_1C36" "i_502F" "i_D732" "i_6B06" "i_56B5"</p> </blockquote> <ul> <li> <p>Pick random 9/73 cases without outlier mutation profile.</p> </li> <li> <p>Deconvolute known human mutational signatures</p> </li> </ul> <blockquote> <p>Fitted mutational signatures, i.e., <code>fit_res</code>, <code>fit_res_hm</code> and <code>fit_res_nonhm</code> were derived using <code>fit_to_signatures</code> function from <em>MutationalPatterns</em> R package.</p> </blockquote> <ul> <li>Merge signatures by proposed mechanism</li> </ul> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>Joining, by = &quot;sig_id&quot;
Joining, by = &quot;sig_id&quot;
Joining, by = &quot;sig_id&quot;
</pre></div> </td></tr></table> <ul> <li>Function to plot signature contribution</li> </ul> <blockquote> <p>Rewrite of <code>plot_contribution</code> function from <em>MutationalPatterns</em> R package.</p> </blockquote> <h3 id=fig-3a_1>Fig 3A<a class=headerlink href=#fig-3a_1 title="Permanent link">&para;</a></h3> <blockquote> <p>Plot deconvoluted signature contribution after grouping signatures based on proposed mechanism.</p> </blockquote> <h3 id=suppl-fig-3a>Suppl Fig 3A<a class=headerlink href=#suppl-fig-3a title="Permanent link">&para;</a></h3> <blockquote> <p>Plot absolute and relative contribution of dominant signatures.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   3.282   2.750  35.465 
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span> [1] &quot;Signature 1&quot;    &quot;Signature 2&quot;    &quot;Signature 8&quot;    &quot;Signature 9&quot;   
 [5] &quot;Signature 11&quot;   &quot;Signature 15&quot;   &quot;Signature 27&quot;   &quot;Signature T-1&quot; 
 [9] &quot;Signature T-3&quot;  &quot;Signature T-10&quot; &quot;Signature T-11&quot;
</pre></div> </td></tr></table> <h3 id=suppl-fig-3b>Suppl Fig 3B<a class=headerlink href=#suppl-fig-3b title="Permanent link">&para;</a></h3> <blockquote> <p>Boxplot comparison of genomic instability related signatures between canine glioma patients separated by outlier mutation profile.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5</pre></div></td><td class=code><div class=codehilite><pre><span></span>notch went outside hinges. Try setting notch=FALSE.
notch went outside hinges. Try setting notch=FALSE.
notch went outside hinges. Try setting notch=FALSE.
notch went outside hinges. Try setting notch=FALSE.
notch went outside hinges. Try setting notch=FALSE.
</pre></div> </td></tr></table> <hr> <h2 id=fig-3b>Fig 3B<a class=headerlink href=#fig-3b title="Permanent link">&para;</a></h2> <blockquote> <p>Compare de-novo mutational signatures in canine and human pediatric and adult glioma with deconvoluted ones from Human Cancers (n=42).</p> </blockquote> <ul> <li> <p>For each of three cohorts (canine, human pediatric and adult glioma), de-novo signatures were constructed as follows:</p> </li> <li> <p>Resulting de-novo signatures from three cohorts were then merged into a single dataframe.</p> </li> <li> <p>Calculate cosine similarity using <code>deconvolution_compare</code> function in <a href=https://github.com/FunGeST/Palimpsest><em>Palimpsest</em></a> R package. Author of Palimpsest package has provided <a href=https://github.com/FunGeST/Palimpsest/raw/master/Files/vignette_palimpsest_2.0.pdf>a detailed documentation</a> on how to run Palimpsest workflow.</p> </li> </ul> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <hr> <h2 id=fig-3c>Fig 3C<a class=headerlink href=#fig-3c title="Permanent link">&para;</a></h2> <blockquote> <p>Plot estimated mutational signature contribution per driver gene.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>Joining, by = &quot;sigs&quot;
Joining, by = &quot;sigs&quot;
Joining, by = &quot;sigs&quot;
</pre></div> </td></tr></table> <hr> <h3 id=winning-tables>Winning Tables<a class=headerlink href=#winning-tables title="Permanent link">&para;</a></h3> <blockquote> <p>Make winning matrix of somatic variants in cancer driver genes (n=73) based on inferred clonality using <a href=https://github.com/FunGeST/Palimpsest><em>Palimpsest</em></a> R package. Author of Palimpsest package has provided <a href=https://github.com/FunGeST/Palimpsest/raw/master/Files/vignette_palimpsest_2.0.pdf>a detailed documentation</a> on how to run Palimpsest workflow.</p> </blockquote> <h2 id=fig-3d>Fig 3D<a class=headerlink href=#fig-3d title="Permanent link">&para;</a></h2> <blockquote> <p>Run <em>Bradley-Terry model</em> to get <a href=/methods/S14_mut_timing/ >estimated relative timing of somatic driver events</a>.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Attaching package: &#39;igraph&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:rtracklayer&#39;:

    blocks, path
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:GenomicRanges&#39;:

    union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:IRanges&#39;:

    union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:S4Vectors&#39;:

    union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:NMF&#39;:

    algorithm, compare
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:BiocGenerics&#39;:

    normalize, path, union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:dplyr&#39;:

    as_data_frame, groups, union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:purrr&#39;:

    compose, simplify
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:tidyr&#39;:

    crossing
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:tibble&#39;:

    as_data_frame
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following objects are masked from &#39;package:stats&#39;:

    decompose, spectrum
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:base&#39;:

    union
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Attaching package: &#39;ggridges&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:ggplot2&#39;:

    scale_discrete_manual
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class=code><div class=codehilite><pre><span></span>Number of items: 36 
Density of wins matrix: 0.2114198 
Fully-connected: FALSE 
Number of fully-connected components: 15 
Summary of fully-connected components: 
  Component size Freq
1              1   10
2              2    2
3              4    1
4              9    2
  Component size Freq
1              1   10
2              2    2
3              4    1
4              9    2
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead

Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class=code><div class=codehilite><pre><span></span>Number of items: 53 
Density of wins matrix: 0.1893912 
Fully-connected: FALSE 
Number of fully-connected components: 35 
Summary of fully-connected components: 
  Component size Freq
1              1   30
2              2    1
3              3    3
4             12    1
  Component size Freq
1              1   30
2              2    1
3              3    3
4             12    1
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead

Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17</pre></div></td><td class=code><div class=codehilite><pre><span></span>Number of items: 41 
Density of wins matrix: 0.2064247 
Fully-connected: FALSE 
Number of fully-connected components: 18 
Summary of fully-connected components: 
  Component size Freq
1              1   13
2              2    2
3              4    1
4              7    1
5             13    1
  Component size Freq
1              1   13
2              2    2
3              4    1
4              7    1
5             13    1
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3
4
5</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead

Warning in ref_check(ref, pi): The value of &#39;ref&#39; is not an item name. Using ref
= NULL instead
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Picking joint bandwidth of 0.0925
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Picking joint bandwidth of 0.118
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Picking joint bandwidth of 0.057
</pre></div> </td></tr></table> <hr> <h2 id=fig-4>Fig 4<a class=headerlink href=#fig-4 title="Permanent link">&para;</a></h2> <blockquote> <p>DNA methylation based classification of canine glioma.</p> </blockquote> <h3 id=preprocessing>Preprocessing<a class=headerlink href=#preprocessing title="Permanent link">&para;</a></h3> <h4 id=format-data-for-plotting>Format data for plotting<a class=headerlink href=#format-data-for-plotting title="Permanent link">&para;</a></h4> <h4 id=create-table-for-plotting-annotations>Create table for plotting annotations<a class=headerlink href=#create-table-for-plotting-annotations title="Permanent link">&para;</a></h4> <h4 id=labelscolors-for-plot>Labels/colors for plot<a class=headerlink href=#labelscolors-for-plot title="Permanent link">&para;</a></h4> <h4 id=plot-probabilities-and-predictions-data-frame-simultaneously>Plot probabilities and predictions data frame simultaneously<a class=headerlink href=#plot-probabilities-and-predictions-data-frame-simultaneously title="Permanent link">&para;</a></h4> <h4 id=add-annotations-for-mutation-rate-purity-and-age>Add annotations for mutation rate, purity, and age<a class=headerlink href=#add-annotations-for-mutation-rate-purity-and-age title="Permanent link">&para;</a></h4> <h3 id=fig-4_1>Fig 4<a class=headerlink href=#fig-4_1 title="Permanent link">&para;</a></h3> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Loading required package: magrittr
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Attaching package: &#39;magrittr&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:purrr&#39;:

    set_names
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:tidyr&#39;:

    extract
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>Attaching package: &#39;ggpubr&#39;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:egg&#39;:

    ggarrange
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2
3</pre></div></td><td class=code><div class=codehilite><pre><span></span>The following object is masked from &#39;package:cowplot&#39;:

    get_legend
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>Warning in as_grob.default(plot): Cannot convert object of class unit into a
grob.
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>$`1`
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>$`2`
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>attr(,&quot;class&quot;)
[1] &quot;list&quot;      &quot;ggarrange&quot;
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=suppl-fig-4a>Suppl Fig 4A<a class=headerlink href=#suppl-fig-4a title="Permanent link">&para;</a></h3> <blockquote> <p>Somatic mutational lanscape, including copy-number aberrations of COSMIC cancer genes, grouped by <a href=/methods/S15_class_predict_methy/ >DNA methylation classification</a>. </p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=suppl-fig-4b>Suppl Fig 4B<a class=headerlink href=#suppl-fig-4b title="Permanent link">&para;</a></h3> <blockquote> <p>DNA methylation age of human and canine glioma.</p> </blockquote> <p>Brewing...</p> <hr> <h2 id=fig-5>Fig 5<a class=headerlink href=#fig-5 title="Permanent link">&para;</a></h2> <blockquote> <p>IHC Panels</p> </blockquote> <h3 id=required-metadata_2>Required metadata<a class=headerlink href=#required-metadata_2 title="Permanent link">&para;</a></h3> <blockquote> <p>There were 33029 total fields measured and 27 fields with missing data. Fields with missing positivies were removed from further analysis.</p> </blockquote> <ul> <li>Number of fields by marker</li> </ul> <p>Marker # Fields</p> <hr> <p>CD14 7298 CD163 6601 CD3 6654 CD79A 5993 IBA1 6456</p> <ul> <li> <p>Counts of number of fields by marker for each sample:</p> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>       Type         Grade       CD14    CD163    CD3     CD79A    IBA1
</pre></div> </td></tr></table> <hr> <p>A1 Adult Low Grade 123 155 189 142 145<br> A10 Adult High Grade 520 675 485 783 792<br> A11 Adult High Grade 240 188 221 276 234<br> A2 Adult Low Grade 422 288 300 245 300<br> A3 Adult High Grade 392 482 362 343 525<br> A4 Adult High Grade 746 689 595 473 623<br> A5 Adult High Grade 695 432 354 357 372<br> A6 Adult High Grade 271 232 130 185 248<br> A7 Adult High Grade 467 489 387 404 227<br> A8 Adult Low Grade 558 687 548 615 467<br> A9 Adult High Grade 220 192 166 194 212<br> N09-531 Canine High Grade 189 149 148 127 166<br> N09-547 Canine High Grade 68 65 57 78 58<br> N09-651 Canine High Grade 22 18 15 40 21<br> N09-868 Canine High Grade 85 47 31 39 56<br> N10-159 Canine Low Grade 55 54 46 33 57<br> N10-702 Canine Low Grade 105 47 38 21 50<br> N12-166 Canine Low Grade 223 125 153 148 163<br> N13-901 Canine High Grade 225 170 181 140 149<br> N15-161 Canine High Grade 77 52 113 64 117<br> N15-604 Canine High Grade 67 70 75 68 60<br> N16-199 Canine High Grade 43 29 41 48 29<br> 11434 Pediatric High Grade 42 33 70 38 84<br> KS18-1269 Pediatric High Grade 480 492 434 468 460<br> KS19-607 Pediatric Low Grade 54 66 48 40 0 <br> S13-28076 Pediatric High Grade 362 375 1222 334 559<br> S14-12021 Pediatric High Grade 547 300 245 290 282 </p> </li> <li> <p>Counts of different types of samples (ped, adult, or breed):</p> </li> </ul> <p>Sample Type Number</p> <hr> <p>Adult 11 Boston terrier 4 Boxer 4 Lab 1 Mixed Dog? 1 Ped 5 Pit Bull 1</p> <ul> <li>Marker Distributions</li> </ul> <blockquote> <p>For each field, the "1+", "2+", and "3+" columns positivity are summed to create a new variable Positivity. The mean of the Positivity is computed across all fields for a given sample and marker. The means are shown in the table below.</p> </blockquote> <p>Sample.Name type Grade CD14 CD163 CD3 CD79A IBA1</p> <hr> <p>A1 Adult Low Grade 0.1 0.6 2.9 0.0 14.6 A10 Adult High Grade 0.2 0.4 2.5 0.0 20.6 A11 Adult High Grade 0.1 1.6 0.3 0.0 22.5 A2 Adult Low Grade 0.0 0.6 3.9 0.0 22.5 A3 Adult High Grade 0.8 0.9 0.2 0.0 42.5 A4 Adult High Grade 1.3 7.0 0.7 0.4 34.0 A5 Adult High Grade 1.6 24.7 2.0 0.0 38.9 A6 Adult High Grade 0.1 0.1 0.5 0.0 9.4 A7 Adult High Grade 0.1 5.7 0.3 0.0 31.9 A8 Adult Low Grade 0.2 3.0 1.7 0.2 14.9 A9 Adult High Grade 0.1 0.1 0.2 0.0 18.5 N09-531 Canine High Grade 0.1 0.0 0.3 0.1 34.1 N09-547 Canine High Grade 0.3 20.0 0.7 0.1 28.9 N09-651 Canine High Grade 0.0 15.2 4.6 0.2 28.1 N09-868 Canine High Grade 3.3 12.4 0.7 0.0 23.6 N10-159 Canine Low Grade 8.9 11.8 2.5 0.0 24.4 N10-702 Canine Low Grade 1.9 21.2 0.3 0.0 23.6 N12-166 Canine Low Grade 0.1 5.1 1.0 0.0 40.3 N13-901 Canine High Grade 2.6 0.3 0.5 0.2 6.2 N15-161 Canine High Grade 0.6 2.6 0.7 0.1 10.6 N15-604 Canine High Grade 0.6 9.1 4.9 0.0 30.6 N16-199 Canine High Grade 24.5 4.1 0.1 0.0 3.3 11434 Pediatric High Grade 0.0 57.9 0.5 0.1 37.3 KS18-1269 Pediatric High Grade 0.0 16.7 0.4 0.3 45.6 KS19-607 Pediatric Low Grade 5.2 46.1 0.1 0.1 NA S13-28076 Pediatric High Grade 0.0 11.1 0.1 0.2 76.8 S14-12021 Pediatric High Grade 0.1 23.8 0.0 0.1 54.6</p> <h3 id=fig-5b>Fig 5B<a class=headerlink href=#fig-5b title="Permanent link">&para;</a></h3> <blockquote> <p>The violin plots below show the distribution of percentage positivity. The violins are constructed using all the fields. The points are subject means. The markers CD14, CD3, and CD79A are low across grade and subject type.</p> </blockquote> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>`stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.
</pre></div> </td></tr></table> <p><img alt src=run_all_files/figure-html/unnamed-chunk-43-1.png><!-- --></p> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>`stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.
</pre></div> </td></tr></table> <p><img alt src=run_all_files/figure-html/unnamed-chunk-43-2.png><!-- --></p> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1</pre></div></td><td class=code><div class=codehilite><pre><span></span>`stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.
</pre></div> </td></tr></table> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <h3 id=p-values>p-values<a class=headerlink href=#p-values title="Permanent link">&para;</a></h3> <blockquote> <p>For CD163 and IBA1, mean percentage positivities are compared across sample types (Pediatric, Adult, Canine) within high grade patients. Specifically a Wilcoxon two sample p--value is computed for Pediatric vs. Canine subject means and Pediatric vs. Adult subject means.</p> </blockquote> <p>Marker Group 1 Group 2 p-value</p> <hr> <p>IBA1 Pediatric Canine 0.0040 IBA1 Pediatric Adult 0.0162 CD163 Pediatric Canine 0.0485 CD163 Pediatric Adult 0.0283</p> <hr> <h2 id=fig-5_1>Fig 5<a class=headerlink href=#fig-5_1 title="Permanent link">&para;</a></h2> <blockquote> <p><a href=/methods/S16_expression_immune/#cibersort-based-expression-analysis>CIBERSORT based deconvolution of immune microenvironment</a> in human and canine glioma. </p> </blockquote> <h3 id=required-metadata_3>Required metadata<a class=headerlink href=#required-metadata_3 title="Permanent link">&para;</a></h3> <h3 id=fig-s5>Fig S5<a class=headerlink href=#fig-s5 title="Permanent link">&para;</a></h3> <table class=codehilitetable><tr><td class=linenos><div class=linenodiv><pre><span></span>1
2</pre></div></td><td class=code><div class=codehilite><pre><span></span>quartz_off_screen 
                2 
</pre></div> </td></tr></table> <p>Last updated: February 8, 2020</p> <h2 id=__comments>Comments</h2> <div class="admonition bug"> <p class=admonition-title>Issues with code ?</p> <p>Please limit comments below for general questions. For code-related query, prefer <a href=https://github.com/TheJacksonLaboratory/canineglioma/issues>submitting a GitHub issue</a>.</p> </div> <div id=disqus_thread></div> <script>
    var disqus_config = function () {
      this.page.url = "https://canineglioma.verhaaklab.com/run_code/run_all/";
      this.page.identifier =
        "/run_code/run_all/";
    };
    (function() {
      var d = document, s = d.createElement("script");
      s.src = "//verhaaklab.disqus.com/embed.js";
      s.setAttribute("data-timestamp", +new Date());
      (d.head || d.body).appendChild(s);
    })();
  </script> </article> </div> </div> </main> <footer class=md-footer> <div class=md-footer-nav> <nav class="md-footer-nav__inner md-grid"> <a href=../ title=How-To class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel=prev> <div class="md-flex__cell md-flex__cell--shrink"> <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i> </div> <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"> <span class=md-flex__ellipsis> <span class=md-footer-nav__direction> Previous </span> How-To </span> </div> </a> </nav> </div> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-footer-copyright> <div class=md-footer-copyright__highlight> Copyright &copy; 2017 - 2020 <a href=https://verhaaklab.com>Verhaak Lab at the Jackson Laboratory for Genomic Medicine</a> </div> powered by <a href=https://www.mkdocs.org>MkDocs</a> and <a href=https://squidfunk.github.io/mkdocs-material/ > Material for MkDocs</a> </div> <div class=md-footer-social> <link rel=stylesheet href=../../assets/fonts/font-awesome.css> <a href=https://verhaaklab.com/contact class="md-footer-social__link fa fa-envelope"></a> <a href=https://twitter.com/roelverhaak class="md-footer-social__link fa fa-twitter"></a> <a href=https://github.com/TheJacksonLaboratory/canineglioma class="md-footer-social__link fa fa-github"></a> <a href=https://twitter.com/sbamin class="md-footer-social__link fa fa-twitter"></a> </div> </div> </div> </footer> </div> <script src=../../assets/javascripts/application.ffb616e8.js></script> <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script> <script src=../../assets/js/katex.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js></script> <script src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js></script> <script src=../../search/main.js></script> </body> </html>